<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>miRetrieve • miRetrieve</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="miRetrieve">
<meta property="og:description" content="miRetrieve">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">miRetrieve</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/miRetrieve.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="miRetrieve_files/header-attrs-2.9/header-attrs.js"></script><script src="miRetrieve_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="miRetrieve_files/plotly-binding-4.9.4.1/plotly.js"></script><script src="miRetrieve_files/typedarray-0.1/typedarray.min.js"></script><script src="miRetrieve_files/jquery-3.5.1/jquery.min.js"></script><link href="miRetrieve_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="miRetrieve_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><link href="miRetrieve_files/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="miRetrieve_files/plotly-main-1.57.1/plotly-latest.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>miRetrieve</h1>
            <h3 class="subtitle">microRNA text mining in PubMed abstracts</h3>
                        <h4 class="author">Julian Friedrich<sup>1,2</sup>, Hans-Peter Hammes<sup>2, 3</sup>, Guido Krenning<sup>1</sup><br><br><font size="2"> <sup>1</sup>Cardiovascular Regenerative Medicine (CAVAREM), Department of Pathology and Medical Biology, University Medical Center Groningen, University of Groningen, The Netherlands. <br><sup>2</sup>Section of Endocrinology, 5th Medical Department, Medical Faculty of Mannheim, University of Heidelberg, Mannheim, Germany. <br><sup>3</sup>European Center of Angioscience, Medical Faculty Mannheim, Heidelberg University, Germany</font>
</h4>
            
            <h4 class="date">2021-08-12</h4>
      
      
      <div class="hidden name"><code>miRetrieve.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>miRetrieve is an R package designed to facilitate text mining with microRNAs (miRNAs) in PubMed abstracts. By extracting miRNA names from large amounts of text, miRetrieve is able to provide insights into thousands of articles within a short amount of time.</p>
<p>In this vignette, we describe how to use miRetrieve. First, we are going to illustrate the mechanisms underlying miRetrieve by introducing basic tools of text analysis, which is a part of <em>Natural Language Processing</em> (NLP). Next, we are going to explain how to use the functions in miRetrieve, before applying miRetrieve in a case study.</p>
</div>
<div id="natural-language-processing" class="section level1">
<h1 class="hasAnchor">
<a href="#natural-language-processing" class="anchor"></a>Natural Language Processing</h1>
<p><strong>Natural Language Processing (NLP)</strong> describes the application of computational methods to process and analyze language<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<p>In this section, we are going to present common tools in NLP, namely <em>tokenization</em>, <em>tf-idf</em>, <em>stop word removal</em>, and <em>topic modeling</em>. These tools aid in gaining and extracting insights from a collection of texts, which are contained in a <strong>corpus</strong>.<br>
While this list is not exhaustive, it illustrates the mechanisms underlying many functions in miRetrieve and makes their use transparent.</p>
<p>To demonstrate these techniques, we are going to use a toy dataset, generated from abstracts about <em>diabetes</em>. The abstracts have a heavy focus on miR-34 and include some sampled abstracts and therefore do not represent the real data on miRNAs in diabetes.</p>
<div id="regex" class="section level2">
<h2 class="hasAnchor">
<a href="#regex" class="anchor"></a>Regular expressions</h2>
<p>Regular expressions are “a sequence of characters that specify a search pattern”<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<p>In miRetrieve, regular expressions help to extract miRNA names in a multi-steps process: First, minor modifications such as inserting a “-“ between any combination of “RNA” and a digit are made to prepare further extraction. Afterwards, miRNA-like strings are extracted with the regular expression</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="st">'[Mm][Ii][Cc]?[Rr][Oo]?[Rr]?[Nn]?[Aa]?[\\‑\\-\\s]?\\d+[a-zA-Z]?[\\‑\\-]?[12345]?[\\‑\\-]?[35]?[Pp]?'</span></code></pre></div>
<p>while let-7 is extracted using the regular expression</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="st">'[Ll][Ee][Tt][\\‑\\-\\s]?\\d+[a-zA-Z]?[\\‑\\-]?[35]?[Pp]?'</span></code></pre></div>
<p>Then, summarized (e.g. <em>miR-33a/b</em>) and chained miRNA names (e.g. <em>miR-34/-27/-29</em>) are separated, and possible -3, -5 suffices or trailing letters are removed.</p>
<p>What remains are miRNAs that are a combination of <em>miR-/miRNA-/MIR-/microRNA/… + number + (trailing letter)</em>. The miRNA numbers and trailing letters are extracted and joined with “miR-“ to unify the name. For let-7, any combination of <em>let-/LET-/Let/…</em> is converted to lower case “let-“.</p>
<p>Finally, miRNA names of older miRBase versions are converted to the most recent miRBase<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a> version (version 22 as of time of writing) (e.g. <em>miR-97, miR-102, miR-180(a/b)</em> become <em>miR-30a, miR-29a, and miR-172(a/b)</em>, respectively).</p>
</div>
<div id="tokenization" class="section level2">
<h2 class="hasAnchor">
<a href="#tokenization" class="anchor"></a>Tokenization</h2>
<p><strong>Tokenization</strong> refers to splitting text into smaller pieces, called <em>tokens</em>.</p>
<p>While text can be tokenized in many different ways, two common approaches are <strong>single word tokenization</strong> and <strong>n-gram tokenization</strong> (Fig. @ref(fig:tokenize)).<br>
Whereas single word tokenization splits text into single words (Fig. @ref(fig:tokenize) <strong>A</strong>), n-gram tokenization splits text into each combination of <em>n</em> adjacent words, which are referred to as <em>2-grams</em>, <em>3-grams</em> etc. (Fig. @ref(fig:tokenize) <strong>B</strong>).<br>
During tokenization, capital letters are transformed into lower case letters and punctuation such as <strong>.</strong> , <strong>,</strong> , or <strong>-</strong> is substituted with a space, if not specified otherwise. As a result, terms such as <em>T1DM</em> are transformed into <em>t1dm</em>, while compound terms such as <em>low-density</em> are tokenized into <em>low</em> and <em>density</em>.</p>
<div class="figure">
<img src="Tokenization_miRetrieve_res.png" alt=""><p class="caption"><font size="1"><b>Tokenization in NLP.</b> Tokenization refers to splitting text into smaller pieces, called tokens. Two common tokenization approaches are single word tokenization <b>(A)</b> and n-gram tokenization, splitting text into each combination of n adjacent words such as 2-grams <b>(B)</b></font>.</p>
</div>
<p> After a text is tokenized, the frequency of each token can provide first insights into the overall subject of the text. If <em>inflammation</em> is one of the most frequent tokens in a text, it can also be assumed that <em>inflammation</em> partially describes the topic of the text. Furthermore, token frequency can be compared between texts and serve as a starting point to determine text (dis)similarity.</p>
<p>While single word tokenization makes text comparison easy, complex n-grams may impede direct comparison. To meaningfully compare the token frequency of the 2-gram <em>low grade</em>, for example, requires many texts to have the exact same combination of <em>low</em> and <em>grade</em> multiple times, whereas it is clear that the chance of same word combinations decreases when n-gram size increases.</p>
<p>However, n-grams shine when word context matters. Single word tokenization of <em>Low grade inflammation</em> (<em>low</em>, <em>grade</em>, <em>inflammation</em>) loses the context of <em>low</em>, and if <em>low</em> is used in the context of <em>low grade inflammation</em>, <em>low expression</em>, or even <em>low-density lipoprotein</em> is not clear. After tokenizing <em>Low grade inflammation</em> into 2-grams, however, the token <em>low grade</em> hints that <em>low</em> is neither associated with <em>expression</em> nor with <em>low-density lipoprotein</em>, which already provides more insight into the context of <em>low</em>.</p>
<p>miRetrieve uses tokenization to generate term associations from text. Per miRNA, all abstracts mentioning this miRNA are tokenized, and stop words are removed. The tokens/terms are then displayed in decreasing frequency, thus displaying terms commonly used in combination with this miRNA throughout different abstracts.</p>
<p>When a function depends on tokenization in miRetrieve, the tokenization type can usually be set via the <code>token</code> argument. <code>token = "words"</code> performs single word tokenization, while <code>token = "ngrams"</code> and a specified <code>n</code> argument performs n-gram tokenization.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Analyze miRNA-term association of miR-34 by single word tokenization</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
               mir <span class="op">=</span> <span class="st">"miR-34"</span>,
               token <span class="op">=</span> <span class="st">"words"</span><span class="op">)</span>

<span class="co"># Analyze miRNA-term association of miR-34 by 2-gram tokenization</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
               mir <span class="op">=</span> <span class="st">"miR-34"</span>,
               token <span class="op">=</span> <span class="st">"ngrams"</span>,
               n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/tokenizeexample-1.png" width="50%"><img src="miRetrieve_files/figure-html/tokenizeexample-2.png" width="50%"></p>
</div>
<div id="tf-idf" class="section level2">
<h2 class="hasAnchor">
<a href="#tf-idf" class="anchor"></a>tf-idf</h2>
<p><strong>term frequency–inverse document frequency</strong>, or <strong>tf-idf</strong> in short, determines how unique and important a token is to one text compared to other texts.</p>
<p>Instead of comparing raw token frequency between texts, tokens are “weighed” depending on how often they are mentioned in one text compared to all other texts under investigation.<br>
When tokenizing the texts</p>
<ul>
<li><em>Cancer can affect any organ.</em></li>
<li><em>Cancer can be caused by mutations.</em></li>
<li><em>Cancer can be caused by viruses</em></li>
</ul>
<p>into single words, the tokens <em>cancer</em> and <em>can</em> are present in all three texts, while the tokens <em>be</em>, <em>caused</em>, and <em>by</em> are present in two out of three texts. The tokens <em>affect</em>, <em>any</em>, <em>organ</em>, <em>mutations</em>, and <em>viruses</em>, however, are only present in one text compared to all other texts. Taken together, this suggests that the tokens <em>cancer</em> and <em>can</em> offer no information when distinguishing these texts, while the text specific tokens <em>organ</em>, <em>mutations</em>, and <em>viruses</em> make the texts distinguishable and are thus more important for each text.</p>
<p>In miRetrieve, tf-idf can be used to determine how important a term is to a miRNA compared to other miRNAs in a corpus. If <em>inflammation</em> is only associated with miR-146, but not associated with miR-374 or miR-23, then <em>inflammation</em> is very specific for miR-146.<br>
When a miRetrieve function offers tf-idf analysis, such as <code><a href="../reference/plot_mir_terms.html">plot_mir_terms()</a></code> or <code><a href="../reference/plot_wordcloud.html">plot_wordcloud()</a></code>, it can be applied by setting <code>tf.idf = TRUE</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Analyze miRNA-term association of miR-34 with tf-idf</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
               mir <span class="op">=</span> <span class="st">"miR-34"</span>,
               tf.idf <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/tfidfexample-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="stop-words" class="section level2">
<h2 class="hasAnchor">
<a href="#stop-words" class="anchor"></a>Stop words</h2>
<p><strong>Stop words</strong> refer to common words that offer no information for text analysis, such as <em>a</em>, <em>is</em>, or <em>whether</em>, and are thus often removed (Fig. @ref(fig:stop)).</p>
<div class="figure">
<img src="Stopwordremoval_miRetrieve_res.png" alt=""><p class="caption"><font size="1"><b>Stop word removal.</b> Stop words refer to common words that offer no information for text analysis. It is therefore common practice to remove stop words in text analysis.</font></p>
</div>
<p><br> In miRetrieve, stop words are removed in two ways depending on tokenization type, namely stop word removal for single word tokenization and stop word removal for n-gram tokenization.</p>
<div id="stop-words-for-single-word-tokenization" class="section level3">
<h3 class="hasAnchor">
<a href="#stop-words-for-single-word-tokenization" class="anchor"></a>Stop words for single word tokenization</h3>
<p>To remove stop words for single word tokenization, stop words must be provided in a data frame. miRetrieve comes with predefined stop words, namely <code>stopwords_miretrieve</code>, which combines English stop words<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>, such as <em>a</em>, <em>is</em>, or <em>whether</em>, with stop words of PubMed abstracts, such as <em>western</em>, <em>qpcr</em>, or <em>significant</em>.</p>
<p>If only English stop words shall be removed, <code>stop_words</code> from the <strong>tidytext</strong> package can be used.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove common English and PubMed words with `stopwords_miretrieve`</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
               mir <span class="op">=</span> <span class="st">"miR-34"</span>,
               stopwords <span class="op">=</span> <span class="va">stopwords_miretrieve</span><span class="op">)</span>

<span class="co"># Remove common English words with `stop_words` from tidytext</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
               mir <span class="op">=</span> <span class="st">"miR-34"</span>,
               stopwords <span class="op">=</span> <span class="fu">tidytext</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/tidytext/man/stop_words.html">stop_words</a></span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/stopwordremoval_miretrieve_tidytext-1.png" width="50%"><img src="miRetrieve_files/figure-html/stopwordremoval_miretrieve_tidytext-2.png" width="50%"></p>
<p>Additionally, stop words can be generated from custom terms with <code><a href="../reference/generate_stopwords.html">generate_stopwords()</a></code>. These generated stop words can be added to an existing stop word data frame using the argument <code>combine_with</code>. Additionally, several stop words data frames can be combined with <code><a href="../reference/combine_stopwords.html">combine_stopwords()</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Vector of custom stop words</span>
<span class="va">custom_stopwords</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"these"</span>, <span class="st">"are"</span>, <span class="st">"some"</span>, <span class="st">"custom"</span>, <span class="st">"stop"</span>, <span class="st">"words"</span><span class="op">)</span>

<span class="co"># Generate custom stop words data frame</span>
<span class="co"># Combine custom stop words with `stopwords_miretrieve`</span>
<span class="va">custom_stopwords_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_stopwords.html">generate_stopwords</a></span><span class="op">(</span><span class="va">custom_stopwords</span>,
                                          combine_with <span class="op">=</span> <span class="va">stopwords_miretrieve</span><span class="op">)</span>

<span class="co"># Generate custom stop words data frame</span>
<span class="co"># Combine with `tidytext::stop_words` separately</span>
<span class="va">custom_stopwords_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_stopwords.html">generate_stopwords</a></span><span class="op">(</span><span class="va">custom_stopwords</span><span class="op">)</span>
<span class="va">stopwords_combined</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_stopwords.html">combine_stopwords</a></span><span class="op">(</span><span class="va">custom_stopwords_df</span>,
                                        <span class="fu">tidytext</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/tidytext/man/stop_words.html">stop_words</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="stop-words-for-n-gram-tokenization" class="section level3">
<h3 class="hasAnchor">
<a href="#stop-words-for-n-gram-tokenization" class="anchor"></a>Stop words for n-gram tokenization</h3>
<p>For n-gram tokenization, miRetrieve removes English stop words and PubMed stop words separately. First, English stop words<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a> are removed, and n-gram stop words are removed afterwards. The separation in stop word removal for English and n-gram stop words is explained by the increase in complexity with bigger n-grams.<br>
While English stop words are removed by setting <code>stopwords_ngram = TRUE</code>, n-grams to be removed must be provided in a data frame. <code>stopwords_miretrieve</code> removes common 2-grams, while stop words for bigger n-grams must be generated manually with <code><a href="../reference/generate_stopwords.html">generate_stopwords()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove English stop words for 2-gram tokenization</span>
<span class="co"># Remove PubMed 2-grams with `stopwords_miretrieve`</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
               mir <span class="op">=</span> <span class="st">"miR-34"</span>,
               token <span class="op">=</span> <span class="st">"ngrams"</span>,
               n <span class="op">=</span> <span class="fl">2</span>,
               stopwords <span class="op">=</span> <span class="va">stopwords_miretrieve</span>, 
               stopwords_ngram <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/stoptwogram-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="topic-modeling" class="section level2">
<h2 class="hasAnchor">
<a href="#topic-modeling" class="anchor"></a>Topic modeling</h2>
<p><strong>topic modeling</strong> describes the identification of topics in a corpus. Topics can either be identified in a supervised, e.g.  controlled manner, or in an unsupervised, e.g. blind manner.</p>
<div id="supervised-topic-modeling" class="section level3">
<h3 class="hasAnchor">
<a href="#supervised-topic-modeling" class="anchor"></a>Supervised topic modeling</h3>
<p><strong>Supervised topic modeling</strong> refers to identifying known topics in a corpus, and thus requires prior knowledge.</p>
<p>miRetrieve offers supervised topic modeling by heuristically identifying topics with keywords: First, a topic is defined by keywords, and these keywords are then used to calculate a <em>topic score</em> for each text in the corpus. The topic score reflects how well a text matches the keywords, and if the topic score surpasses a threshold, the text is considered to match the topic.</p>
<p>There are three pre-implemented heuristic topic models in miRetrieve, namely</p>
<ul>
<li>abstracts that investigate miRNAs in patients,</li>
<li>abstracts that investigate miRNAs in animal models,</li>
<li>abstracts that investigate miRNAs as biomarkers.</li>
</ul>
<p>While knowledge about how often a miRNA has been investigated in patients or in animal models can estimate the translation between bench and bedside, identifying which miRNA is most likely a biomarker aids in estimating its specificity compared to other fields.</p>
<p>For each heuristic model, a topic score can be calculated using <code><a href="../reference/calculate_score_patients.html">calculate_score_patients()</a></code>, <code><a href="../reference/calculate_score_animals.html">calculate_score_animals()</a></code>, and <code><a href="../reference/calculate_score_biomarker.html">calculate_score_biomarker()</a></code>. Furthermore, each <code>calculate_score_*()</code> function has a corresponding <code>plot_score_*()</code> function (<code><a href="../reference/plot_score_patients.html">plot_score_patients()</a></code>, <code><a href="../reference/plot_score_animals.html">plot_score_animals()</a></code>, and <code>plot_score_biomarkers()</code>), which plots the distribution of scores across all abstracts, and helps in choosing a threshold for topic assignment.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot score distribution, determine threshold</span>
<span class="fu"><a href="../reference/plot_score_patients.html">plot_score_patients</a></span><span class="op">(</span><span class="va">df_diabetes</span><span class="op">)</span>

<span class="co"># Calculate score for abstracts investigating miRNAs in patients</span>
<span class="va">df_patients</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_score_patients.html">calculate_score_patients</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
                                        threshold <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/plotscore-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>Next to the pre-implemented models, custom topics can be defined with custom keywords using <code><a href="../reference/calculate_score_topic.html">calculate_score_topic()</a></code> and its corresponding <code><a href="../reference/plot_score_topic.html">plot_score_topic()</a></code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Define keywords of custom topic "angiogenesis"</span>
<span class="va">keywords_angiogenesis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"angiogenesis"</span>, <span class="st">"vegf"</span>, <span class="st">"vascularization"</span>,
                           <span class="st">"sprouting"</span><span class="op">)</span>

<span class="co"># Plot distribution of "angiogenesis" scores</span>
<span class="fu"><a href="../reference/plot_score_topic.html">plot_score_topic</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
                 keywords <span class="op">=</span> <span class="va">keywords_angiogenesis</span>,
                 name.topic <span class="op">=</span> <span class="st">"Angiogenesis"</span><span class="op">)</span>

<span class="co"># Calculate angiogenesis score for each abstract</span>
<span class="va">df_angio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_score_topic.html">calculate_score_topic</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
                                  keywords <span class="op">=</span> <span class="va">keywords_angiogenesis</span>,
                                  threshold <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p>While one abstract can belong to multiple topics, abstracts can also be assigned to only one out of two or more topics: First, topic scores for all topics of interest are calculated, using <code><a href="../reference/calculate_score_topic.html">calculate_score_topic()</a></code>. Afterwards, each abstract is assigned to the topic where it surpasses a threshold and achieves the highest topic score,<br>
using <code><a href="../reference/assign_topic.html">assign_topic()</a></code>. If the topic score of an abstract does not surpass the threshold in any topic, the topic of the abstract is labelled as <code>"Unknown"</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Define keywords for type 1 diabetes</span>
<span class="va">keywords_t1dm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"pancreas"</span>, <span class="st">"beta cells"</span>, <span class="st">"gada"</span><span class="op">)</span>

<span class="co"># Define keywords for type 2 diabetes</span>
<span class="va">keywords_t2dm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"insulin resistance"</span>, <span class="st">"obesity"</span>, <span class="st">"metformin"</span><span class="op">)</span>

<span class="co"># Calculate type 1 diabetes scores for each abstract</span>
<span class="va">df_diabetes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_score_topic.html">calculate_score_topic</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
                                     keywords <span class="op">=</span> <span class="va">keywords_t1dm</span>,
                                     col.score <span class="op">=</span> <span class="st">"T1DM"</span><span class="op">)</span>

<span class="co"># Calculate type 2 diabetes scores for each abstract</span>
<span class="va">df_diabetes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_score_topic.html">calculate_score_topic</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
                                     keywords <span class="op">=</span> <span class="va">keywords_t2dm</span>,
                                     col.score <span class="op">=</span> <span class="st">"T2DM"</span><span class="op">)</span>

<span class="co"># Assign abstracts with a score of &gt;= 3 in "T1DM" to type 1 diabetes</span>
<span class="co"># Assign abstracts with a score of &gt;= 3 in "T2DM" to type 2 diabetes</span>
<span class="co"># Abstracts with a score &lt; 3 in "T1DM" and "T2DM" are assigned to</span>
<span class="co"># "Unknown".</span>
<span class="fu"><a href="../reference/assign_topic.html">assign_topic</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
             col.topic <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"T1DM"</span>, <span class="st">"T2DM"</span><span class="op">)</span>,
             threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="unsupervised-topic-modeling" class="section level3">
<h3 class="hasAnchor">
<a href="#unsupervised-topic-modeling" class="anchor"></a>Unsupervised topic modeling</h3>
<p><strong>Unsupervised topic modeling</strong> refers to identifying topics in a corpus with algorithms. As unsupervised topic modeling does not require prior knowledge, unsupervised topic modeling can be used to detect and uncover hidden topics in a corpus.</p>
<p>In miRetrieve, unsupervised topic modeling can be conducted with the Latent Dirichlet Algorithm (LDA), based on the <strong>topicmodels</strong> package<span class="citation">[<a href="#ref-topicmodels" role="doc-biblioref">2</a>]</span>.</p>
<p>To perform topic modeling with LDA, LDA requires the user to specify the number of topics. Based on different criteria and probability distributions, LDA then identifies as many topics as specified in the corpus, and assigns each text in the corpus a <em>topic probability</em> to belong to either topic. Ultimately, each text is assigned to the topic with its highest topic probability (Fig. @ref(fig:lda) <strong>A</strong>).<br>
Based on the texts within each topic, the subjects of the topics identified without supervision can then be determined by comparing their token frequency (Fig. @ref(fig:lda) <strong>B</strong>).</p>
<div class="figure">
<img src="LDA_miRetrieve_resized.png" alt=""><p class="caption"><font size="1"><b>Unsupervised topic modeling with LDA.</b> <b>(A)</b> LDA is fit on a corpus and identifies k = 3 topics based on different criteria. For each text in the corpus, a topic probability to belong to either topic is calculated and the text is assigned to the topic with the highest topic probability. <br><b>(B)</b> Tokenizing the texts in a topic and determining their most frequent tokens aids in inferring the main subject of discovered topics.</font></p>
</div>
<p><br> This whole process of identifying topics and calculating topic probabilities is referred to as <em>model fitting</em>. While an LDA model can be fit with <code><a href="../reference/fit_lda.html">fit_lda()</a></code>, the topics are ultimately assigned to each text with <code><a href="../reference/assign_topic_lda.html">assign_topic_lda()</a></code>. The subjects of the topics can be identified with <code>plot_lda_terms()</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Fit LDA model with k = 3 topics</span>
<span class="co"># Identify 3 topics in df</span>
<span class="va">lda_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_lda.html">fit_lda</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
                     k <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>

<span class="co"># Identify subject of topics</span>
<span class="fu"><a href="../reference/plot_lda_term.html">plot_lda_term</a></span><span class="op">(</span><span class="va">lda_model</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/fitlda-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>The first topic embraces <em>insulin</em>, <em>t2d(m)</em>, <em>metabolic</em>, and <em>liver</em> and might therefore summarize abstracts on diabetes type 2 and insulin resistance. The second topic, however, stresses <em>nephropathy</em>, <em>urinary</em>, and <em>dn</em> - a common abbreviation for <em>diabetic nephropathy</em> (or <em>neuropathy</em> in some cases) and likely encompasses abstracts about diabetic nephropathy. Finally, topic number 3 enlists <em>endothelial</em>, <em>heart</em>, and <em>dr</em>, an abbreviation for <em>diabetic retinopathy</em>, which could hint at abstracts about endothelial dysfunction in diabetes.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Assign LDA topics</span>
<span class="fu"><a href="../reference/assign_topic_lda.html">assign_topic_lda</a></span><span class="op">(</span><span class="va">df_diabetes</span>,
                 lda_model <span class="op">=</span> <span class="va">lda_model</span>,
                 topic.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Insulin resistance"</span>,
                                 <span class="st">"Nephropathy"</span>,
                                 <span class="st">"Endothelial dysfunction"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>As the optimal number of topics for LDA modeling is often unknown, one approach is to fit many LDA models, which differ in topic number, and to compare their <em>perplexity</em>. In LDA, perplexity measures how well a model fits the topics, while a lower perplexity corresponds to a better model. When comparing the perplexity of LDA models with different topic numbers, an increase in topic number often leads to a steep decrease in model perplexity at the beginning, indicating a model improvement with an increase in topic number. After a certain point, however, further increasing the topic number often leads to a marginal decrease in perplexity only, indicating that the model improves only marginally with an increase in topic number. The topic number where the decrease in perplexity starts to flatten is usually a good starting point for LDA modeling in practice.</p>
<p>In miRetrieve, the perplexity of different LDA models can be compared with <code><a href="../reference/plot_perplexity.html">plot_perplexity()</a></code>. <code><a href="../reference/plot_perplexity.html">plot_perplexity()</a></code> fits LDA models over different topic numbers and compares their perplexity in an <em>elbow plot</em>.</p>
<p>Using the elbow plot on our diabetes data set, <code><a href="../reference/plot_perplexity.html">plot_perplexity()</a></code> generates</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot perplexity for 2 to 5 topics</span>
<span class="co"># Identify optimal topic number</span>
<span class="fu"><a href="../reference/plot_perplexity.html">plot_perplexity</a></span><span class="op">(</span><span class="va">df_diabetes</span>, start <span class="op">=</span> <span class="fl">2</span>, end <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/perplexityplotexplanation-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>As the curve describes an “elbow” at <code>k = 3</code>, our previous choice with <code><a href="../reference/fit_lda.html">fit_lda(df, k = 3)</a></code> seems to have been reasonable.</p>
</div>
</div>
</div>
<div id="miretrieve-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#miretrieve-functions" class="anchor"></a>miRetrieve functions</h1>
<p>In the following section, we are going to describe how to use and combine the functions in miRetrieve.</p>
<p>First, we are going to outline how to install miRetrieve, before we explain how to load, prepare, and save data for and from analysis. Afterwards, we are going to illustrate how to analyze the miRNA landscape in one subject, before describing how to compare the miRNA landscape of multiple subjects.</p>
<p>To demonstrate miRetrieve’s functions, we are going to use another toy dataset that contains miRNAs in Colorectal Cancer (CRC). This dataset contains many abstracts on miR-21 combined with some sampled abstracts and therefore does not represent a real world cross-section on miRNAs in CRC.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install miRetrieve, run</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"miRetrieve"</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-prepare-and-save-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-prepare-and-save-data" class="anchor"></a>Load, prepare, and save data</h2>
<div id="load-data" class="section level3">
<h3 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h3>
<p>miRetrieve is optimized to work with PubMed abstracts in PubMed-format, which can be downloaded from PubMed via <em>Save</em> –&gt; <em>Selection: All results</em>, <em>Format: PubMed</em> –&gt; <em>Create File</em>.</p>
<p>The resulting PubMed-file can be loaded into R with <code><a href="../reference/read_pubmed.html">read_pubmed()</a></code> .</p>
<p>When loading abstracts with <code><a href="../reference/read_pubmed.html">read_pubmed()</a></code>, all abstracts can be assigned a <code>Topic</code> column, which denotes the subject of a file and facilitates miRNA comparison between topics. If a <code>Topic</code> column is not specified while loading, it can also be added with <code><a href="../reference/add_col_topic.html">add_col_topic()</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read in PubMed-file from colorectal cancer abstracts</span>
<span class="co"># Denote abstracts as "CRC"</span>
<span class="va">df_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="st">"CRC_PubMed.txt"</span>, topic <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span>

<span class="co"># Is the same as</span>
<span class="va">df_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="st">"CRC_PubMed.txt"</span><span class="op">)</span>
<span class="va">df_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_col_topic.html">add_col_topic</a></span><span class="op">(</span><span class="va">df</span>, topic.name <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span></code></pre></div>
<p>Multiple files can be combined into one data frame with <code><a href="../reference/combine_df.html">combine_df()</a></code>, which is crucial when comparing miRNAs of multiple topics.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load first PubMed-file</span>
<span class="va">df1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="st">"pubmedfile1.txt"</span>,
                           topic <span class="op">=</span> <span class="st">"cANCA"</span><span class="op">)</span>

<span class="co"># Load second PubMed-file</span>
<span class="va">df2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="st">"pubmedfile2.txt"</span>,
                           topic <span class="op">=</span> <span class="st">"pANCA"</span><span class="op">)</span>

<span class="co"># Combine df1 and df2</span>
<span class="va">df_large</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_df.html">combine_df</a></span><span class="op">(</span><span class="va">df1</span>, <span class="va">df2</span><span class="op">)</span></code></pre></div>
</div>
<div id="prepare-data" class="section level3">
<h3 class="hasAnchor">
<a href="#prepare-data" class="anchor"></a>Prepare data</h3>
<div id="subset-abstracts" class="section level4">
<h4 class="hasAnchor">
<a href="#subset-abstracts" class="anchor"></a>Subset abstracts</h4>
<p>Abstracts loaded into R can be subset for original research or review articles with <code><a href="../reference/subset_research.html">subset_research()</a></code> and <code><a href="../reference/subset_review.html">subset_review()</a></code> respectively. Furthermore, abstracts can also be subset for a specific publishing period with <code><a href="../reference/subset_year.html">subset_year()</a></code>.<br>
Subsetting abstracts with <code>subset_*()</code> keeps only abstracts of interest, while abstracts belonging to another article type or published outside the defined period are dropped.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Subset for abstracts of original research articles</span>
<span class="va">df_research</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_research.html">subset_research</a></span><span class="op">(</span><span class="va">df_crc</span><span class="op">)</span></code></pre></div>
</div>
<div id="extract-mirna-names" class="section level4">
<h4 class="hasAnchor">
<a href="#extract-mirna-names" class="anchor"></a>Extract miRNA names</h4>
<p>One of the core functions of miRetrieve the extraction of miRNA names from abstracts with <code><a href="../reference/extract_mir_df.html">extract_mir_df()</a></code>. miRNA names are extracted using <a href="#regex">regular expressions</a>, and extracted miRNA names are stored in a separate <code>miRNA</code> column, where each miRNA name occupies one row.</p>
<p>Next to extracting miRNA names from abstracts, miRNA names can also be extracted from single strings with <code><a href="../reference/extract_mir_string.html">extract_mir_string()</a></code>.</p>
<p>Both <code>extract_mir_*()</code> functions extract miRNA names either without or with a possible trailing letter (e.g. <em>miR-23</em> or <em>miR-23a</em>). As the use of miRNA nomenclature is rather inconsistent throughout literature, it is recommended to ignore trailing letters with <code>extract_letters = FALSE</code><a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a>.</p>
<p>Furthermore, <code>extract_mir_*()</code> has a <code>threshold</code> argument determining how often a miRNA must be in an abstract/string to be extracted. As miRNA names also change over time, deprecated miRNA names are automatically converted to the most recent miRBase version (e.g. miR-97, miR-102, miR-180(a/b) become miR-30a, miR-29a, and miR-172(a/b), respectively).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Extract miRNA names mentioned at least twice in an abstract without trailing letters</span>
<span class="va">df_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_mir_df.html">extract_mir_df</a></span><span class="op">(</span><span class="va">df_crc</span>,
                         threshold <span class="op">=</span> <span class="fl">2</span>,
                         extract_letters <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Extract miRNA names from a string with trailing letters</span>
<span class="fu"><a href="../reference/extract_mir_string.html">extract_mir_string</a></span><span class="op">(</span><span class="st">"miR-146a is an important miRNA in inflammation."</span>,
                   extract_letters <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>An exemplary output is</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">df_crc</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 8</span>
<span class="co">#&gt;      PMID  Year Title             Abstract           Language Type   Topic miRNA</span>
<span class="co">#&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;             &lt;chr&gt;              &lt;chr&gt;    &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;</span>
<span class="co">#&gt; 1  3.15e7  2019 MicroRNAs expres… Egypt has increas… eng      Journ… CRC   miR-…</span>
<span class="co">#&gt; 2  3.15e7  2019 In Vitro and In … Although chemothe… eng      Journ… CRC   miR-…</span>
<span class="co">#&gt; 3  3.15e7  2019 Assessment of Se… BACKGROUND: Micro… eng      Journ… CRC   miR-…</span>
<span class="co">#&gt; 4  3.15e7  2019 MicroRNA regulat… Colorectal cancer… eng      Journ… CRC   miR-…</span>
<span class="co">#&gt; 5  3.14e7  2019 MicroRNAs play a… Arsenic exposure … eng      Journ… CRC   miR-…</span>
<span class="co">#&gt; 6  3.13e7  2019 Faecal microRNAs… MicroRNAs (miRNAs… eng      Journ… CRC   miR-…</span></code></pre></div>
</div>
<div id="subset-and-indicate-mirna-names" class="section level4">
<h4 class="hasAnchor">
<a href="#subset-and-indicate-mirna-names" class="anchor"></a>Subset and indicate miRNA names</h4>
<p>After extracting miRNA names, abstracts can be subset for miRNAs with <code><a href="../reference/subset_mir.html">subset_mir()</a></code> and <code><a href="../reference/subset_mir_threshold.html">subset_mir_threshold()</a></code>.<br>
While <code><a href="../reference/subset_mir.html">subset_mir()</a></code> subsets abstracts for specified miRNAs, <code><a href="../reference/subset_mir_threshold.html">subset_mir_threshold()</a></code> subsets abstracts for miRNAs that are mentioned with a determined frequency. This frequency can either be an integer, corresponding to the minimal number of abstracts a miRNA is mentioned in, or it can be a decimal between 0 and 1, corresponding to the minimal relative number of abstracts a miRNA is mentioned in.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Keep only abstracts with miR-126 and miR-146</span>
<span class="va">df_mir126_miR_146</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_mir.html">subset_mir</a></span><span class="op">(</span><span class="va">df_crc</span>,
                                mir.retain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"miR-126"</span>, <span class="st">"miR-146"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Keep only abstracts with miRNAs mentioned in at least 5% of all abstracts</span>
<span class="va">df_five_ab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_mir_threshold.html">subset_mir_threshold</a></span><span class="op">(</span><span class="va">df_crc</span>,
                                   threshold <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></code></pre></div>
<p>Instead of subsetting abstracts for a specific miRNA, abstracts with a specific miRNA can also be labelled with <code><a href="../reference/indicate_mir.html">indicate_mir()</a></code>. Per specified miRNA name in <code><a href="../reference/indicate_mir.html">indicate_mir()</a></code>, a separate <em>Yes/No</em> column is added indicating the presence of the miRNA in the abstract.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Indicate abstracts with miR-126 and miR-146</span>
<span class="va">df_mir126_miR_146</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/indicate_mir.html">indicate_mir</a></span><span class="op">(</span><span class="va">df_crc</span>,
                                  indicate.mir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"miR-126"</span>, <span class="st">"miR-146"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Save data frame as an .xlsx file</span>
<span class="co"># Filter for miR-126 and miR-146 in excel</span>
<span class="fu"><a href="../reference/save_excel.html">save_excel</a></span><span class="op">(</span><span class="va">df_mir126_miR_146</span>,
           excel_file <span class="op">=</span> <span class="st">"df_mir_126146.xlsx"</span><span class="op">)</span></code></pre></div>
</div>
<div id="subset-data-frame" class="section level4">
<h4 class="hasAnchor">
<a href="#subset-data-frame" class="anchor"></a>Subset data frame</h4>
<p>While many functions of the <code>subset_*()</code> and <code>indicate_*()</code> family provide the possibility of subsetting a data frame, each data frame can also be individually subset with <code><a href="../reference/subset_df.html">subset_df()</a></code>.<br><code><a href="../reference/subset_df.html">subset_df()</a></code> is a wrapper of <strong>dplyr</strong>’s <code><a href="https://rdrr.io/r/stats/filter.html">filter()</a></code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Subset data frame with customized arguments</span>
<span class="fu"><a href="../reference/subset_df.html">subset_df</a></span><span class="op">(</span><span class="va">df_crc</span>,
          col.filter <span class="op">=</span> <span class="st">"miRNA"</span>,
          filter_for <span class="op">=</span> <span class="st">"miR-126"</span><span class="op">)</span>

<span class="co"># `subset_df()` is a more general version of</span>
<span class="fu"><a href="../reference/subset_mir.html">subset_mir</a></span><span class="op">(</span><span class="va">df_crc</span>, <span class="st">"miR-126"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="save-data" class="section level3">
<h3 class="hasAnchor">
<a href="#save-data" class="anchor"></a>Save data</h3>
<p>During analysis, any data frame or graph can be saved locally with <code><a href="../reference/save_excel.html">save_excel()</a></code> or <code><a href="../reference/save_plot.html">save_plot()</a></code> respectively.</p>
<div id="save-data-frame-to-excel" class="section level4">
<h4 class="hasAnchor">
<a href="#save-data-frame-to-excel" class="anchor"></a>Save data frame to excel</h4>
<p><code><a href="../reference/save_excel.html">save_excel()</a></code> saves a data frame as an .xlsx-file. When more than one data frame is passed to <code><a href="../reference/save_excel.html">save_excel()</a></code>, each data frame is saved as a separate work sheet in the same .xlsx-file.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Save df_crc and df_diabetes to the same .xlsx-file</span>
<span class="fu"><a href="../reference/save_excel.html">save_excel</a></span><span class="op">(</span><span class="va">df_crc</span>, <span class="va">df_diabetes</span>,
           excel_file <span class="op">=</span> <span class="st">"miRetrieve_df.xlsx"</span><span class="op">)</span></code></pre></div>
</div>
<div id="save-plots" class="section level4">
<h4 class="hasAnchor">
<a href="#save-plots" class="anchor"></a>Save plots</h4>
<p><code><a href="../reference/save_plot.html">save_plot()</a></code> saves the last generated plot, while the plot properties can be defined with <code>width</code>, <code>height</code>, and <code>dpi</code>. <code><a href="../reference/save_plot.html">save_plot()</a></code> is a wrapper of <strong>ggplot2</strong>’s <code>ggsave()</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Save last plot</span>
<span class="fu"><a href="../reference/save_plot.html">save_plot</a></span><span class="op">(</span><span class="st">"Last_plot.pdf"</span>,
          height <span class="op">=</span> <span class="fl">5</span>,
          width <span class="op">=</span> <span class="fl">7</span>,
          dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="extract-pubmed-ids" class="section level3">
<h3 class="hasAnchor">
<a href="#extract-pubmed-ids" class="anchor"></a>Extract PubMed-IDs</h3>
<p>PubMed-IDs can be extracted from a data frame with <code><a href="../reference/get_pmid.html">get_pmid()</a></code>. By default, <code><a href="../reference/get_pmid.html">get_pmid()</a></code> copies the PubMed-IDs to the clipboard, which can be used further outside R.<br>
Additionally, <code><a href="../reference/get_pmid.html">get_pmid()</a></code> can also extract PubMed-IDs as a string by setting <code>copy = FALSE</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Copy PubMed-IDs to clipboard</span>
<span class="fu"><a href="../reference/get_pmid.html">get_pmid</a></span><span class="op">(</span><span class="va">df_crc</span>,
         copy <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="mirna-text-mining-in-one-subject" class="section level2">
<h2 class="hasAnchor">
<a href="#mirna-text-mining-in-one-subject" class="anchor"></a>miRNA text mining in one subject</h2>
<p>The following section focuses on miRNA text mining in one subject, opposed to miRNA text mining in several subjects.</p>
<p>Here, we are going to describe how to count miRNAs and display their development. Next, we are going to explain how to display which terms a miRNA is associated with, before illustrating how to visualize which targets miRNAs regulate.</p>
<p>The functions in this section require the miRNAs names to be extracted with <code>extract_mir_*()</code>.</p>
<div id="count-mirnas" class="section level3">
<h3 class="hasAnchor">
<a href="#count-mirnas" class="anchor"></a>Count miRNAs</h3>
<p>How many abstracts mention a miRNA can be identified either with <code><a href="../reference/count_mir.html">count_mir()</a></code> or <code><a href="../reference/plot_mir_count.html">plot_mir_count()</a></code>. While <code><a href="../reference/count_mir.html">count_mir()</a></code> displays the miRNA count in a data frame, <code><a href="../reference/plot_mir_count.html">plot_mir_count()</a></code> visualizes the count of the most frequently mentioned miRNAs.</p>
<p><em>Again, please bear in mind that this dataset contains randomly sampled abstracts and many abstracts on miR-21.</em></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Count how many abstracts mention a miRNA</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/count_mir.html">count_mir</a></span><span class="op">(</span><span class="va">df_crc</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   miRNA   Mentioned_n</span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;int&gt;</span>
<span class="co">#&gt; 1 miR-21          267</span>
<span class="co">#&gt; 2 miR-145          10</span>
<span class="co">#&gt; 3 let-7             9</span>
<span class="co">#&gt; 4 miR-34            9</span>
<span class="co">#&gt; 5 miR-29            7</span>
<span class="co">#&gt; 6 miR-196           6</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot the count of the five most frequently mentioned miRNAs</span>
<span class="fu"><a href="../reference/plot_mir_count.html">plot_mir_count</a></span><span class="op">(</span><span class="va">df_crc</span>,
               top <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-12-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="count-exceeding-mirnas" class="section level3">
<h3 class="hasAnchor">
<a href="#count-exceeding-mirnas" class="anchor"></a>Count exceeding miRNAs</h3>
<p>Next to counting how many abstracts mention one miRNA, counting how many miRNAs are mentioned in a minimal number of abstracts can be done with <code><a href="../reference/count_mir_threshold.html">count_mir_threshold()</a></code> or <code><a href="../reference/plot_mir_count_threshold.html">plot_mir_count_threshold()</a></code>.</p>
<p>Counting how many miRNAs are mentioned in a minimal number of abstracts provides information if the majority of abstracts focus on a few miRNAs only, or if the interest in several miRNAs is evenly distributed across a field.</p>
<p><code><a href="../reference/count_mir_threshold.html">count_mir_threshold()</a></code> accepts a <code>threshold</code> argument and counts how many miRNAs are mentioned in at least <code>threshold</code> abstracts. <code>threshold</code> can either be an integer, counting how many miRNAs are mentioned in a minimal number of abstracts, or it can be a decimal between 0 and 1, counting how many miRNAs are mentioned in a relative number of abstracts compared to all abstracts.</p>
<p><code><a href="../reference/plot_mir_count_threshold.html">plot_mir_count_threshold()</a></code> displays the count of miRNAs over several thresholds. Similar to <code>count_mir_threshold</code>, the thresholds can either be integers or decimals.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Count how many miRNAs are mentioned in at least 5 abstracts</span>
<span class="fu"><a href="../reference/count_mir_threshold.html">count_mir_threshold</a></span><span class="op">(</span><span class="va">df_crc</span>,
                    threshold <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; [1] 10</span>

<span class="co"># Plot how many miRNAs are mentioned in at least 5 to 10 abstracts</span>
<span class="fu"><a href="../reference/plot_mir_count_threshold.html">plot_mir_count_threshold</a></span><span class="op">(</span><span class="va">df_crc</span>,
                         start <span class="op">=</span> <span class="fl">5</span>,
                         end <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-13-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="count-mirnas-per-year" class="section level3">
<h3 class="hasAnchor">
<a href="#count-mirnas-per-year" class="anchor"></a>Count miRNAs per year</h3>
<p>How often a miRNA was mentioned per year can be visualized with <code><a href="../reference/plot_mir_development.html">plot_mir_development()</a></code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot development of miR-21 until 2018</span>
<span class="fu"><a href="../reference/plot_mir_development.html">plot_mir_development</a></span><span class="op">(</span><span class="va">df_crc</span>,
                     mir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"miR-21"</span><span class="op">)</span>,
                     end <span class="op">=</span> <span class="fl">2018</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-14-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="count-new-mirnas-per-year" class="section level3">
<h3 class="hasAnchor">
<a href="#count-new-mirnas-per-year" class="anchor"></a>Count new miRNAs per year</h3>
<p>How many miRNAs are mentioned for the first time in a year can be displayed with <code><a href="../reference/plot_mir_new.html">plot_mir_new()</a></code>.</p>
<p>Displaying how many miRNAs are mentioned for the first time in a year estimates the dynamism of a field, e.g. if recent abstracts mention miRNAs previously not reported, or if recent abstracts focus on miRNAs already mentioned in previous years.</p>
<p><code><a href="../reference/plot_mir_new.html">plot_mir_new()</a></code> also provides a <code>threshold</code> argument determining in how many abstracts of a year a miRNA must be mentioned to be considered <em>mentioned</em>. By setting a <code>threshold</code>, miRNAs that are only sparsely mentioned in a year are ignored.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot newly mentioned miRNAs per year until 2018;</span>
<span class="co"># miRNAs need to be reported in at least 3 abstracts/year</span>
<span class="co"># to be considered "mentioned"</span>
<span class="fu"><a href="../reference/plot_mir_new.html">plot_mir_new</a></span><span class="op">(</span><span class="va">df_crc</span>,
             threshold <span class="op">=</span> <span class="fl">3</span>,
             end <span class="op">=</span> <span class="fl">2018</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-15-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="associate-mirnas-with-terms" class="section level3">
<h3 class="hasAnchor">
<a href="#associate-mirnas-with-terms" class="anchor"></a>Associate miRNAs with terms</h3>
<p>Terms often associated with a miRNA can be visualized using <code><a href="../reference/plot_mir_terms.html">plot_mir_terms()</a></code>.</p>
<p>While <code><a href="../reference/plot_mir_terms.html">plot_mir_terms()</a></code> performs single word tokenization by default, <code><a href="../reference/plot_mir_terms.html">plot_mir_terms()</a></code> can also perform n-gram tokenization by setting <code>token  = ngrams</code> and specifying a separate <code>n</code> argument.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot top terms of miR-21</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_crc</span>,
               mir <span class="op">=</span> <span class="st">"miR-21"</span><span class="op">)</span>

<span class="co"># Plot top 2-grams of miR-21</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_crc</span>,
               mir <span class="op">=</span> <span class="st">"miR-21"</span>,
               token <span class="op">=</span> <span class="st">"ngrams"</span>,
               n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-16-1.png" width="50%"><img src="miRetrieve_files/figure-html/unnamed-chunk-16-2.png" width="50%"></p>
<p>Next to plotting the top terms as a bar plot, top terms can also be visualized as a word cloud with <code><a href="../reference/plot_wordcloud.html">plot_wordcloud()</a></code>.<a href="#fn7" class="footnote-ref" id="fnref7"><sup>7</sup></a></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Word cloud of miR-21</span>
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span><span class="op">(</span><span class="va">df_crc</span>,
               mir <span class="op">=</span> <span class="st">"miR-21"</span>,
               max.terms <span class="op">=</span> <span class="fl">40</span>,
               colours <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"lightblue"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1,118 x 3</span>
<span class="co">#&gt;        PMID word           n</span>
<span class="co">#&gt;       &lt;dbl&gt; &lt;chr&gt;      &lt;int&gt;</span>
<span class="co">#&gt;  1 17702597 fu             3</span>
<span class="co">#&gt;  2 17702597 neoplastic     3</span>
<span class="co">#&gt;  3 17968323 metastasis     3</span>
<span class="co">#&gt;  4 17968323 pdcd4         11</span>
<span class="co">#&gt;  5 18196926 crc            6</span>
<span class="co">#&gt;  6 18196926 tumor          3</span>
<span class="co">#&gt;  7 18196926 tumors         3</span>
<span class="co">#&gt;  8 18230780 cancer         4</span>
<span class="co">#&gt;  9 18230780 colon          4</span>
<span class="co">#&gt; 10 18230780 patients       3</span>
<span class="co">#&gt; # … with 1,108 more rows</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-17-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="indicate-terms" class="section level3">
<h3 class="hasAnchor">
<a href="#indicate-terms" class="anchor"></a>Indicate terms</h3>
<p>Abstracts can be screened for terms with <code><a href="../reference/indicate_term.html">indicate_term()</a></code>. Per term, <code><a href="../reference/indicate_term.html">indicate_term()</a></code> signals it presence in an abstract with a separate <em>Yes/No</em> column.</p>
<p>How often a term must be in an abstract to be considered <em>present</em> can be controlled with a <code>threshold</code> argument. Furthermore, <code><a href="../reference/indicate_term.html">indicate_term()</a></code> can also keep only abstracts containing the term(s) of interest via the <code>discard</code> argument. A possible application is to keep abstracts that mention a certain drug, and to re-count the most frequent miRNAs in this subset.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Indicate and keep abstracts that mention "bevacizumab" at least twice</span>
<span class="va">abstracts_beva</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/indicate_term.html">indicate_term</a></span><span class="op">(</span><span class="va">df_crc</span>,
                                term <span class="op">=</span> <span class="st">"bevacizumab"</span>,
                                threshold <span class="op">=</span> <span class="fl">2</span>,
                                discard <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Count miRNAs in "bevacizumab" abstracts</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/count_mir.html">count_mir</a></span><span class="op">(</span><span class="va">abstracts_beva</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 2</span>
<span class="co">#&gt;   miRNA   Mentioned_n</span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;int&gt;</span>
<span class="co">#&gt; 1 miR-579           1</span></code></pre></div>
</div>
<div id="identify-mirna-targets" class="section level3">
<h3 class="hasAnchor">
<a href="#identify-mirna-targets" class="anchor"></a>Identify miRNA targets</h3>
<p>miRetrieve can integrate miRNA targets from excel files such as <em>miRTarBase</em><span class="citation">[<a href="#ref-mirtarbase" role="doc-biblioref">4</a>]</span> with <code><a href="../reference/join_targets.html">join_targets()</a></code>.</p>
<p><code><a href="../reference/join_targets.html">join_targets()</a></code> loads an excel-file with PubMed-IDs and miRNA targets and adds it to a miRetrieve data frame by matching their PubMed-IDs.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Adds targets from miRTarBase from an excel file (see "References") to df</span>
<span class="va">df_targets_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/join_targets.html">join_targets</a></span><span class="op">(</span><span class="va">df_crc</span>,
                               excel_file <span class="op">=</span><span class="st">"miRTarBase_MTI.xlsx"</span>,
                               col.pmid.excel <span class="op">=</span> <span class="st">"References (PMID)"</span>,
                               col.target.excel <span class="op">=</span> <span class="st">"Target Gene"</span>,
                               col.mir.excel <span class="op">=</span> <span class="st">"miRNA"</span><span class="op">)</span></code></pre></div>
<p>Additionally, miRetrieve comes with the latest miRTarBase version 8.0 and can be queried via <code><a href="../reference/join_mirtarbase.html">join_mirtarbase()</a></code>. If you use miRTarBase with miRetrieve, please make sure to cite <span class="citation">[<a href="#ref-mirtarbase" role="doc-biblioref">4</a>]</span>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Adds targets from miRTarBase (see "References") to df</span>
<span class="va">df_targets_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/join_mirtarbase.html">join_mirtarbase</a></span><span class="op">(</span><span class="va">df_crc</span>,
                                  col.topic.df <span class="op">=</span> <span class="va">Topic</span>,
                                  reduce <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>After adding the targets, target frequency can be counted with <code><a href="../reference/count_target.html">count_target()</a></code> or visualized with <code><a href="../reference/plot_target_count.html">plot_target_count()</a></code>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot target frequency</span>
<span class="fu"><a href="../reference/plot_target_count.html">plot_target_count</a></span><span class="op">(</span><span class="va">df_targets_crc</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-22-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>Furthermore, miRNA-target interactions can be plotted with <code><a href="../reference/plot_target_mir_scatter.html">plot_target_mir_scatter()</a></code>.<br><code><a href="../reference/plot_target_mir_scatter.html">plot_target_mir_scatter()</a></code> plots either the most frequently targeted genes, or it plots the top miRNAs targeting genes. If the focus shall be on the top targets or top targeting miRNAs, can be regulated via the <code>filter_for</code> argument.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot most frequently targeted genes</span>
<span class="fu"><a href="../reference/plot_target_mir_scatter.html">plot_target_mir_scatter</a></span><span class="op">(</span><span class="va">df_targets_crc</span>,
                        filter_for <span class="op">=</span> <span class="st">"target"</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-23-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="single-nucleotide-polymorphism" class="section level3">
<h3 class="hasAnchor">
<a href="#single-nucleotide-polymorphism" class="anchor"></a>Single Nucleotide Polymorphism</h3>
<p>Single Nucleotide Polymorphisms (SNPs) can be extracted from abstracts with <code><a href="../reference/extract_snp.html">extract_snp()</a></code>.</p>
<p><code><a href="../reference/extract_snp.html">extract_snp()</a></code> retrieves SNPs from abstracts and stores them in a column. Unlike <code><a href="../reference/extract_mir_df.html">extract_mir_df()</a></code>, however, all extracted SNPs of an abstract are stored in the same row. Furthermore, <code><a href="../reference/extract_snp.html">extract_snp()</a></code> can also subset abstracts containing SNPs via the <code>discard</code> argument.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Exctract SNPs</span>
<span class="co"># Keep only abstracts with SNPs</span>
<span class="va">snp_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_snp.html">extract_snp</a></span><span class="op">(</span><span class="va">df_crc</span>,
                      discard <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Extracted SNPs can be counted with <code><a href="../reference/count_snp.html">count_snp()</a></code>, while abstracts can be subset for specific SNPs with <code><a href="../reference/subset_snp.html">subset_snp()</a></code>.<br>
To facilitate filtering for SNPs, SNP names can be extracted from a data frame with <code><a href="../reference/get_snp.html">get_snp()</a></code>. <code><a href="../reference/get_snp.html">get_snp()</a></code> retrieves the string of a SNP by row, which can be passed to <code><a href="../reference/subset_snp.html">subset_snp()</a></code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Count SNPs</span>
<span class="va">snp_count_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/count_snp.html">count_snp</a></span><span class="op">(</span><span class="va">snp_df</span><span class="op">)</span>

<span class="co"># Extract SNP name in the second row of snp_count_df</span>
<span class="va">second_snp_string</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_snp.html">get_snp</a></span><span class="op">(</span><span class="va">snp_count_df</span>,
                             row <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="co"># Subset `snp_df` for abstracts containing `second_snp_string`</span>
<span class="fu"><a href="../reference/subset_snp.html">subset_snp</a></span><span class="op">(</span><span class="va">snp_df</span>,
           snp.retain <span class="op">=</span> <span class="va">second_snp_string</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="mirna-text-mining-in-multiple-subjects" class="section level2">
<h2 class="hasAnchor">
<a href="#mirna-text-mining-in-multiple-subjects" class="anchor"></a>miRNA text mining in multiple subjects</h2>
<p>Next to miRNA text mining in a single subject, miRetrieve also offers tools to compare the results of miRNA text mining in multiple subjects.</p>
<p>In this section, we are going to explain how to compare miRNA count and miRNA-term association, using <code>df_crc</code> and <code>df_diabetes</code> from previous examples. Furthermore, we are going to describe how to visualize miRNA-target interactions across fields.</p>
<div id="data-preparation" class="section level3">
<h3 class="hasAnchor">
<a href="#data-preparation" class="anchor"></a>Data preparation</h3>
<p>To compare different subjects, each subject is loaded separately with <code>read_pubmed_*()</code>. Furthermore, each field must be assigned a distinct topic name, using either the <code>topic</code> argument of <code>read_pubmed_*()</code> or <code><a href="../reference/add_col_topic.html">add_col_topic()</a></code>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load abstracts of diabetes</span>
<span class="va">df_diabetes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="st">"pubmed_diabetes.txt"</span>, topic <span class="op">=</span> <span class="st">"Diabetes"</span><span class="op">)</span>

<span class="co"># Load abstracts of colorectal cancer</span>
<span class="va">df_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="st">"pubmed_crc.txt"</span>, topic <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span></code></pre></div>
<p>Afterwards, all files are combined for further analysis with <code><a href="../reference/combine_df.html">combine_df()</a></code>.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Combine abstracts of topics</span>
<span class="va">df_combined</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_df.html">combine_df</a></span><span class="op">(</span><span class="va">df_diabetes</span>, <span class="va">df_crc</span><span class="op">)</span></code></pre></div>
</div>
<div id="extract-mirna-names-as-strings" class="section level3">
<h3 class="hasAnchor">
<a href="#extract-mirna-names-as-strings" class="anchor"></a>Extract miRNA names as strings</h3>
<p>A key difference to text mining in one subject is that the miRNAs to analyze must be specified.</p>
<p>For this, miRNA names can be extracted as strings from a data frame using <code><a href="../reference/get_mir.html">get_mir()</a></code>, <code>get_shared_mir*()</code>, <code>get_distinct_mir_*()</code>, and <code><a href="../reference/combine_mir.html">combine_mir()</a></code>.</p>
<ul>
<li>
<p><code><a href="../reference/get_mir.html">get_mir()</a></code> extracts either the most frequently mentioned miRNAs or the miRNAs that are mentioned in a minimum number of abstracts.<br>
Moreover, <code><a href="../reference/get_mir.html">get_mir()</a></code> can also extract the top miRNAs of one subject with a <code>topic</code> argument.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get top 5 miRNAs of all topics</span>
<span class="fu"><a href="../reference/get_mir.html">get_mir</a></span><span class="op">(</span><span class="va">df_combined</span>,
        top <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>

<span class="co"># Get top 5 miRNAs of the topic "CRC"</span>
<span class="fu"><a href="../reference/get_mir.html">get_mir</a></span><span class="op">(</span><span class="va">df_combined</span>,
        top <span class="op">=</span> <span class="fl">5</span>,
        topic <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span></code></pre></div>
</li>
<li>
<p><code>get_shared_mir*()</code> provides the most frequent miRNAs that are shared between two subjects. <code><a href="../reference/get_shared_mir_df.html">get_shared_mir_df()</a></code> extracts the shared miRNAs from a data frame, while <code><a href="../reference/get_shared_mir_vec.html">get_shared_mir_vec()</a></code> extracts the shared miRNAs from two character vectors.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get top shared miRNAs of "Diabetes" and "CRC"</span>
<span class="fu"><a href="../reference/get_shared_mir_df.html">get_shared_mir_df</a></span><span class="op">(</span><span class="va">df_combined</span>,
                  topic <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Diabetes"</span>, <span class="st">"CRC"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</li>
<li>
<p><code>get_distinct_mir*()</code> provides the most frequent miRNAs that are distinct for one topic, but do not belong to the top miRNAs of another topic. <code><a href="../reference/get_distinct_mir_df.html">get_distinct_mir_df()</a></code> extracts the distinct miRNAs from a data frame, while the topic to extract the distinct miRNAs from is determined with the <code>distinct</code> argument.<br><code><a href="../reference/get_distinct_mir_vec.html">get_distinct_mir_vec()</a></code> extracts the distinct miRNAs of the first character vector from two character vectors.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get top miRNAs distinct for "CRC" compared to the top miRNAs of "Diabetes"</span>
<span class="fu"><a href="../reference/get_distinct_mir_df.html">get_distinct_mir_df</a></span><span class="op">(</span><span class="va">df_combined</span>, 
                    distinct <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span></code></pre></div>
</li>
<li>
<p><code><a href="../reference/combine_mir.html">combine_mir()</a></code> combines character vectors with miRNA names into one vector.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get top 5 miRNAs of "CRC"</span>
<span class="va">top_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mir.html">get_mir</a></span><span class="op">(</span><span class="va">df_combined</span>,
                      top <span class="op">=</span> <span class="fl">5</span>,
                      topic <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span>

<span class="co"># Get top 5 miRNAs of "Diabetes"</span>
<span class="va">top_diabetes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mir.html">get_mir</a></span><span class="op">(</span><span class="va">df_combined</span>,
                      top <span class="op">=</span> <span class="fl">5</span>,
                      topic <span class="op">=</span> <span class="st">"Diabetes"</span><span class="op">)</span>

<span class="co"># Combine vectors</span>
<span class="va">top_combined</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_mir.html">combine_mir</a></span><span class="op">(</span><span class="va">top_crc</span>, <span class="va">top_diabetes</span><span class="op">)</span></code></pre></div>
</li>
</ul>
</div>
<div id="compare-mirna-count" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-mirna-count" class="anchor"></a>Compare miRNA count</h3>
<p>How many abstracts per subject mention a miRNA can be compared with <code><a href="../reference/compare_mir_count.html">compare_mir_count()</a></code>.</p>
<p><code><a href="../reference/compare_mir_count.html">compare_mir_count()</a></code> displays either the absolute number of abstracts per subject mentioning a miRNA, or it can display the relative number of abstracts per subject mentioning a miRNA, referring to the number of abstracts with a miRNA relative to all abstracts per subject.<br>
Furthermore, the relative count of miRNAs can be compared between two subjects on a log2-scale with <code><a href="../reference/compare_mir_count_log2.html">compare_mir_count_log2()</a></code>.</p>
<p><em>Please bear in mind that both datasets are randomly sampled. Additionally, the diabetes dataset has a heavy focus on miR-34, while the CRC dataset stresses miR-34. The following figures are therefore for illustration purposes only.</em></p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Compare miRNA frequency between topics</span>
<span class="fu"><a href="../reference/compare_mir_count.html">compare_mir_count</a></span><span class="op">(</span><span class="va">df_combined</span>,
                  mir <span class="op">=</span> <span class="va">top_combined</span><span class="op">)</span>

<span class="co"># Compare miRNA frequency between subjects on a log2-scale</span>
<span class="fu"><a href="../reference/compare_mir_count_log2.html">compare_mir_count_log2</a></span><span class="op">(</span><span class="va">df_combined</span>,
                       mir <span class="op">=</span> <span class="va">top_combined</span><span class="op">)</span>
<span class="co">#&gt; $data.frame</span>
<span class="co">#&gt; # A tibble: 6 x 4</span>
<span class="co">#&gt;   miRNA       CRC Diabetes log2_ratio</span>
<span class="co">#&gt;   &lt;fct&gt;     &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;</span>
<span class="co">#&gt; 1 let-7   0.0196   0.0253      -0.367</span>
<span class="co">#&gt; 2 miR-145 0.0218   0.00562      1.96 </span>
<span class="co">#&gt; 3 miR-21  0.582    0.0281       4.37 </span>
<span class="co">#&gt; 4 miR-29  0.0153   0.0253      -0.729</span>
<span class="co">#&gt; 5 miR-30  0.00654  0.0225      -1.78 </span>
<span class="co">#&gt; 6 miR-34  0.0196   0.399       -4.35 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $plot</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-33-1.png" width="50%"><img src="miRetrieve_files/figure-html/unnamed-chunk-33-2.png" width="50%"></p>
</div>
<div id="identify-unique-mirnas" class="section level3">
<h3 class="hasAnchor">
<a href="#identify-unique-mirnas" class="anchor"></a>Identify unique miRNAs</h3>
<p>Instead of comparing shared miRNAs, most frequent unique miRNAs of each topic can be displayed with <code><a href="../reference/compare_mir_count_unique.html">compare_mir_count_unique()</a></code>.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identify 5 most frequent unique miRNAs per topic</span>
<span class="fu"><a href="../reference/compare_mir_count_unique.html">compare_mir_count_unique</a></span><span class="op">(</span><span class="va">df_combined</span>,
                         top <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-34-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="compare-term-associations" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-term-associations" class="anchor"></a>Compare term associations</h3>
<p>There are three functions to compare miRNA-term associations across subjects, namely <code><a href="../reference/compare_mir_terms.html">compare_mir_terms()</a></code>, <code><a href="../reference/compare_mir_terms_log2.html">compare_mir_terms_log2()</a></code>, and <code><a href="../reference/compare_mir_terms_scatter.html">compare_mir_terms_scatter()</a></code>.</p>
<p>While <code><a href="../reference/compare_mir_terms.html">compare_mir_terms()</a></code> can compare the top term count of a miRNA over many subjects, <code><a href="../reference/compare_mir_terms_log2.html">compare_mir_terms_log2()</a></code> and <code><a href="../reference/compare_mir_terms_scatter.html">compare_mir_terms_scatter()</a></code> can compare the top term count of a miRNA over two subjects only.</p>
<p><code><a href="../reference/compare_mir_terms.html">compare_mir_terms()</a></code> plots the count of top miRNA-term associations, whereas<br><code><a href="../reference/compare_mir_terms_log2.html">compare_mir_terms_log2()</a></code> compares the miRNA-term association between two subjects on a log2-scale.<a href="#fn8" class="footnote-ref" id="fnref8"><sup>8</sup></a></p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Compare term frequency for miR-21 between topics</span>
<span class="fu"><a href="../reference/compare_mir_terms.html">compare_mir_terms</a></span><span class="op">(</span><span class="va">df_combined</span>,
                  mir <span class="op">=</span> <span class="st">"miR-21"</span>, 
                  top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>

<span class="co"># Compare term frequency for miR-21 between two topics on a log2-scale</span>
<span class="fu"><a href="../reference/compare_mir_terms_log2.html">compare_mir_terms_log2</a></span><span class="op">(</span><span class="va">df_combined</span>,
                       mir <span class="op">=</span> <span class="st">"miR-21"</span>,
                       top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; $data.frame</span>
<span class="co">#&gt; # A tibble: 23 x 5</span>
<span class="co">#&gt;    word         number_topic     CRC Diabetes log2_ratio</span>
<span class="co">#&gt;    &lt;fct&gt;               &lt;int&gt;   &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;</span>
<span class="co">#&gt;  1 blood                   2 0.0899       0.1     -0.154</span>
<span class="co">#&gt;  2 fibrosis                2 0.00375      0.2     -5.74 </span>
<span class="co">#&gt;  3 hemoglobin              2 0.00375      0.2     -5.74 </span>
<span class="co">#&gt;  4 progenitor              2 0.00375      0.2     -5.74 </span>
<span class="co">#&gt;  5 rats                    2 0.00375      0.2     -5.74 </span>
<span class="co">#&gt;  6 tubular                 2 0.00749      0.2     -4.74 </span>
<span class="co">#&gt;  7 tnf                     2 0.0112       0.2     -4.15 </span>
<span class="co">#&gt;  8 egfr                    2 0.0150       0.2     -3.74 </span>
<span class="co">#&gt;  9 pathogenesis            2 0.0337       0.2     -2.57 </span>
<span class="co">#&gt; 10 resistance              2 0.0449       0.2     -2.15 </span>
<span class="co">#&gt; # … with 13 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $plot</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-35-1.png" width="50%"><img src="miRetrieve_files/figure-html/unnamed-chunk-35-2.png" width="50%"></p>
<p>Finally, <code><a href="../reference/compare_mir_terms_scatter.html">compare_mir_terms_scatter()</a></code> creates a scatter plot, displaying the frequency of the shared miRNA-associated terms.<a href="#fn9" class="footnote-ref" id="fnref9"><sup>9</sup></a></p>
<p><em>Note: The created scatter plot might not be displayed correctly if the vignette is downloaded. To see its interactivity in action, please generate the vignette straight in R.</em></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Compare terms of miR-21 between two topics</span>
<span class="fu"><a href="../reference/compare_mir_terms_scatter.html">compare_mir_terms_scatter</a></span><span class="op">(</span><span class="va">df_combined</span>,
                          mir <span class="op">=</span> <span class="st">"miR-21"</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-de15714eb3d1e2020e58" style="width:700px;height:432.632880098888px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-de15714eb3d1e2020e58">{"x":{"data":[{"x":[-1.09449353411648,-0.108721221054746],"y":[-1.09449353411648,-0.108721221054746],"text":"intercept: 0<br />slope: 1","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(102,102,102,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[null,null,-1.04600986188743,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.951188311469741,null,null,-1.01334854129236,null,null,null,null,null,null,null,null,null,-0.978183092689142,-1.01392573106568,null,-1.01741590192541,-1.04968570170458,null,null,null,null,null,null,-1.03926375112496,null,null,null,null,null,-0.984856918617152,-1.04531979414169,null,null,null,null,null,-0.962951313517988,null,-0.96898765177466,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.957617441331968,null,null,null,null,null,null,null,-1.04618996256031,null,null,-0.99962270827964,-0.956063560536131,null,null,-1.01958601905499,null,null,null,null,null,null,null,null,null,null,-0.996683785459027,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.959669818147086,null,null,null,null,null,null,null,null,-1.02308619762771,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.04059512894601,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.0321136839455,null,null,-0.974452811735682,null,null,null,-0.965506916609593,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.952726979297586,null,-0.997175386501476,null,null,-1.02682512637693,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.98085384608712,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.02755698612891,-0.995675293821842,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.970788447000086,null,null,null,null,-0.997456632670946,null,null,null,null,null,null,null,null,-1.04545699011069,null,-1.03372403045651,null,null,null,-0.981899241032079,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.9850080732489,-0.962746128346771,null,null,-0.993041130108759,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.961778638185933,null,null,null,-1.01497616504785,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.153529053466643,-0.316214778994168,null,-1.00229862893466,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.995899333246052,null,null,null,null,null,-1.00055044039618,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.978716790466569,-0.960183750744909,-0.979860513820313,null,null,null,null,null,null,null,-0.695101774811786,null,null,null,null,null,-1.03266618160997,-1.01592076048255,null,null,null,-0.98424544245936,-0.970912445825525,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.98221948472783,null,null,-0.969944269838743,null,null,null,null,-1.00813597759698,-1.03803173869383,null,null,null,-0.745282283892897,-0.980054197390564,null,null,null,null,null,null,null,null,null,-1.03780487568583,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.01341259041801,null,-1.02312257632148,null,null,null,-0.993194985855371,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.72817338186041,-1.02799434238113,null,null,null,-0.982547934097238,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.02748108641244,null,null,null,null,null,-0.959332934534177,null,null,null,null,null,null,null,null,-1.0420818816172,null,null,null,null,null,null,null,-1.0131751973182,null,null,null,null,null,null,null,null,null,-0.95530731966719,-0.43938388021578,null,null,-0.981895634299144,null,null,null,-0.955476822797209,null,null,null,null,null,null,null,null,null,-0.651321453440045,null,null,null,null,null,null,null,null,null,null,-1.01313110354822,null,null,null,null,null,null,null,null,-0.983058196748607,null,-1.00634948599618,-0.978823459194973,-0.964459875063039,null,null,null,null,null,null,null,-1.02813413338736,null,-0.963493741676211,-1.03009011324029,null,null,-0.960392886446789,null,-0.706896347601008,null,null,null,null,null,null,null,-0.982572118774988,null,null,null,null,null,null,null,-1.0325319991447,null,null,null,null,null,null,null,null,null,-1.02594667377416,-0.665680150945713,null,null,null,-0.980023241345771,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.968339614290744,null,-1.04404664144386,-0.99620203806553,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.02533884162549,null,null,null,null,null,null,null,null,-0.991318556107581,null,null,-0.560095206543935,null,null,null,null,null,-0.979981297277845,null,null,null,null,-0.483904168627356,null,null,null,-1.02588926362805,null,null,-0.959905046154745,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.996479711402208,null,-0.964931944641285,null,null,null,null,null,null,null,null,null,null,null,null,-1.01353460238315,null,null,null,null,null,null,null,null,null,-0.988520259037614,-0.990029299841262,null,null,null,null,null,null,null,null,-0.970323106483556,null,null,-0.568328241923024,-0.693323081966531,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.992013621330261,null,null,null,null,-0.999341961042955,null,null,null,null,null,null,null,null,-0.952887921640649,null,null,null,null,null,null,null,-1.02958650069777,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.01272234281059,null,null,null,null,null,null,null,null,null,null,null,null,-1.02395243514329,null,null,null,null,null,null,null,null,-1.03111962710973,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.545023650176023,null,null,null,-0.989244480314664,null,null,null,null,null,-0.983741236082278,null,null,-0.991249393322505,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.732686531429221,-0.969891528482549,null,null,-1.01222936953418,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.985552437440492,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.02091631747317,null,null,null,null,null,null,null,null,null,-0.974545464501716,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.954066552058794,null,null,null,null,null,null,null,null,-1.02780693299137,null,null,null,null,null,-0.725660962496926,null,null,null,null,null,-1.02687149543781,null,null,null,null,null,null,null,null,null,-0.969910652772523,null,null,-1.04806044674478,null,null,null,null,null,null,null,null,null,null,-1.00710812627804,null,null,null,null,null,-0.971245885104872,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.958579202229157,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.951663538231514,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.976364591764286,null,null,-0.973586978088133,-1.00802986109629,null,null,null,null,null,null,-1.01451224666089,null,-1.01300551807508,null,null,null,null,null,null,null,null,null,-1.03380978966597,null,null,-0.744323155053002,null,null,-0.55070523129603,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.951594758173451,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.477552137163599,null,-0.979867882095277,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.961605917150155,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.977555721183307,-0.714677821520758,null,null,null,-0.717218567828689,null,-0.738754075325136,null,null,null,null,null,null,null,null,null,null,-1.04124607911799,-1.02447373338509,-1.03993866860401,-0.981602651788853,-1.01844986893702,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.965434943139553,-0.702797098483001,-0.998705557920039,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.489977534290535,null,null,null,null,null,null,null,null,null,null,-0.655898140815128,null,null,null,null,-1.02561603116337,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.00209334918763,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-1.04713627528399,null,null,null,null,null,null,null,null,null,-0.95263352135662,null,null,-0.693870104897809,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.966211546747945,null,-1.01486940924078,null,null,null,null,null,null,null,null,null,null,-0.971459925570525,null,null,null,null,null,null,null,null,null,null,null,null,-0.982441885024309,null,-0.999714916432276,null,-0.993502678023651,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.953444334957749,null,-0.977376236859709,null,null,null,null,null,-0.746765397956636,null,null,null,null,null,-0.693303563796957,null,null,null,null,null,null,null,null,null,null,null,-1.0031574459048,-0.975006542727351,-1.00727963990066,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.983219366637059,-0.492382390949075,null,null,null,null,-0.989697840576991,-1.03937986341771,null,null,null,null,null,-1.04965337407775,null,null,null,null,null,-0.958844006876461,null,null,null,null,null,null,null,null,null,-1.03171444248874,-0.48190919346003,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.995588388270699,null,null,null,null,null,null,-0.695363372216974,null,null,null,-0.963535451795906,-1.00161536419764,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.969396822387353,-1.03927690896671,null,null,null,null,null,-0.998071192856878,null,null,null,null,null,null,null,-0.725433855971363,null,-0.979991540871561,null,null,null,null,null,null,null,null,null,-0.951628101291135,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.951054369332269,null,null,null,null,-0.951487303571776,null,null,null,null,null,null,null,null,-0.984141256520525,null,null,null,null,null,null,null,-1.01325299015734,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.986716794734821,null,null,-0.952716196910478,-0.993592128809542,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.661503627618816,-0.47719092700923,-0.7435649962882,-0.958544351300225,-1.02735291786958],"y":[-2.37835155902811,-2.3541651782385,null,-2.39739715494596,-2.42778367295776,-2.36716426450674,-2.39853689913087,-2.14472275577138,-2.40105356527873,-2.3922391409791,-2.35726875404742,-2.43126837357794,-2.34962932994039,-2.37014253908218,-2.39270475556058,-2.42187781881456,null,-2.39821418305308,-2.37314219246999,null,-2.36912491843485,-2.42959859539204,-1.16583244173067,-1.45610953851171,-1.16679542445323,-1.18182570870274,-1.68673681969754,-2.35008606629361,-2.37684765682005,null,-2.4023370625955,-1.4526134828548,null,null,-2.40744615057864,-2.13785835349119,-2.41343551006083,-2.42539107983847,-2.36956353332297,-2.43188650967003,-2.37594136037313,-2.17070618235158,-2.18279644211872,-1.97499910759549,-1.37212570418417,-2.38179270506752,null,null,-2.41119604735085,-2.38831228805155,-2.12034724546249,-2.34948096319948,-2.14542333491265,-1.47685410021572,-2.37950336146549,null,-2.00151520014084,-2.38170269600031,-2.39949398478874,-1.9553954839827,-2.38202593734403,-2.34935661787548,-2.42423553441566,-2.35248007097197,-2.3563914936074,-2.42525173579723,-2.42939339613457,-2.42025657415305,-2.38165014277128,-2.43728340720207,-2.36077579397623,-2.42188239102633,-2.18875216763119,-2.12875831302373,-2.42004168627699,-2.1995077919739,-2.42314415446398,-2.40706888200165,-1.92780308105326,-2.43794308936139,-1.38936408488113,-2.44296543118925,-2.39412153675602,null,null,-2.39973739432552,-2.37341135249388,null,-1.86524470032761,-2.42160991521467,-1.39527765461277,-2.40518001148392,-0.930419689758896,-2.17388106391883,-2.44333498406139,-2.44485024996483,-2.15396452644492,-2.39478864572392,-2.1622720671876,-2.37176574271087,-2.43852424826854,-2.42848929476839,-2.19310509083445,-2.40512895866212,-2.39051185374063,-2.39677253326316,-2.40129414946222,-2.43380618542814,-2.37567904699043,-2.17075726939949,-2.36518443180629,-2.36407123387706,-2.39163630486381,-2.39659848979589,-1.84462522127302,-2.17611784737675,-2.39971300450069,-1.93137237165398,-2.38734701730226,-2.41666909980503,-2.43623539682393,-2.16956261017478,-2.13445831589053,-2.40312692197421,-2.3886789924736,-2.41296475772195,-1.21181087394922,-2.37439069187184,-2.42470155212012,-2.43535975928334,-2.44106556391724,-2.35931450685401,-2.41621919107073,-2.35176169868008,-2.37859594636046,null,-2.36060025685382,-2.37875247642313,-2.39017119203144,-2.35988273623315,-2.38406388969541,-0.773724375293724,-2.36789318076716,-1.57160750179405,-1.87243682659333,-2.17529504567503,-2.39854324285787,-2.42534046394666,-2.39938187355091,-2.36166760505092,-1.94196677509504,-2.11141928132276,-2.38005293156401,-1.08596807560072,-2.36726885908664,-2.35160857723475,-2.12229217982969,-2.36416286435359,-2.44725038032428,-2.37245558738158,-2.38455369070088,-2.38859355668918,-1.68466001906557,-2.35299854667772,-2.39719659693454,-1.94525602910477,-2.40233301830393,-2.3734439698086,-2.41134999953591,-2.44686970506244,-2.35956928148315,-2.39524478745096,-2.42989307965998,-2.11028985481827,-2.39020437400703,-2.44597769289193,-2.42478375484151,-2.39486079358411,-2.40474122101822,-2.40357397706629,-2.39437840297268,-1.70603733797683,-2.37795494867404,-2.44158837020312,-2.41275151910176,-2.44283225348134,-2.42225747632709,-1.82784151172833,-1.85242803870802,null,-2.20429726089935,-2.4382380323057,-0.0689153121279597,-1.55563600122755,-0.768817814599316,-2.43697203726423,null,-1.99742737963139,-1.39324142236762,-2.00823784539285,-1.09878700883971,-2.41875415817284,-2.35414986980994,-2.44280871432655,-0.938570670211038,-1.85607741301516,-2.15149871749456,-2.38316735967909,-2.44057513624811,-2.40629656361797,-2.40525447430936,-2.40699867930178,-2.36295664087319,-1.98014914413257,-2.41184442819011,-2.43707699869209,-2.20237843457727,-2.44402940739327,null,-2.3551709430796,-2.3934204527215,-1.44616500788602,-2.10911188452157,-2.44335836877962,-2.36584447785661,-2.37697014348031,-2.37877192788673,-1.96621036935872,-2.38506456213453,-2.35723958455873,-2.39889917245981,-2.13668818576845,-2.39137723818802,-2.39132038373881,-2.37487444712696,-2.37283498871055,-1.7052393082669,-2.40216028525409,-2.36155236354479,-2.34818867897064,-2.37735096379564,-2.44558663317383,-2.43640651510012,-2.42875074299899,-2.37463846658737,-2.12492671597867,-2.4108389440962,-2.13214989191039,-1.31973150870692,-2.11256825137536,-2.36054153023669,-2.357623724131,-2.3529271820831,-2.43404312180736,-2.35805704270405,-2.12363966282575,-2.43603705347431,-2.41124567841561,-2.3893640556178,-2.37770150627856,-2.11801933909884,-2.41524690546011,-2.1574570911762,-1.91497369521572,-2.39313513487958,-1.92269093244146,-2.35257049845912,-1.64279064453672,-1.95609382247827,-2.41865497180642,-2.44423878956967,-2.11737481044258,-1.95411231479417,-2.12067945851597,-1.15901470534743,-2.39768411966887,-2.18978191107164,-1.71316508575468,-2.42634869590783,-2.41766637477939,-2.36994141943668,null,null,-2.37968065541153,-2.43265637483363,-2.40512931624343,-1.97076112815454,-1.93578447370133,-2.3637616271424,-2.40371637668376,-2.38208140794103,-2.42736916705416,-2.14026807924385,-2.34980598830533,-2.349315124265,-2.43052137131252,-2.14657209532453,null,-2.38396128719122,-2.39805919034787,-2.44094816551459,-2.1235581264191,null,-2.40794305862308,-2.4390513611297,-2.39796783964065,-2.34979686609385,-2.41124222348825,-2.38277701845308,-2.41766702065159,-2.12196934468231,null,-2.42624598902565,null,-2.19342165033451,-1.98259147304385,-1.49020237204446,null,-2.4390900285487,-2.41075201798611,-2.40424510923129,-2.431902300221,-2.38325523976617,-2.35755527208474,-0.401879445888855,-1.12572053403971,-1.78751649669008,-2.35977123289284,-2.43103003886444,-1.96362907198532,-2.18912174283179,-2.44211051967376,-0.136467040036188,-2.18259624536733,-2.35659305113749,-1.95695807394846,-2.10917325360353,-1.91993052099175,-2.20148227250321,-2.3765458950811,-2.37490764416669,-2.39654060809788,-2.13430801036886,-2.43937541151334,-2.19714858779336,-2.35249576582954,-2.40500095349652,-2.37778783410703,-2.39158564384942,-2.40685894359284,-2.34913268927866,-2.43103568735913,-2.37473486173347,-2.43090287447357,-2.15548805681175,-2.38754465387118,null,null,-2.38324435940393,-2.37433921571736,-2.19438102723217,-1.22852838424983,-2.11234316956835,-2.39365597332743,-1.98193693024847,-2.16077205148178,-0.362705348972423,-2.36220405563162,-2.38804394342386,-1.32663085858732,-2.40337714353026,-2.39922761754514,-2.42577804681295,-1.95864816420986,-1.95579104576069,-2.40370333141562,-2.2025688729773,-1.91096076654545,-2.4050574631934,-2.14143803346089,-2.19068073966193,-2.12492357874873,-2.37342546855015,-2.37259010004726,-2.41208498883442,-1.91574439719594,-2.42430042492141,-2.1207528678293,-2.15511439140538,-2.37896762714171,-2.39926953649064,-1.81256028195945,-1.6037778625157,-2.36077765766912,-2.382573875486,-2.35227024325085,-2.38807044367221,-2.42951321306554,-2.38035720084243,-2.38959657936738,-2.12726745582769,-2.41141314255455,-2.36674327985764,-2.19366343336298,-2.43205684831281,-2.42784733609677,-2.4368125158633,-2.36977689862632,-2.44284124728673,-2.37895792986181,-2.3788265510966,-2.43272101171696,-2.41674769337204,-2.4467899787045,-2.44005751116582,-2.38007344832093,-2.39941648240582,-2.38409409670074,-2.38784442474515,null,-1.91665843582056,-2.4448689902705,-2.35158283829604,null,-2.44255977741358,-2.42975499392868,-2.4170443278077,-2.36961198731241,-2.38299990587783,-2.4159457234471,-2.43341209770095,-2.42142729482529,-2.36520168830556,-2.36316915965089,-2.44406410013706,-2.36575123591264,-2.40942055327267,-2.36591023153541,-1.95262602478075,-2.35226949018341,-2.39536733664953,-2.42991710917939,null,null,-2.35174784697973,-1.97829995082035,-2.44444068477404,-2.35704849625853,-1.76762853957838,-2.44293100177691,-2.16438555475908,-2.11121120313766,-2.42429981690747,-2.36941138110504,-2.3539005102395,-2.4046070391816,-2.39076077407607,-2.38337506729093,-1.70087353011252,-2.20052703444964,-2.00034894835013,-2.39322967180432,-2.42524750321553,-2.41087635234379,-2.4432064802424,-2.40373975001656,-2.39619550523729,-2.19776372870548,-1.76888664838983,-2.42062047769231,-1.61929254959191,-2.35268287293528,-2.37806095239156,-2.38617814213884,null,-1.19988440269622,-2.37455644950186,-2.40793423306112,-2.41902776162987,-2.39404370175124,-2.35467328934845,-2.15768387262362,-1.92610952259316,-2.35940576619849,-2.38729030900084,-2.41047074776524,-2.42831514310473,-2.38240360336293,-2.41622808864721,-2.42284336435058,-2.44574542809659,-2.40121354593811,-2.44617065585134,-2.17383148884984,-1.86738048953572,-1.97835789049319,-2.37967544307847,-2.38732454753862,-1.81068915187949,-2.15530603293447,-2.38556797926229,null,null,null,-2.38685814810832,-1.99305559826358,-2.43913936466359,-1.92887334103039,-2.36621888722674,-2.37616419815839,-2.15680099686726,-1.84197963793864,-2.13753772085185,-2.43175575309814,-2.44754723345776,-2.40061602751174,-2.41933319045146,null,null,-2.42525161765897,-2.40692933476911,-2.1483267714095,null,null,-2.43623847512443,-2.42240063008302,-1.63563890682067,-2.42412016916004,-2.36436059486967,-2.41872320874044,-2.11092637924595,-2.41259163634767,-2.39724600080316,-2.35137144902636,-1.94801864373772,-2.36737159591083,-2.18322756980198,-1.77512296637157,-2.3556913470189,-2.44760065505196,null,-2.4277641939503,-2.39280344744128,-2.44715620763981,-2.35196413256143,-2.4263550187732,null,-2.39264992098705,-1.56935726277384,-2.40883573671252,-1.13893279391448,null,-2.14108279995273,-2.36631032375534,-2.37065664478959,-2.44707372479987,-1.96033099262319,-2.14316134844559,-2.37459643343844,-2.12952499879176,-2.4039456992681,null,-2.15100224475029,-2.39387761284978,-2.15164474147024,-2.39923495836937,-2.35606249296487,-2.3975865421622,-2.37305685296104,-2.18404300870026,-2.39241045290054,-2.3518721794282,-2.3554635007671,-1.60831650806444,-2.43452002103821,-2.37125890027181,-2.15658472353245,-2.3506336943983,-1.98838929588716,-2.2022838706209,-2.44078631201209,null,-1.64244629155378,-1.93555372190936,-2.42342710975007,-2.40129376252097,-2.36590103330472,-1.84211566474856,-2.40301913414531,-2.38641105792866,-2.35428475882308,-2.40015044080009,-2.38610651265279,-2.40204570063305,-2.4036299875277,-1.98207628183342,-2.14654230857852,-2.41084286235654,-2.3684848859013,-2.35862031646919,-2.42397189443381,-2.10751174360632,-1.96491566962837,-1.40726234390679,-2.38101170943499,-1.68083322454709,-1.99101077829297,-1.56447494294348,-2.4049214928449,-2.40306700512682,null,-2.43919221335472,-2.40713313920804,-2.36743117263921,null,-2.19725802541363,-1.91414478839184,-2.36430753935509,-2.19792878551527,-2.39088432235093,-1.37884404211584,-2.38150607783263,-2.38169799533588,-2.38055520090655,-2.41911227840845,-2.41153300167442,-2.38263930695047,-2.37633794565135,-2.42124110295352,-1.82841165858423,-2.38867122717583,-2.3528915881683,-2.14661237614347,-2.1747468077636,-2.42963288016998,-2.38360759737351,-2.15654937036453,-2.36501662615136,-1.50659037070641,null,-2.3830340997354,-2.38995442716786,-2.17580426806236,-2.16965911584559,-2.44390510439788,null,-2.39317702274219,-2.34875465724998,-1.65043849690351,-2.16626125712066,-2.37879731944212,-2.40136475607668,-1.79039706961354,-2.43789264138703,-2.38102659583473,-2.35357929068067,-2.41348806436089,-2.43904770338357,-2.4130940567973,-2.40356404350986,-2.37626591851384,-2.41346783715215,null,-2.35348732213405,-2.43205199728635,-2.35874827490569,-2.37355826748916,-2.41476893166927,-2.4450064719227,-1.81697535841828,-2.43783939034128,-2.36545529084195,null,null,-2.36569810242307,-2.44086864481595,null,-2.44219821934038,-2.42284456091382,-2.35662876610094,null,-2.38202042126012,-2.39769171756142,-2.4070347217881,-2.35660094230351,-2.41718029922721,-2.39302575395358,-2.3882884023982,-1.93761815301883,-2.35648387207565,-0.868401124165775,-2.35563711503436,-2.41703832631846,-2.35474371496038,-2.42566421598111,-2.38596161650875,-2.42934901332864,-2.39502367941112,-2.41417237001386,-2.36836138072339,-1.92137320454973,null,-2.39063610367392,-2.38074859928255,-2.42477084459186,-2.43858724246182,-2.4181921490739,-2.37838492300742,-2.40845299677839,-2.41571287750647,-2.15540430476541,-2.37534908478063,null,null,null,-2.40748458258257,-2.35400813063987,-1.4774649472212,-1.32866650801081,-2.40261088097849,-2.37910419090566,-2.37866116606546,null,-2.16491070304184,-2.38618880219673,-1.66425986794289,-0.713599723678463,-2.39256932340012,null,-2.38565354710826,-2.38774104526182,-2.35283553833523,-2.38220180068546,-2.13407611367876,-2.19286490965753,-2.4088247565125,-2.36834070592103,-2.15238049526817,null,-2.37730695072477,-2.3490858023093,-2.42871835516215,-2.38702925313645,-2.42756668308665,-2.43305188543544,-2.44546065726698,-2.41499569476963,-2.1382412389212,-2.37136385841676,-2.20386016015547,-2.37831227858932,-2.39282769821965,-2.35833201592703,-2.1277567783429,-2.42294729613149,-2.35258643289943,null,null,-2.35254800471392,-2.43538351781116,-2.43007655765776,-1.42063727646634,-2.4213301723861,-2.18629311870902,-2.43970779388127,-2.36765477646057,-2.36669319034935,-2.38234162903772,-2.39117818235335,-2.43659154542637,-2.43277086615943,-2.4078831087836,-2.39371110313022,-1.47291113491869,-1.53087252002502,-2.43540316871748,-2.15013385878994,-2.40599791473159,-1.97239403176396,-2.12064977458796,-2.388972799996,-2.42876505963885,-2.13502569615959,null,-2.35477179238022,null,null,-2.14721190108104,-2.3552780782842,-2.3781158611915,-2.19925981220442,-2.1422038866731,-2.38062394413081,-2.39271991909862,-2.40966718495273,-2.36228975642388,-2.41526197712318,-2.3921459614803,-2.36188204016452,-2.4381236042167,-2.39122963799545,-2.39365136733194,-2.43834576488388,-2.37003327720248,-2.4241971977326,-1.70415884795598,-2.40770652110604,-1.984347798082,-2.39386686487504,-2.3828985888549,-2.35646309652747,-2.39781369390347,-2.38327474536289,-2.3701450501141,-2.38202243349896,-1.64816818026825,-2.18780626247755,-2.37639056975403,-2.39483037146655,-2.44546919865915,-2.44700501824644,-2.39272701989775,-1.76727413986003,-2.38097942059775,-2.4003417905691,-2.36572951681827,-2.40719091308218,-2.43934021594849,-2.35636317339175,null,-2.35551610333295,-2.44676735834599,-2.43887865986125,-2.10872220828316,-2.00080579085085,-1.99971846361919,-2.12628114536057,-2.36238087076143,null,-2.37684342246884,-2.44293479125862,-1.50604309680699,-1.84490287251407,-2.13343204663253,-2.37439302709334,-2.43940975231034,-2.39854572085113,null,-2.13534582700233,-2.36056459942282,-2.44372415994675,-2.35559572102883,null,-2.15092555855519,-2.43649868225128,-2.35357989909043,-1.84562487389063,-2.41831509189595,-2.36909158996222,null,-2.4271555923662,-2.4383173369375,-2.42140535500747,-2.14836566004089,-2.38161780974277,-2.40698260728185,-2.44532438342375,-2.14866551096203,-2.37721376757053,-2.42893038823185,-2.15702140750947,-2.18260913424312,-2.36554985086349,-2.44736461455849,-2.37823692472101,-2.41274137678937,-2.14341083916953,null,-2.35373706140471,null,-2.39199577113994,-2.34853059607939,-2.10783421905364,-2.43786970817399,-2.43782393092205,-2.19464536761517,-2.39153068373361,-2.40618783687622,-2.167134891107,-2.42825563419991,-2.43451311157932,-2.42617121984166,null,-2.14404610703352,-2.4383926013057,-2.34901567888827,-2.43957892889538,-2.38893961476077,-2.10847304913333,-2.36626740368464,-2.39407363268928,-2.44528167171319,null,null,-2.36558034834833,-1.43703287469037,-1.65944868083111,-2.38241284129125,-2.43677265479145,-2.44407189269931,-2.38750072478232,-2.44316734350667,null,-2.4380146395686,-2.41195608915676,-2.42617564353076,null,-2.37407250174236,-2.40327655263626,-2.3842880708902,-2.18146168154782,-2.40824420256921,-2.43012004916471,-1.53760479321996,-2.40612705792681,-2.11027463540087,-2.11149654051407,-2.39044313128487,-2.41589971729935,-2.38822430207309,-2.42398532920432,-2.12873404117419,-2.43861565983769,-2.17095643492068,-2.3848008128154,-2.36725026953881,null,-2.42874643155754,-2.15767503493118,-2.12625216849266,-2.18819847617416,-2.36516097630755,-2.44641242202287,-2.34954802089286,-2.11196661387223,-2.43548897204836,-2.41448487462857,-2.40530960301784,-2.44490385787011,-2.42122618039519,null,-2.37419190868185,-2.4272883909094,-2.41746109065515,-2.42787581009705,-2.42205433210273,-2.40766368699641,-2.37384258099281,null,-2.36091995246624,-2.39901955542535,-1.17108099384611,-2.36427663113371,-2.39169066038829,-1.54927095434482,-1.84522371134823,-2.42288321569011,-2.42975040779364,-2.39924400563267,-2.40618659614501,-2.36158473065969,-1.99871447560186,-2.37415913588283,-2.41380217192144,-1.77683562832987,-2.43425671274689,-2.20160860724865,-1.26417605910674,-1.30527725146388,-2.34985429128081,-2.44685944347092,-2.3779251579699,-2.11265527357465,-2.17354401772028,-2.34842819013082,-2.42972886967854,-1.96704045449193,-2.42872950220861,-2.44040681408168,-2.40772320896302,-2.35758622233205,-2.11177996597836,-2.40234579858751,-2.13694593767288,-1.94975310546621,-2.44401027646762,-2.40599460297179,-2.13870641673567,-2.41362774190874,-2.40693045801398,-2.43520381015641,-2.18604962025421,-2.35391078566116,-1.40224905316751,-1.39760587059461,-1.17805507433346,-2.40851104376287,null,-2.43486419611526,-2.40480560660743,-2.16186126946124,-2.3801646804708,-2.39265634308484,-2.43910576610741,-2.37460850447331,-2.10985173597584,-0.825456077532155,-2.39512690327045,-2.40390771096953,-2.40185015194965,-2.37482311828443,-2.43446221601133,-2.35944426339501,-2.42386605574665,-2.36188359400638,-2.39366747255352,-2.00385902534514,-2.44465915793934,-2.36082456258972,-2.41399705098391,-2.39268037572418,-2.37923220380125,-2.43103375262962,-2.36364174487916,-2.13927646005178,-2.4014560003803,-1.97745776547859,-2.43633530249734,-2.41322582862721,-1.3380212674085,-2.42127573346169,-1.95988568553044,null,-1.99526231462411,-2.37664651137692,-2.40806879199025,null,-1.49245723951376,-2.37242170507469,-2.41036905278833,-2.41523525170431,-2.42794284983919,-2.17798483800727,-2.3515246793433,-2.4230897252486,null,-1.38709829603251,-0.726578118543522,-2.4389733879865,-1.07094523004401,-2.40587750967042,-2.17297129408109,-2.43175019096033,-2.14462921191591,-1.70550344785675,-2.10958381831298,-2.41457748073534,-2.37988782548726,-2.41654293109112,-2.41485671713398,-2.35602636277125,-2.16683093635949,null,null,-2.14874239389903,-2.16428987016554,-1.54072782552162,-1.96325272192522,-2.37901699890298,-2.35420848857992,-1.62828309067059,-2.43166870698714,-2.39203531053298,-2.43628364059058,-2.40549700067294,-2.37159291704101,-2.37109479089648,-2.41486927802124,-1.95382043252508,-2.20163313684172,-2.40459666122132,-2.42687451481269,-2.43910313856703,-2.1853768111265,-2.00134760433617,-2.44787581460931,-2.13890291920981,-2.4322696604804,-2.36599897321229,-2.40008579655931,null,-2.3837848428442,-2.36586762187452,-2.15993133362501,-2.36750626254742,-2.18144016697189,-2.3728173094374,-2.34813635291447,-2.1106844709546,-2.38413856905335,-2.3967835902973,-2.43825278350067,-2.44040123722897,-2.15700994048386,-2.13640209937586,-2.42243108477359,-2.34909630797719,-2.43314145340499,-2.39158128661019,-2.42942335559436,-2.39962477868338,-1.60818114659283,-2.40609511607707,-2.36226783970286,-2.38573728220452,-2.42762637497307,-2.42830870319539,-2.40232954440088,-2.12877761319398,-2.38886261964598,-2.43798640785188,-2.42879962094975,-2.38862464512148,-2.4034973017211,-2.40553189795845,-2.35741336872251,-2.42894721890715,-2.11708286444775,-2.35838172531602,-1.97211294394261,-2.38658732478888,-2.40554836546452,-2.39274573305537,-2.44444235009522,-1.85843208181774,-2.36230970940468,-2.42612307237376,-2.42853127175857,-2.38003932980736,-2.41112311794803,-1.09107239327979,-2.43858717123893,-1.7105397467188,-2.36679233746053,-2.13254817380222,-2.44187154097398,-2.40749662819234,-2.37484060144433,-2.44262762095694,-2.17466576037115,-2.15454526837568,-2.43820955753707,-2.42374566486486,-2.18622071636493,-2.13839693293559,-2.37807376908985,-1.76394731597168,-2.11663140189486,-1.41018461457022,-2.43064660968174,-2.12118830942201,-2.38672188436993,-2.40669270337474,-2.44054001460698,-2.17794072402759,-2.41005930588962,-2.15153783355492,-2.40591464245607,-2.40563147382726,-2.38786887189289,-2.16346137910734,-2.41873222927478,null,-2.35702094470997,-2.36267723432192,-2.41951613163491,-2.36181415729606,-2.39549834586595,-2.40605656365314,-2.44337510676504,-2.19031911469648,-1.98878304612632,-1.93155033959816,-1.54884842600527,-1.6735246128322,-1.80575810567112,-1.69746902479449,null,-2.38950454085515,-2.41802204826762,-2.42785666620274,-2.42503355692987,-2.40567608029031,-1.60186029517877,-2.38856904199728,-2.13804334581948,-2.3695401020619,-2.44081424104896,-2.3926971977149,-2.38572036798959,-2.36014278282327,-1.94080264939736,-2.36169523795051,null,-2.42453601351389,-2.36099329507359,null,-2.37193066627168,-2.41900811416944,-2.38219083804087,-2.35404646287587,-2.44328737813429,-2.3592329944522,-2.42891071993307,-2.39144718128172,-2.39021470016249,-2.43480272994292,null,-2.44481038581961,-2.40710549222221,-2.39592649811328,-2.44742157564414,-2.40844624000982,null,-2.39471218204507,-2.43674953033131,-2.37709744382099,-2.4348380459045,-2.1878862541693,-1.44368471079274,-2.11459629745806,-2.43667814338495,-1.57746995739958,-2.17879757781631,-2.39821704451841,-2.37575117829637,-2.15685137257098,-2.40603560984981,-2.43164780578026,null,-2.1890707227227,-2.42835792913482,-2.1443114230002,-2.4003675288326,-2.41908021304206,-2.39103613114365,-1.7360044521199,-1.47906415960519,-1.58776843120663,-1.98515555424425,-1.10918471778369,-2.20164196505795,-2.14445564840573,-2.15782833120904,-2.37535302129489,-2.13471468946083,-2.43174133962401,-2.35042050525461,-2.40459410937418,null,-2.42551317850974,-2.43233889660713,-2.36555334173989,-2.3877042510352,-2.40455873757982,-2.42180422554173,-2.34873966949907,-2.38072502133817,-2.35702792057217,-2.38955524380234,-2.42188436785162,-2.37586287523535,-2.3504135844103,-2.43948903514919,-2.40153127040629,-1.68270833153743,null,-2.38151745617305,-2.40433907164642,-2.1938199556491,-2.0065473818108,-2.42247692936015,-1.87094742370406,-2.38172235116743,-2.41938996937041,-2.4452449526443,-2.18532963484178,-2.16292737899932,-2.38905730538917,-1.68468936135992,-2.3608168949644,-2.44660231715915,-2.34953036313326,-2.36429155737077,-2.40389002494429,-2.4212419213998,-2.41108980654188,-1.95910828191596,-2.44109590510287,null,-2.38106010142163,-2.41424995854412,-1.43712774282089,-2.43496072893412,-2.437548403469,-0.193958918242249,-2.44381069916216,-2.35175206433431,-2.43322761706457,-2.39538632349958,-2.4251705098004,-2.3770395875248,-2.40116524845743,-2.37691945454952,-2.41414967254185,-2.40363003025212,-1.12717115679667,-2.39486702003431,-2.35787183789452,-2.3711526863315,-2.41464716887948,-2.41777654965789,-2.37551932183941,-2.40253355377735,null,-2.17504244207523,-2.38187654959419,-2.41399549125143,-2.400042255506,-2.42129817691547,-2.37064495999747,-1.91413100742544,-2.13390139467667,-2.3860466299151,-2.37677951755849,-2.41434663285416,-1.33236946758392,-2.36889390428931,-2.16699078931126,-2.40579245610078,-2.3872178180647,-2.34882476544389,-1.6713195204176,-1.93231706799748,-2.37913752701514,-2.35916655010459,-2.35102532026739,-2.37881966205695,-0.977791951993589,-1.92221026088851,null,-2.42409792675552,-2.3658265903863,-2.14303577722168,-2.37186513520227,-2.38972999311899,-2.39122990642589,-2.43182387778451,-2.38526884422553,-2.43534147444096,-2.37874292596624,-2.40198822745969,-2.4457972557322,-2.41530815221284,-2.36063628958453,-2.17730278005476,-2.40969281872184,-1.92316431944533,-2.12490963389284,-1.78513475347878,-1.47080874401178,-2.39773793644653,-2.43890362256587,-2.37498356565283,-2.40760346798141,-2.42363268498246,-2.43924373404969,-2.41575651725669,-2.39209666098849,null,-2.11667300835835,-2.38144220207267,-2.37098477883645,-2.44029668236998,-2.12734194514925,-2.36768667614468,-2.39455850297035,-2.4063404690978,-2.12257048918984,-2.40053539951713,-2.36638563452669,-2.36620131577746,-2.40988877532453,-1.50318568930729,-2.37767229692926,-2.44366877763913,-2.43817700350572,-2.37841127631635,-2.13120889352287,-2.42595038493545,-2.41330998894208,-0.9833302120347,-2.14587001093316,-2.38774332062256,-2.40723466687285,-2.36175327834588,-2.41926076622129,-2.40978696765759,-2.36510881160301,-2.14271690797179,null,-2.40764694086191,-2.43097214235839,-2.43196111272936,-2.39845727230803,null,-1.93603525696131,-0.843264155535211,-2.40192104299002,-2.44611259303437,-1.48089410492718,-2.10708668251174,-2.41603926241313,-2.36382995571856,-2.16938012568864,-1.62062209038063,-2.37085444964849,-2.41694559366823,null,-2.44102685887282,null,null,null,-2.36641057709147,-2.43456711955749,-2.38295797028494,-2.3870971079721,-2.35228466408242,-2.37642943305344,-2.43034850068306,-1.16771200567861,-2.35665844492995,-2.3890276383181,-2.43384006051728,-2.3789686897342,-2.38830756366168,-2.1477486776712,-2.39495398810049,-2.20091285352247,-2.40276252015409,-2.38644711294127,-2.44112880621486,-2.36362590792504,-2.35224318216004,-2.43389352958565,-2.34955481330779,-2.37439632164695,-2.41889399940063,-2.39693486816631,-2.42027645905417,-1.86997946753269,-2.10749837684563,-2.40477836458393,-2.43323946106628,-2.3590126877223,-2.38756567340259,-2.39256343287797,-2.40595905501776,-1.95283033001511,-2.44544535661497,-2.3985326766773,-2.39211273237978,-2.35884792928499,-2.41965924827763,-2.35114681115755,-2.37049579769754,-2.18920040648083,-2.40793216491965,-2.41701989521188,-2.39994206887585,-1.39104900224413,-2.18536200672972,-2.15288841361764,-2.43122437880755,null,-2.39115945545872,-2.43515337431253,-2.20024156591973,-2.38776450206441,-2.37579348571979,-1.86240362591212,-2.37007803323594,-2.40699962526941,-2.10504993836513,-2.35932211999697,-2.41689718207725,-1.19296977688798,-1.94643365686103,-2.20097215623247,-2.18126776035062,-2.12059418447899,-2.4149569134266,-2.44307728913062,-2.44632176661779,-2.17986406959246,-2.36425217448649,-2.35334027982698,-2.35954437617128,-1.96761103923242,-2.37574270775457,-2.39969429525462,-2.42367864717179,-2.19137827240049,-2.17704255225255,-2.38793628672519,-2.39827197939538,-2.40517458173362,-1.49908641411225,-2.42358244224832,-1.33945277465902,-2.40918354059485,-2.40138883456037,-2.41074761362196,-2.41829657338475,null,-2.12675116800821,-2.41688291053669,-2.37154312108559,-2.39932403241199,-2.39960524861616,-2.43883959006433,-1.75778897853227,-2.34899556620893,-2.43320362328467,-2.40106964342211,-1.91966079055606,-2.37182391146579,-2.15038995796933,null,-2.19056463000017,-2.16040423582892,-2.4043347849083,-2.44096504953139,-2.42846188194034,-2.44750734770586,-2.40607301029758,-2.44230892085851,-2.35135655927387,-2.37116184453861,-2.43160578241021,-2.39003043761858,-2.44696334693962,-2.41740244885822,-2.41138695199423,-2.36087845173114,-2.38781563872852,-2.37021015601148,-2.4008193046824,-2.38733527842359,-2.40055498704695,-2.37785269564864,-2.42492934314449,-2.43274380793245,-2.41158129466147,-1.95029741857476,-2.39067949981344,-2.37612075920132,-2.40817835722499,-2.3980294347546,-2.41271107015581,-2.39256635392466,-1.91913730652346,-2.38742418780838,-2.40693336727944,null,-2.37287575856698,-1.92594058626606,-2.44399459166833,-2.16216081143941,-2.37676478552547,-2.38795703279701,-1.80325173329198,-2.38270826253955,-2.39554634008472,-2.44473080293191,-2.36781638698102,-1.93720302433982,-0.806941662320026,-2.14315332803528,-2.38782272180923,-2.36013756662522,-2.38934288169638,-2.16977312127712,-2.44393260386222,-2.38275677268774,-2.42263712501534,-2.43722644814205,-2.38582844438878,-2.40338055529081,-2.12520708407393,-2.44719900389117,-2.40157822126438,-2.43772111558736,-2.41356974884783,null,-2.37654529980299,null,-2.41281700947237,-2.145557526598,-2.3977392198556,-2.40836523266424,-2.15358190639528,-1.92855782277226,-2.34903225214601,-2.38496592864775,-2.42390863290437,-2.37865369709124,-2.36607865002387,-2.37922693940726,-2.42326783505649,-2.43725136388474,-2.39711086636324,-2.3577354493589,-2.43485561872748,-2.4394620605084,-2.44152073285458,-2.39483293991451,-2.37866405519106,-2.44688303925012,-2.36687216107183,null,-1.99518054433907,-2.40824054533531,-2.44253235829909,null,-2.41753994787958,-2.40403779265852,-2.3583784751865,-2.3964451949383,-2.37588740478185,-2.38661085318689,-2.40660764945586,-2.39958294919141,-2.38126124686477,-2.4174252784849,-1.9607638986669,-1.63113428828306,-2.42628049732101,-2.34889745463532,-2.38845427493014,-2.36884559144181,-2.35595466313333,-2.00118160363096,-2.00615754463985,-2.15706733193367,-2.35254499973179,-2.18359446717078,-2.12648065549439,-2.38228751721707,-2.35365818594963,-2.39948847294235,-2.3999584673942,-2.36172358442944,-2.42283648685933,-2.41942909441804,-2.44470953720537,-2.35547208537263,-1.78782181461444,-1.36187518825683,-2.41003273365003,null,-2.38443868568082,-2.13802699479858,-1.55577678500063,-2.34994960716375,-2.41151660737031,null,-1.68786421759645,-2.36084685204931,-1.36839415943179,-2.39098295355597,-2.40160090674561,null,-2.18693003755181,-2.43011920303487,-2.37588520883963,-2.42579476917543,-2.41364074899265,-2.38711971077518,-2.12555493240363,-2.37435961477839,-2.00625892289723,-2.35984640107871,-2.39434371248734,null,-2.37648120329631,-2.3764325917038,-2.43107311705043,-2.41693810376254,-1.30840707876763,-1.44936740768375,-1.8545301591848,-2.40280642568049,-0.66360563184788,-2.10809837841081,-2.106963312655,-1.2134976362691,-1.98832781504649,-2.40253745750905,-2.36944002418694,-2.44388281361115,-2.42008745219008,-1.83515856031896,-2.42317401126874,-2.12269374027966,-2.15374574587333,null,null,-2.43074078723238,-2.39662303261374,-2.40586411951025,-2.39872459120032,null,-2.44317619821903,-2.4317264677762,-2.36891675684982,-2.38408516194121,-2.38993383336727,-2.39893699386538,-2.35011617413637,-2.20318280219218,-1.97634523926578,-2.44449813778427,-1.93795014856192,-2.44455214911521,-2.19298037812697,-2.40332343355202,-2.17617717479008,-1.42780836133786,-2.35465218385597,-2.41135084412907,-2.38436659856615,-2.42463516106744,-2.13788392868011,-2.4401984496313,null,-1.59003757658734,-2.2039152512618,-2.37623002604031,-2.38085397116755,-2.44302837134254,-1.98743745669532,-2.43893659164113,-2.44106129156806,-2.40350530415689,-2.35555952187077,-2.4393753633174,-2.4452212459213,-2.41339101512366,-2.40064798209731,-2.20061575992151,null,-2.36522532031321,-2.35399817617525,-2.40033013180745,-2.41216075651263,-2.38184197096069,-2.40947434227363,-2.00005711173913,-2.40602337145627,-1.34063483452703,-2.401116731627,null,null,-2.11302928252264,-2.39316966545683,-2.11797136883835,-2.15478604678648,-2.11189421706951,-2.36949165491768,-1.94892409030623,-2.44766937634357,-2.42246210207612,-1.38822581713833,-2.43398360545197,-2.44738256889776,-1.33444127845507,-2.17557875784698,-2.44115985145708,-1.98452027590289,-1.92849707197207,-1.92736171229086,null,-2.36911921564405,-2.18022530587042,-2.42934822207698,-2.417451988048,-2.37445524840531,null,-2.40542035140478,-2.35663715445352,-2.39622875033327,-2.35058815861502,-1.96582300515189,-2.41160344337494,-2.383096141046,-2.18163950073651,-2.4234776755833,-0.284041910625347,-2.14186446935056,-1.25227027136411,-1.75912825911431,-0.744711090539032,-0.963698494459799,-1.24834977797277,-2.36279461325993,-2.37793765659825,-2.4291721303054,-1.93733226876981,-2.44342578836166,-2.38600882185073,-2.42167305504353,-2.2004488019331,-1.94428191353842,-2.4141490162854,-1.68348538593048,-2.40013623109934,-2.40144439754457,-2.34863617483442,-2.38215722411786,-2.37217699517877,-2.13216806320126,-2.40419809739486,-2.40938095041,-2.38709248938047,-2.41040829073553,-2.4212047692892,-2.38331636886959,-2.12091785057566,-2.12262594763308,-2.35630322276064,-2.11375576831053,-2.37084176128645,-2.44173516126265,null,-2.11477452742985,-2.35507238381923,-2.44356569296596,-2.42985560698816,null,-2.3794133166355,-2.39240995273375,-2.44137462131129,-2.39454773939595,-2.38590814152648,-2.39969105399595,-2.4314339759332,-2.41261565501717,-1.99247314013815,-2.39853396441703,-2.36087543676057,-2.35216637498614,-2.43388972285119,-2.15089419950383,-1.76957440501204,-2.12500617902844,null,-2.43080398752735,-2.39023423844864,-2.1799795498897,-2.37102999608413,-2.43874055373398,-2.3996829962396,-2.35547363656023,-2.42839516941839,-2.41125866440506,-2.36355510126249,-2.15117789079087,-2.20194616756054,-2.3782591393954,-2.17112479245408,-2.20257762948169,-1.68401423324024,-1.92757147237468,-2.38912418433846,null,null,-2.18630929825014,-2.36484790658959,-2.43117492515975,-2.36787898152051,-2.36436961444941,-1.39093199809361,-2.19748974003016,-2.38613764976998,-1.5599371483439,-2.43725161431738,-2.35146552145396,-1.74182472272435,-2.36347188877788,-2.40482458982532,-2.36955125334588,-2.35456418656194,-2.42845014047724,null,null,null,null,null],"text":["Diabetes:  NA<br />CRC: 0.004<br />a1","Diabetes:  NA<br />CRC: 0.004<br />a13u","Diabetes: 0.1<br />CRC:    NA<br />a1c","Diabetes:  NA<br />CRC: 0.004<br />a498","Diabetes:  NA<br />CRC: 0.004<br />a549","Diabetes:  NA<br />CRC: 0.004<br />a723","Diabetes:  NA<br />CRC: 0.004<br />aa","Diabetes:  NA<br />CRC: 0.007<br />aacr","Diabetes:  NA<br />CRC: 0.004<br />aas","Diabetes:  NA<br />CRC: 0.004<br />abc","Diabetes:  NA<br />CRC: 0.004<br />abcc5","Diabetes:  NA<br />CRC: 0.004<br />abl","Diabetes:  NA<br />CRC: 0.004<br />ablated","Diabetes:  NA<br />CRC: 0.004<br />ablative","Diabetes:  NA<br />CRC: 0.004<br />abnormalities","Diabetes:  NA<br />CRC: 0.004<br />acap2","Diabetes: 0.1<br />CRC:    NA<br />acr","Diabetes:  NA<br />CRC: 0.004<br />actb","Diabetes:  NA<br />CRC: 0.004<br />actin","Diabetes: 0.1<br />CRC:    NA<br />acvr1c","Diabetes:  NA<br />CRC: 0.004<br />adamts1","Diabetes:  NA<br />CRC: 0.004<br />additive","Diabetes:  NA<br />CRC: 0.064<br />adenocarcinoma","Diabetes:  NA<br />CRC: 0.037<br />adenocarcinomas","Diabetes:  NA<br />CRC: 0.064<br />adenoma","Diabetes:  NA<br />CRC: 0.067<br />adenomas","Diabetes:  NA<br />CRC: 0.022<br />adenomatous","Diabetes:  NA<br />CRC: 0.004<br />adenosine","Diabetes:  NA<br />CRC: 0.004<br />adenovirus","Diabetes: 0.1<br />CRC:    NA<br />adiponectin","Diabetes: 0.1<br />CRC: 0.004<br />adipose","Diabetes:  NA<br />CRC: 0.034<br />adjuvant","Diabetes: 0.1<br />CRC:    NA<br />adrenal","Diabetes: 0.1<br />CRC:    NA<br />adrenals","Diabetes:  NA<br />CRC: 0.004<br />adrenergic","Diabetes:  NA<br />CRC: 0.007<br />african","Diabetes:  NA<br />CRC: 0.004<br />agar","Diabetes:  NA<br />CRC: 0.004<br />ages","Diabetes:  NA<br />CRC: 0.004<br />aggravation","Diabetes:  NA<br />CRC: 0.004<br />aggressively","Diabetes: 0.1<br />CRC: 0.004<br />aging","Diabetes:  NA<br />CRC: 0.007<br />ago","Diabetes:  NA<br />CRC: 0.007<br />ago2","Diabetes:  NA<br />CRC: 0.011<br />aid","Diabetes:  NA<br />CRC: 0.041<br />akt","Diabetes:  NA<br />CRC: 0.004<br />alanine","Diabetes: 0.1<br />CRC:    NA<br />albumin:creatinine","Diabetes: 0.1<br />CRC:    NA<br />albuminuria","Diabetes:  NA<br />CRC: 0.004<br />aldehyde","Diabetes:  NA<br />CRC: 0.004<br />aldh","Diabetes:  NA<br />CRC: 0.007<br />aldose","Diabetes:  NA<br />CRC: 0.004<br />alkaline","Diabetes:  NA<br />CRC: 0.007<br />allele","Diabetes: 0.1<br />CRC: 0.034<br />alpha","Diabetes:  NA<br />CRC: 0.004<br />alternations","Diabetes: 0.1<br />CRC:    NA<br />ameliorated","Diabetes:  NA<br />CRC: 0.011<br />american","Diabetes:  NA<br />CRC: 0.004<br />americans","Diabetes:  NA<br />CRC: 0.004<br />amines","Diabetes:  NA<br />CRC: 0.011<br />amino","Diabetes:  NA<br />CRC: 0.004<br />aminoacyl","Diabetes:  NA<br />CRC: 0.004<br />amo","Diabetes:  NA<br />CRC: 0.004<br />amp","Diabetes:  NA<br />CRC: 0.004<br />ampk","Diabetes:  NA<br />CRC: 0.004<br />amplicon","Diabetes:  NA<br />CRC: 0.004<br />amplicons","Diabetes:  NA<br />CRC: 0.004<br />amylose","Diabetes:  NA<br />CRC: 0.004<br />analog","Diabetes:  NA<br />CRC: 0.004<br />anatomical","Diabetes:  NA<br />CRC: 0.004<br />anchorage","Diabetes:  NA<br />CRC: 0.004<br />ancillary","Diabetes:  NA<br />CRC: 0.004<br />angiodynamics","Diabetes: 0.1<br />CRC: 0.007<br />angiogenesis","Diabetes:  NA<br />CRC: 0.007<br />animals","Diabetes:  NA<br />CRC: 0.004<br />annex","Diabetes:  NA<br />CRC: 0.007<br />annexin","Diabetes:  NA<br />CRC: 0.004<br />anoxia","Diabetes:  NA<br />CRC: 0.004<br />antago","Diabetes:  NA<br />CRC: 0.011<br />anticancer","Diabetes:  NA<br />CRC: 0.004<br />antidromic","Diabetes: 0.1<br />CRC: 0.037<br />antigen","Diabetes:  NA<br />CRC: 0.004<br />antimetabolite","Diabetes:  NA<br />CRC: 0.004<br />antineoplastic","Diabetes: 0.1<br />CRC:    NA<br />antioxidant","Diabetes: 0.1<br />CRC:    NA<br />antisenescence","Diabetes:  NA<br />CRC: 0.004<br />antitumor","Diabetes:  NA<br />CRC: 0.004<br />anxa1","Diabetes: 0.1<br />CRC:    NA<br />aortic","Diabetes:  NA<br />CRC: 0.015<br />ap","Diabetes:  NA<br />CRC: 0.004<br />ap1","Diabetes:  NA<br />CRC: 0.037<br />apc","Diabetes:  NA<br />CRC: 0.004<br />apical","Diabetes:  NA<br />CRC: 0.120<br />apoptosis","Diabetes:  NA<br />CRC: 0.007<br />apoptotic","Diabetes:  NA<br />CRC: 0.004<br />archival","Diabetes:  NA<br />CRC: 0.004<br />argonaute","Diabetes:  NA<br />CRC: 0.007<br />argonaute2","Diabetes:  NA<br />CRC: 0.004<br />arsenic","Diabetes: 0.1<br />CRC: 0.007<br />ascertain","Diabetes:  NA<br />CRC: 0.004<br />asian","Diabetes:  NA<br />CRC: 0.004<br />asians","Diabetes:  NA<br />CRC: 0.004<br />aspartate","Diabetes:  NA<br />CRC: 0.007<br />aspirin","Diabetes:  NA<br />CRC: 0.004<br />assertion","Diabetes:  NA<br />CRC: 0.004<br />assisted","Diabetes:  NA<br />CRC: 0.004<br />assists","Diabetes:  NA<br />CRC: 0.004<br />assumed","Diabetes:  NA<br />CRC: 0.004<br />attribute","Diabetes:  NA<br />CRC: 0.004<br />austria","Diabetes:  NA<br />CRC: 0.007<br />autocrine","Diabetes:  NA<br />CRC: 0.004<br />automatically","Diabetes:  NA<br />CRC: 0.004<br />autophagic","Diabetes:  NA<br />CRC: 0.004<br />autophagy","Diabetes:  NA<br />CRC: 0.004<br />auxiliary","Diabetes:  NA<br />CRC: 0.015<br />azoxymethane","Diabetes:  NA<br />CRC: 0.007<br />b7","Diabetes:  NA<br />CRC: 0.004<br />bacteria","Diabetes:  NA<br />CRC: 0.011<br />balb","Diabetes:  NA<br />CRC: 0.004<br />bar","Diabetes:  NA<br />CRC: 0.004<br />basement","Diabetes:  NA<br />CRC: 0.004<br />bases","Diabetes:  NA<br />CRC: 0.007<br />bax","Diabetes:  NA<br />CRC: 0.007<br />bcl","Diabetes:  NA<br />CRC: 0.004<br />bcl2","Diabetes:  NA<br />CRC: 0.004<br />bcl2l2","Diabetes:  NA<br />CRC: 0.004<br />bengal","Diabetes: 0.1<br />CRC: 0.064<br />beta","Diabetes:  NA<br />CRC: 0.004<br />beta1","Diabetes:  NA<br />CRC: 0.004<br />beta4","Diabetes:  NA<br />CRC: 0.004<br />betaar","Diabetes:  NA<br />CRC: 0.004<br />bevacizumab","Diabetes:  NA<br />CRC: 0.004<br />bifidobacterium","Diabetes:  NA<br />CRC: 0.004<br />binary","Diabetes:  NA<br />CRC: 0.004<br />bioactives","Diabetes:  NA<br />CRC: 0.004<br />bioassay","Diabetes: 0.1<br />CRC:    NA<br />biocompatibility","Diabetes:  NA<br />CRC: 0.004<br />biocompatibles","Diabetes:  NA<br />CRC: 0.004<br />biodistribution","Diabetes:  NA<br />CRC: 0.004<br />biogenesis","Diabetes:  NA<br />CRC: 0.004<br />bioimpedance","Diabetes:  NA<br />CRC: 0.004<br />biomark","Diabetes:  NA<br />CRC: 0.157<br />biomarker","Diabetes:  NA<br />CRC: 0.004<br />biomedical","Diabetes:  NA<br />CRC: 0.026<br />biopsies","Diabetes:  NA<br />CRC: 0.015<br />biopsy","Diabetes:  NA<br />CRC: 0.007<br />biosystems","Diabetes:  NA<br />CRC: 0.004<br />biotin","Diabetes:  NA<br />CRC: 0.004<br />bl","Diabetes:  NA<br />CRC: 0.004<br />black","Diabetes:  NA<br />CRC: 0.004<br />blacks","Diabetes:  NA<br />CRC: 0.011<br />bladder","Diabetes:  NA<br />CRC: 0.007<br />blinded","Diabetes:  NA<br />CRC: 0.004<br />blockage","Diabetes: 0.1<br />CRC: 0.090<br />blood","Diabetes:  NA<br />CRC: 0.004<br />bnip3","Diabetes:  NA<br />CRC: 0.004<br />bona","Diabetes:  NA<br />CRC: 0.007<br />bonding","Diabetes:  NA<br />CRC: 0.004<br />bonds","Diabetes:  NA<br />CRC: 0.004<br />bone","Diabetes:  NA<br />CRC: 0.004<br />bootstrap","Diabetes:  NA<br />CRC: 0.004<br />border","Diabetes:  NA<br />CRC: 0.004<br />bounds","Diabetes:  NA<br />CRC: 0.019<br />bowel","Diabetes:  NA<br />CRC: 0.004<br />box","Diabetes:  NA<br />CRC: 0.004<br />bp1","Diabetes:  NA<br />CRC: 0.011<br />braf","Diabetes:  NA<br />CRC: 0.004<br />branches","Diabetes:  NA<br />CRC: 0.004<br />branching","Diabetes:  NA<br />CRC: 0.004<br />brca","Diabetes:  NA<br />CRC: 0.004<br />brg1","Diabetes:  NA<br />CRC: 0.004<br />broccoli","Diabetes:  NA<br />CRC: 0.004<br />bromodoamin","Diabetes:  NA<br />CRC: 0.004<br />browsers","Diabetes:  NA<br />CRC: 0.007<br />budding","Diabetes:  NA<br />CRC: 0.004<br />building","Diabetes:  NA<br />CRC: 0.004<br />butyrate","Diabetes:  NA<br />CRC: 0.004<br />butyrylated","Diabetes:  NA<br />CRC: 0.004<br />c12g","Diabetes:  NA<br />CRC: 0.004<br />c2","Diabetes:  NA<br />CRC: 0.004<br />c20orf111","Diabetes:  NA<br />CRC: 0.004<br />c57bl","Diabetes:  NA<br />CRC: 0.019<br />ca19","Diabetes:  NA<br />CRC: 0.004<br />caa","Diabetes:  NA<br />CRC: 0.004<br />cab39","Diabetes:  NA<br />CRC: 0.004<br />cabbage","Diabetes:  NA<br />CRC: 0.004<br />cac","Diabetes:  NA<br />CRC: 0.004<br />cachexia","Diabetes:  NA<br />CRC: 0.015<br />caco","Diabetes:  NA<br />CRC: 0.015<br />cadherin","Diabetes: 0.1<br />CRC:    NA<br />caenorhabditis","Diabetes:  NA<br />CRC: 0.007<br />calcium","Diabetes:  NA<br />CRC: 0.004<br />cam","Diabetes: 0.1<br />CRC: 0.929<br />cancer","Diabetes:  NA<br />CRC: 0.030<br />cancerous","Diabetes:  NA<br />CRC: 0.191<br />cancers","Diabetes:  NA<br />CRC: 0.004<br />capecitabine","Diabetes: 0.1<br />CRC:    NA<br />capsule","Diabetes:  NA<br />CRC: 0.011<br />carbohydrate","Diabetes:  NA<br />CRC: 0.037<br />carcinoembryonic","Diabetes:  NA<br />CRC: 0.011<br />carcinogen","Diabetes:  NA<br />CRC: 0.075<br />carcinogenesis","Diabetes:  NA<br />CRC: 0.004<br />carcinogenesis.the","Diabetes:  NA<br />CRC: 0.004<br />carcinogenetic","Diabetes:  NA<br />CRC: 0.004<br />carcinogenicity","Diabetes:  NA<br />CRC: 0.109<br />carcinoma","Diabetes:  NA<br />CRC: 0.015<br />carcinomas","Diabetes:  NA<br />CRC: 0.007<br />cardiac","Diabetes:  NA<br />CRC: 0.004<br />cardiocytes","Diabetes:  NA<br />CRC: 0.004<br />cardiotoxic","Diabetes:  NA<br />CRC: 0.004<br />cardiotoxicity","Diabetes:  NA<br />CRC: 0.004<br />carriers","Diabetes:  NA<br />CRC: 0.004<br />cas","Diabetes:  NA<br />CRC: 0.004<br />cas9","Diabetes:  NA<br />CRC: 0.011<br />casc2","Diabetes:  NA<br />CRC: 0.004<br />casc2mediated","Diabetes:  NA<br />CRC: 0.004<br />casc7","Diabetes: 0.1<br />CRC: 0.007<br />caspase","Diabetes:  NA<br />CRC: 0.004<br />casr","Diabetes: 0.1<br />CRC:    NA<br />catalase","Diabetes:  NA<br />CRC: 0.004<br />cataloged","Diabetes:  NA<br />CRC: 0.004<br />catalytic","Diabetes: 0.1<br />CRC: 0.034<br />catenin","Diabetes:  NA<br />CRC: 0.007<br />caucasian","Diabetes:  NA<br />CRC: 0.004<br />cbs","Diabetes:  NA<br />CRC: 0.004<br />ccaat","Diabetes:  NA<br />CRC: 0.004<br />ccat2","Diabetes:  NA<br />CRC: 0.004<br />cck","Diabetes:  NA<br />CRC: 0.011<br />ccl20","Diabetes:  NA<br />CRC: 0.004<br />ccna2","Diabetes:  NA<br />CRC: 0.004<br />ccr","Diabetes:  NA<br />CRC: 0.004<br />cd133","Diabetes:  NA<br />CRC: 0.007<br />cd166","Diabetes:  NA<br />CRC: 0.004<br />cd24","Diabetes:  NA<br />CRC: 0.004<br />cd28","Diabetes:  NA<br />CRC: 0.004<br />cd3","Diabetes:  NA<br />CRC: 0.004<br />cd34","Diabetes:  NA<br />CRC: 0.022<br />cd44","Diabetes:  NA<br />CRC: 0.004<br />cd45ro","Diabetes:  NA<br />CRC: 0.004<br />cd63","Diabetes:  NA<br />CRC: 0.004<br />cd68","Diabetes:  NA<br />CRC: 0.004<br />cd8","Diabetes:  NA<br />CRC: 0.004<br />cd80","Diabetes:  NA<br />CRC: 0.004<br />cd81","Diabetes:  NA<br />CRC: 0.004<br />cd9","Diabetes:  NA<br />CRC: 0.004<br />cdc25a","Diabetes:  NA<br />CRC: 0.007<br />cdf","Diabetes:  NA<br />CRC: 0.004<br />cdk6","Diabetes:  NA<br />CRC: 0.007<br />cdkn1a","Diabetes:  NA<br />CRC: 0.045<br />cea","Diabetes:  NA<br />CRC: 0.007<br />cel","Diabetes:  NA<br />CRC: 0.004<br />cellsievo","Diabetes: 0.1<br />CRC: 0.004<br />cellulose","Diabetes:  NA<br />CRC: 0.004<br />centre","Diabetes:  NA<br />CRC: 0.004<br />cerebrospinal","Diabetes:  NA<br />CRC: 0.004<br />cervical","Diabetes:  NA<br />CRC: 0.007<br />cetuximab","Diabetes:  NA<br />CRC: 0.004<br />cfdna","Diabetes:  NA<br />CRC: 0.004<br />cflar","Diabetes:  NA<br />CRC: 0.004<br />cfna","Diabetes:  NA<br />CRC: 0.004<br />characterises","Diabetes:  NA<br />CRC: 0.007<br />characterizing","Diabetes:  NA<br />CRC: 0.004<br />checkpoint","Diabetes:  NA<br />CRC: 0.007<br />chemically","Diabetes:  NA<br />CRC: 0.011<br />chemo","Diabetes:  NA<br />CRC: 0.004<br />chemoembolization","Diabetes:  NA<br />CRC: 0.011<br />chemokine","Diabetes:  NA<br />CRC: 0.004<br />chemoprotective","Diabetes:  NA<br />CRC: 0.022<br />chemoradiotherapy","Diabetes:  NA<br />CRC: 0.011<br />chemoresistance","Diabetes:  NA<br />CRC: 0.004<br />chemoresistant","Diabetes:  NA<br />CRC: 0.004<br />chemosensitive","Diabetes:  NA<br />CRC: 0.007<br />chemosensitivity","Diabetes:  NA<br />CRC: 0.011<br />chemotherapeutic","Diabetes:  NA<br />CRC: 0.007<br />chemotherapeutics","Diabetes:  NA<br />CRC: 0.067<br />chemotherapy","Diabetes:  NA<br />CRC: 0.004<br />chick","Diabetes:  NA<br />CRC: 0.007<br />chicken","Diabetes:  NA<br />CRC: 0.019<br />china","Diabetes:  NA<br />CRC: 0.004<br />chips","Diabetes:  NA<br />CRC: 0.004<br />chl1","Diabetes:  NA<br />CRC: 0.004<br />chli","Diabetes: 0.1<br />CRC:    NA<br />cho","Diabetes: 0.1<br />CRC:    NA<br />cholesterol","Diabetes:  NA<br />CRC: 0.004<br />chorioallantoic","Diabetes:  NA<br />CRC: 0.004<br />chorionallantoic","Diabetes:  NA<br />CRC: 0.004<br />chr","Diabetes:  NA<br />CRC: 0.011<br />chromatin","Diabetes:  NA<br />CRC: 0.011<br />chromogenic","Diabetes:  NA<br />CRC: 0.004<br />chromosomes","Diabetes:  NA<br />CRC: 0.004<br />chtn06","Diabetes:  NA<br />CRC: 0.004<br />cinahl","Diabetes:  NA<br />CRC: 0.004<br />circ_0026344","Diabetes:  NA<br />CRC: 0.007<br />circrna","Diabetes:  NA<br />CRC: 0.004<br />circrnas","Diabetes:  NA<br />CRC: 0.004<br />circulatory","Diabetes:  NA<br />CRC: 0.004<br />circumstance","Diabetes:  NA<br />CRC: 0.007<br />cis","Diabetes: 0.1<br />CRC:    NA<br />city","Diabetes:  NA<br />CRC: 0.004<br />ck19","Diabetes:  NA<br />CRC: 0.004<br />ck20","Diabetes:  NA<br />CRC: 0.004<br />classifier","Diabetes:  NA<br />CRC: 0.007<br />cleavage","Diabetes: 0.1<br />CRC:    NA<br />cleaved","Diabetes:  NA<br />CRC: 0.004<br />cleaving","Diabetes:  NA<br />CRC: 0.004<br />clin","Diabetes:  NA<br />CRC: 0.004<br />clinicopath","Diabetes:  NA<br />CRC: 0.004<br />clm","Diabetes:  NA<br />CRC: 0.004<br />closest","Diabetes:  NA<br />CRC: 0.004<br />cmv","Diabetes:  NA<br />CRC: 0.004<br />coa","Diabetes:  NA<br />CRC: 0.007<br />coad","Diabetes: 0.1<br />CRC:    NA<br />coded","Diabetes:  NA<br />CRC: 0.004<br />codes","Diabetes: 0.1<br />CRC:    NA<br />coincident","Diabetes:  NA<br />CRC: 0.007<br />colectomy","Diabetes:  NA<br />CRC: 0.011<br />coli","Diabetes:  NA<br />CRC: 0.030<br />colitis","Diabetes: 0.1<br />CRC:    NA<br />collagen","Diabetes:  NA<br />CRC: 0.004<br />colo201","Diabetes:  NA<br />CRC: 0.004<br />colo206f","Diabetes:  NA<br />CRC: 0.004<br />colo32","Diabetes:  NA<br />CRC: 0.004<br />colo320","Diabetes:  NA<br />CRC: 0.004<br />colocalization","Diabetes:  NA<br />CRC: 0.004<br />colocytes","Diabetes:  NA<br />CRC: 0.356<br />colon","Diabetes:  NA<br />CRC: 0.075<br />colonic","Diabetes:  NA<br />CRC: 0.015<br />colonocytes","Diabetes:  NA<br />CRC: 0.004<br />colonoscopic","Diabetes:  NA<br />CRC: 0.004<br />colonoscopies","Diabetes:  NA<br />CRC: 0.011<br />colonoscopy","Diabetes:  NA<br />CRC: 0.007<br />colonosphere","Diabetes:  NA<br />CRC: 0.004<br />colonospheres","Diabetes:  NA<br />CRC: 0.779<br />colorectal","Diabetes:  NA<br />CRC: 0.007<br />colorectum","Diabetes:  NA<br />CRC: 0.004<br />commonest","Diabetes:  NA<br />CRC: 0.011<br />communication","Diabetes:  NA<br />CRC: 0.007<br />comparable","Diabetes:  NA<br />CRC: 0.011<br />compartment","Diabetes:  NA<br />CRC: 0.007<br />compartments","Diabetes:  NA<br />CRC: 0.004<br />compatible","Diabetes:  NA<br />CRC: 0.004<br />compensated","Diabetes:  NA<br />CRC: 0.004<br />compiled","Diabetes:  NA<br />CRC: 0.007<br />complement","Diabetes:  NA<br />CRC: 0.004<br />complicating","Diabetes:  NA<br />CRC: 0.007<br />computationally","Diabetes:  NA<br />CRC: 0.004<br />conclusively","Diabetes:  NA<br />CRC: 0.004<br />confl","Diabetes:  NA<br />CRC: 0.004<br />confound","Diabetes:  NA<br />CRC: 0.004<br />conjunctival","Diabetes:  NA<br />CRC: 0.004<br />connecting","Diabetes:  NA<br />CRC: 0.004<br />contamination","Diabetes:  NA<br />CRC: 0.004<br />contention","Diabetes:  NA<br />CRC: 0.004<br />continuation","Diabetes:  NA<br />CRC: 0.004<br />corn","Diabetes:  NA<br />CRC: 0.007<br />correctly","Diabetes:  NA<br />CRC: 0.004<br />correlationship","Diabetes: 0.1<br />CRC:    NA<br />cortex","Diabetes: 0.1<br />CRC:    NA<br />cortical","Diabetes:  NA<br />CRC: 0.004<br />counselling","Diabetes:  NA<br />CRC: 0.004<br />counted","Diabetes: 0.1<br />CRC: 0.007<br />covariates","Diabetes:  NA<br />CRC: 0.060<br />cox","Diabetes:  NA<br />CRC: 0.007<br />cpg","Diabetes:  NA<br />CRC: 0.004<br />cq","Diabetes:  NA<br />CRC: 0.011<br />cra","Diabetes:  NA<br />CRC: 0.007<br />cras","Diabetes:  NA<br />CRC: 0.464<br />crc","Diabetes:  NA<br />CRC: 0.004<br />crch37","Diabetes:  NA<br />CRC: 0.004<br />crch38","Diabetes:  NA<br />CRC: 0.045<br />crcs","Diabetes:  NA<br />CRC: 0.004<br />crispr","Diabetes:  NA<br />CRC: 0.004<br />criterion","Diabetes:  NA<br />CRC: 0.004<br />crohn's","Diabetes:  NA<br />CRC: 0.011<br />crosstalk","Diabetes:  NA<br />CRC: 0.011<br />crt","Diabetes:  NA<br />CRC: 0.004<br />cruciferous","Diabetes:  NA<br />CRC: 0.007<br />crypt","Diabetes:  NA<br />CRC: 0.011<br />cscs","Diabetes:  NA<br />CRC: 0.004<br />csf","Diabetes:  NA<br />CRC: 0.007<br />cslc","Diabetes:  NA<br />CRC: 0.007<br />cslcs","Diabetes:  NA<br />CRC: 0.007<br />css","Diabetes:  NA<br />CRC: 0.004<br />ctnnb1","Diabetes:  NA<br />CRC: 0.004<br />ctsc","Diabetes:  NA<br />CRC: 0.004<br />curable","Diabetes:  NA<br />CRC: 0.011<br />curcumin","Diabetes:  NA<br />CRC: 0.004<br />curcumol","Diabetes:  NA<br />CRC: 0.007<br />cyclin","Diabetes:  NA<br />CRC: 0.007<br />cyclooxygenase","Diabetes:  NA<br />CRC: 0.004<br />cyld","Diabetes:  NA<br />CRC: 0.004<br />cysteine","Diabetes:  NA<br />CRC: 0.015<br />cytokeratin","Diabetes:  NA<br />CRC: 0.026<br />cytokine","Diabetes:  NA<br />CRC: 0.004<br />cytologic","Diabetes:  NA<br />CRC: 0.004<br />cytological","Diabetes:  NA<br />CRC: 0.004<br />cytolysis","Diabetes:  NA<br />CRC: 0.004<br />cytoplasm","Diabetes:  NA<br />CRC: 0.004<br />cytoplasmic","Diabetes:  NA<br />CRC: 0.004<br />cytotoxic","Diabetes:  NA<br />CRC: 0.004<br />czech","Diabetes:  NA<br />CRC: 0.007<br />d1","Diabetes:  NA<br />CRC: 0.004<br />d15","Diabetes:  NA<br />CRC: 0.004<br />dairy","Diabetes:  NA<br />CRC: 0.007<br />damage","Diabetes:  NA<br />CRC: 0.004<br />dams","Diabetes:  NA<br />CRC: 0.004<br />dawley","Diabetes:  NA<br />CRC: 0.004<br />dcbeads","Diabetes:  NA<br />CRC: 0.004<br />dcc","Diabetes:  NA<br />CRC: 0.004<br />dcr","Diabetes:  NA<br />CRC: 0.004<br />ddpcr","Diabetes:  NA<br />CRC: 0.004<br />deacetylase","Diabetes:  NA<br />CRC: 0.004<br />deactivation","Diabetes:  NA<br />CRC: 0.004<br />december","Diabetes:  NA<br />CRC: 0.004<br />decide","Diabetes:  NA<br />CRC: 0.004<br />declare","Diabetes:  NA<br />CRC: 0.004<br />declares","Diabetes:  NA<br />CRC: 0.004<br />deemed","Diabetes:  NA<br />CRC: 0.004<br />deepest","Diabetes:  NA<br />CRC: 0.004<br />deficiencies","Diabetes: 0.1<br />CRC:    NA<br />degrades","Diabetes:  NA<br />CRC: 0.011<br />dehydrogenase","Diabetes:  NA<br />CRC: 0.004<br />delaying","Diabetes:  NA<br />CRC: 0.004<br />delineated","Diabetes: 0.1<br />CRC:    NA<br />delta","Diabetes:  NA<br />CRC: 0.004<br />deltact","Diabetes:  NA<br />CRC: 0.004<br />deltadeltact","Diabetes:  NA<br />CRC: 0.004<br />demarcate","Diabetes:  NA<br />CRC: 0.004<br />demethylation","Diabetes:  NA<br />CRC: 0.004<br />denmark","Diabetes:  NA<br />CRC: 0.004<br />denoted","Diabetes:  NA<br />CRC: 0.004<br />deoxyribonucleic","Diabetes:  NA<br />CRC: 0.004<br />depended","Diabetes:  NA<br />CRC: 0.004<br />deranged","Diabetes:  NA<br />CRC: 0.004<br />dermatological","Diabetes:  NA<br />CRC: 0.004<br />deseq2","Diabetes:  NA<br />CRC: 0.004<br />desmoid","Diabetes:  NA<br />CRC: 0.004<br />destruction","Diabetes:  NA<br />CRC: 0.004<br />devel","Diabetes:  NA<br />CRC: 0.011<br />dextran","Diabetes:  NA<br />CRC: 0.004<br />dfi","Diabetes:  NA<br />CRC: 0.004<br />dha","Diabetes:  NA<br />CRC: 0.004<br />dhfr","Diabetes: 0.7<br />CRC:    NA<br />diabetes","Diabetes: 0.5<br />CRC:    NA<br />diabetic","Diabetes:  NA<br />CRC: 0.004<br />diarrhea","Diabetes: 0.1<br />CRC: 0.011<br />dicer","Diabetes:  NA<br />CRC: 0.004<br />die","Diabetes:  NA<br />CRC: 0.004<br />diets","Diabetes:  NA<br />CRC: 0.019<br />differentiating","Diabetes:  NA<br />CRC: 0.004<br />differentstages","Diabetes:  NA<br />CRC: 0.007<br />difficulties","Diabetes:  NA<br />CRC: 0.007<br />difluorinated","Diabetes:  NA<br />CRC: 0.004<br />digested","Diabetes:  NA<br />CRC: 0.004<br />digestion","Diabetes:  NA<br />CRC: 0.004<br />digestive","Diabetes:  NA<br />CRC: 0.004<br />dihydropyrimidine","Diabetes:  NA<br />CRC: 0.004<br />dilutions","Diabetes:  NA<br />CRC: 0.004<br />dimethylhydrazine","Diabetes:  NA<br />CRC: 0.019<br />dimethylthiazol","Diabetes:  NA<br />CRC: 0.007<br />diphenyl","Diabetes:  NA<br />CRC: 0.011<br />diphenyltetrazolium","Diabetes:  NA<br />CRC: 0.004<br />disappearance","Diabetes:  NA<br />CRC: 0.004<br />discern","Diabetes:  NA<br />CRC: 0.004<br />discriminant","Diabetes:  NA<br />CRC: 0.004<br />dispersed","Diabetes:  NA<br />CRC: 0.004<br />disrupting","Diabetes:  NA<br />CRC: 0.004<br />dissection","Diabetes:  NA<br />CRC: 0.007<br />disseminated","Diabetes: 0.1<br />CRC: 0.019<br />distal","Diabetes:  NA<br />CRC: 0.004<br />distinctions","Diabetes:  NA<br />CRC: 0.026<br />dld","Diabetes:  NA<br />CRC: 0.004<br />dld1","Diabetes:  NA<br />CRC: 0.004<br />dmgs","Diabetes:  NA<br />CRC: 0.004<br />dmh","Diabetes: 0.1<br />CRC:    NA<br />dn","Diabetes:  NA<br />CRC: 0.060<br />dna","Diabetes:  NA<br />CRC: 0.004<br />docking","Diabetes:  NA<br />CRC: 0.004<br />docosahexaenoic","Diabetes:  NA<br />CRC: 0.004<br />dominated","Diabetes:  NA<br />CRC: 0.004<br />dor","Diabetes:  NA<br />CRC: 0.004<br />downmodulation","Diabetes:  NA<br />CRC: 0.007<br />dox","Diabetes:  NA<br />CRC: 0.011<br />doxorubicin","Diabetes:  NA<br />CRC: 0.004<br />dpd","Diabetes:  NA<br />CRC: 0.004<br />dra","Diabetes:  NA<br />CRC: 0.004<br />droplet","Diabetes:  NA<br />CRC: 0.004<br />dropped","Diabetes:  NA<br />CRC: 0.004<br />dt","Diabetes:  NA<br />CRC: 0.004<br />dtc","Diabetes:  NA<br />CRC: 0.004<br />dtcs","Diabetes:  NA<br />CRC: 0.004<br />dts","Diabetes:  NA<br />CRC: 0.004<br />dualluciferase","Diabetes:  NA<br />CRC: 0.004<br />duct","Diabetes:  NA<br />CRC: 0.007<br />ductal","Diabetes:  NA<br />CRC: 0.015<br />dukes","Diabetes:  NA<br />CRC: 0.011<br />duplex","Diabetes:  NA<br />CRC: 0.004<br />dvt","Diabetes:  NA<br />CRC: 0.004<br />dworak","Diabetes:  NA<br />CRC: 0.015<br />dysplasia","Diabetes:  NA<br />CRC: 0.007<br />dysplastic","Diabetes:  NA<br />CRC: 0.004<br />dysregulate","Diabetes: 0.1<br />CRC:    NA<br />e15.5","Diabetes: 0.1<br />CRC:    NA<br />e16.5","Diabetes: 0.1<br />CRC:    NA<br />e18.5","Diabetes:  NA<br />CRC: 0.004<br />e1b","Diabetes:  NA<br />CRC: 0.011<br />e2","Diabetes:  NA<br />CRC: 0.004<br />eclipse","Diabetes:  NA<br />CRC: 0.011<br />ectopic","Diabetes:  NA<br />CRC: 0.004<br />edger","Diabetes:  NA<br />CRC: 0.004<br />efficiencies","Diabetes:  NA<br />CRC: 0.007<br />egf","Diabetes: 0.2<br />CRC: 0.015<br />egfr","Diabetes:  NA<br />CRC: 0.007<br />egypt","Diabetes:  NA<br />CRC: 0.004<br />egyptian","Diabetes:  NA<br />CRC: 0.004<br />eif4e","Diabetes:  NA<br />CRC: 0.004<br />electrical","Diabetes:  NA<br />CRC: 0.004<br />electrophoresis","Diabetes: 0.1<br />CRC:    NA<br />electroporation","Diabetes: 0.1<br />CRC:    NA<br />elegans","Diabetes:  NA<br />CRC: 0.004<br />elimination","Diabetes:  NA<br />CRC: 0.004<br />eluting","Diabetes:  NA<br />CRC: 0.007<br />embryo","Diabetes: 0.1<br />CRC:    NA<br />embryonic","Diabetes: 0.1<br />CRC:    NA<br />embryos","Diabetes:  NA<br />CRC: 0.004<br />emerald","Diabetes:  NA<br />CRC: 0.004<br />empty","Diabetes:  NA<br />CRC: 0.022<br />emt","Diabetes:  NA<br />CRC: 0.004<br />enclose","Diabetes:  NA<br />CRC: 0.004<br />enclosed","Diabetes:  NA<br />CRC: 0.004<br />encourage","Diabetes:  NA<br />CRC: 0.007<br />endogenously","Diabetes:  NA<br />CRC: 0.004<br />endometrial","Diabetes:  NA<br />CRC: 0.004<br />endonuclease","Diabetes:  NA<br />CRC: 0.004<br />endoperoxide","Diabetes:  NA<br />CRC: 0.011<br />endoscopic","Diabetes:  NA<br />CRC: 0.004<br />endoscopically","Diabetes:  NA<br />CRC: 0.007<br />endoscopy","Diabetes: 0.1<br />CRC: 0.015<br />endothelial","Diabetes:  NA<br />CRC: 0.004<br />energetic","Diabetes:  NA<br />CRC: 0.004<br />enigma","Diabetes: 0.1<br />CRC:    NA<br />enos","Diabetes:  NA<br />CRC: 0.004<br />enumerated","Diabetes:  NA<br />CRC: 0.004<br />environ","Diabetes:  NA<br />CRC: 0.004<br />enzymatic","Diabetes:  NA<br />CRC: 0.004<br />epcam","Diabetes: 0.1<br />CRC: 0.004<br />epcs","Diabetes: 0.1<br />CRC:    NA<br />epidemic","Diabetes:  NA<br />CRC: 0.004<br />epidemiol","Diabetes:  NA<br />CRC: 0.026<br />epidermal","Diabetes:  NA<br />CRC: 0.004<br />epigenetically","Diabetes: 0.2<br />CRC: 0.071<br />epithelial","Diabetes: 0.1<br />CRC:    NA<br />epithelialization","Diabetes:  NA<br />CRC: 0.007<br />epithelium","Diabetes:  NA<br />CRC: 0.004<br />equivalently","Diabetes:  NA<br />CRC: 0.004<br />er","Diabetes:  NA<br />CRC: 0.004<br />eraser","Diabetes:  NA<br />CRC: 0.011<br />erk","Diabetes:  NA<br />CRC: 0.007<br />erk1","Diabetes:  NA<br />CRC: 0.004<br />escc","Diabetes:  NA<br />CRC: 0.007<br />esophageal","Diabetes:  NA<br />CRC: 0.004<br />esophagus","Diabetes: 0.1<br />CRC:    NA<br />esrd","Diabetes:  NA<br />CRC: 0.007<br />estimates","Diabetes:  NA<br />CRC: 0.004<br />estrogen","Diabetes:  NA<br />CRC: 0.007<br />ethnicity","Diabetes:  NA<br />CRC: 0.004<br />etiologies","Diabetes:  NA<br />CRC: 0.004<br />etoposide","Diabetes:  NA<br />CRC: 0.004<br />ets","Diabetes:  NA<br />CRC: 0.004<br />ets1","Diabetes:  NA<br />CRC: 0.007<br />eukaryotic","Diabetes:  NA<br />CRC: 0.004<br />ev","Diabetes:  NA<br />CRC: 0.004<br />evasion","Diabetes:  NA<br />CRC: 0.004<br />evolves","Diabetes:  NA<br />CRC: 0.026<br />examination","Diabetes:  NA<br />CRC: 0.004<br />exclusive","Diabetes:  NA<br />CRC: 0.004<br />execute","Diabetes:  NA<br />CRC: 0.007<br />exfoliated","Diabetes:  NA<br />CRC: 0.004<br />exhaustive","Diabetes:  NA<br />CRC: 0.011<br />exiqon","Diabetes:  NA<br />CRC: 0.007<br />exogenous","Diabetes:  NA<br />CRC: 0.004<br />exophytic","Diabetes: 0.1<br />CRC:    NA<br />exos","Diabetes:  NA<br />CRC: 0.022<br />exosomal","Diabetes: 0.1<br />CRC: 0.011<br />exosome","Diabetes:  NA<br />CRC: 0.004<br />expired","Diabetes:  NA<br />CRC: 0.004<br />expose","Diabetes:  NA<br />CRC: 0.004<br />extending","Diabetes: 0.1<br />CRC: 0.015<br />extracellular","Diabetes:  NA<br />CRC: 0.004<br />extracting","Diabetes:  NA<br />CRC: 0.004<br />extraterminal","Diabetes:  NA<br />CRC: 0.004<br />eyes","Diabetes:  NA<br />CRC: 0.004<br />ezh2","Diabetes:  NA<br />CRC: 0.004<br />ezna","Diabetes:  NA<br />CRC: 0.004<br />faecal","Diabetes:  NA<br />CRC: 0.004<br />failing","Diabetes:  NA<br />CRC: 0.011<br />false","Diabetes:  NA<br />CRC: 0.007<br />fap","Diabetes:  NA<br />CRC: 0.004<br />fatality","Diabetes:  NA<br />CRC: 0.004<br />fatty","Diabetes:  NA<br />CRC: 0.004<br />favourable","Diabetes:  NA<br />CRC: 0.004<br />february","Diabetes:  NA<br />CRC: 0.007<br />feces","Diabetes:  NA<br />CRC: 0.011<br />females","Diabetes:  NA<br />CRC: 0.041<br />ffpe","Diabetes:  NA<br />CRC: 0.004<br />fhit","Diabetes:  NA<br />CRC: 0.022<br />fibroblast","Diabetes:  NA<br />CRC: 0.011<br />fibroblastic","Diabetes:  NA<br />CRC: 0.026<br />fibroblasts","Diabetes:  NA<br />CRC: 0.004<br />fibronectin","Diabetes: 0.2<br />CRC: 0.004<br />fibrosis","Diabetes: 0.1<br />CRC:    NA<br />fibrotic","Diabetes:  NA<br />CRC: 0.004<br />fic","Diabetes:  NA<br />CRC: 0.004<br />fidarestat","Diabetes:  NA<br />CRC: 0.004<br />fide","Diabetes: 0.1<br />CRC:    NA<br />filtration","Diabetes:  NA<br />CRC: 0.007<br />fisher's","Diabetes:  NA<br />CRC: 0.011<br />flat","Diabetes:  NA<br />CRC: 0.004<br />flight","Diabetes:  NA<br />CRC: 0.007<br />fluidigm","Diabetes:  NA<br />CRC: 0.004<br />fluids","Diabetes:  NA<br />CRC: 0.041<br />fluorouracil","Diabetes:  NA<br />CRC: 0.004<br />flurouracil","Diabetes:  NA<br />CRC: 0.004<br />flux","Diabetes:  NA<br />CRC: 0.004<br />foci","Diabetes:  NA<br />CRC: 0.004<br />folding","Diabetes:  NA<br />CRC: 0.004<br />folfox","Diabetes:  NA<br />CRC: 0.004<br />forecasted","Diabetes:  NA<br />CRC: 0.004<br />forkhead","Diabetes:  NA<br />CRC: 0.004<br />formulated","Diabetes:  NA<br />CRC: 0.015<br />fos","Diabetes:  NA<br />CRC: 0.004<br />foxc2","Diabetes:  NA<br />CRC: 0.004<br />foxo","Diabetes:  NA<br />CRC: 0.007<br />foxo3a","Diabetes:  NA<br />CRC: 0.007<br />foxp3","Diabetes:  NA<br />CRC: 0.004<br />francis","Diabetes:  NA<br />CRC: 0.004<br />freely","Diabetes:  NA<br />CRC: 0.007<br />freshly","Diabetes:  NA<br />CRC: 0.004<br />friendly","Diabetes:  NA<br />CRC: 0.030<br />fu","Diabetes: 0.1<br />CRC:    NA<br />fundamentals","Diabetes:  NA<br />CRC: 0.004<br />funding","Diabetes:  NA<br />CRC: 0.004<br />fuox","Diabetes:  NA<br />CRC: 0.007<br />fusobacterium","Diabetes:  NA<br />CRC: 0.007<br />g2","Diabetes:  NA<br />CRC: 0.004<br />g6c","Diabetes: 0.1<br />CRC:    NA<br />galactosidase","Diabetes:  NA<br />CRC: 0.004<br />gapdh","Diabetes:  NA<br />CRC: 0.004<br />gas","Diabetes:  NA<br />CRC: 0.022<br />gastric","Diabetes:  NA<br />CRC: 0.007<br />gastrointestinal","Diabetes:  NA<br />CRC: 0.004<br />gavage","Diabetes:  NA<br />CRC: 0.004<br />gavaged","Diabetes:  NA<br />CRC: 0.015<br />gc","Diabetes:  NA<br />CRC: 0.004<br />gd1","Diabetes: 0.1<br />CRC: 0.004<br />gel","Diabetes:  NA<br />CRC: 0.004<br />generator","Diabetes:  NA<br />CRC: 0.004<br />genesis","Diabetes:  NA<br />CRC: 0.004<br />genotoxic","Diabetes:  NA<br />CRC: 0.004<br />geo2r","Diabetes:  NA<br />CRC: 0.004<br />geometrically","Diabetes:  NA<br />CRC: 0.004<br />german","Diabetes:  NA<br />CRC: 0.004<br />germany","Diabetes: 0.1<br />CRC:    NA<br />gestation","Diabetes:  NA<br />CRC: 0.004<br />gestational","Diabetes:  NA<br />CRC: 0.004<br />giemsa","Diabetes:  NA<br />CRC: 0.004<br />git","Diabetes:  NA<br />CRC: 0.004<br />github.com","Diabetes:  NA<br />CRC: 0.004<br />gland","Diabetes:  NA<br />CRC: 0.004<br />glands","Diabetes:  NA<br />CRC: 0.015<br />glioma","Diabetes:  NA<br />CRC: 0.004<br />gliomas","Diabetes:  NA<br />CRC: 0.004<br />globule","Diabetes: 0.1<br />CRC:    NA<br />glomerular","Diabetes: 0.4<br />CRC:    NA<br />glucose","Diabetes:  NA<br />CRC: 0.004<br />glutamate","Diabetes:  NA<br />CRC: 0.004<br />glycation","Diabetes: 0.1<br />CRC:    NA<br />glycemic","Diabetes:  NA<br />CRC: 0.004<br />glycerophospholipid","Diabetes:  NA<br />CRC: 0.004<br />glycolysis","Diabetes:  NA<br />CRC: 0.004<br />glycosphingolipid","Diabetes: 0.1<br />CRC:    NA<br />glycosylated","Diabetes:  NA<br />CRC: 0.004<br />gmos","Diabetes:  NA<br />CRC: 0.004<br />graded","Diabetes:  NA<br />CRC: 0.004<br />grades","Diabetes:  NA<br />CRC: 0.004<br />granular","Diabetes:  NA<br />CRC: 0.004<br />graphical","Diabetes:  NA<br />CRC: 0.004<br />gravity","Diabetes:  NA<br />CRC: 0.004<br />grew","Diabetes:  NA<br />CRC: 0.011<br />groove","Diabetes:  NA<br />CRC: 0.004<br />groundwater","Diabetes: 0.2<br />CRC: 0.146<br />growth","Diabetes:  NA<br />CRC: 0.004<br />gse35834","Diabetes:  NA<br />CRC: 0.004<br />gse37178","Diabetes:  NA<br />CRC: 0.004<br />gse68468","Diabetes:  NA<br />CRC: 0.004<br />gse77955","Diabetes:  NA<br />CRC: 0.004<br />gse81558","Diabetes:  NA<br />CRC: 0.004<br />gse81582","Diabetes:  NA<br />CRC: 0.004<br />gse90709","Diabetes:  NA<br />CRC: 0.004<br />gt","Diabetes:  NA<br />CRC: 0.004<br />gtp","Diabetes:  NA<br />CRC: 0.011<br />gtpase","Diabetes: 0.1<br />CRC:    NA<br />gtt","Diabetes:  NA<br />CRC: 0.004<br />guanidine","Diabetes:  NA<br />CRC: 0.004<br />h1","Diabetes:  NA<br />CRC: 0.004<br />h3k27","Diabetes:  NA<br />CRC: 0.004<br />h3k27m3","Diabetes:  NA<br />CRC: 0.004<br />h3k27me3","Diabetes:  NA<br />CRC: 0.004<br />h4","Diabetes:  NA<br />CRC: 0.004<br />haemoglobin","Diabetes:  NA<br />CRC: 0.004<br />haemolysis","Diabetes: 0.1<br />CRC: 0.007<br />hairpin","Diabetes:  NA<br />CRC: 0.004<br />hamsb","Diabetes: 0.1<br />CRC:    NA<br />hasc","Diabetes: 0.1<br />CRC:    NA<br />hba1c","Diabetes: 0.1<br />CRC:    NA<br />hbac1","Diabetes:  NA<br />CRC: 0.004<br />hcas","Diabetes:  NA<br />CRC: 0.004<br />hcs","Diabetes:  NA<br />CRC: 0.037<br />hct","Diabetes:  NA<br />CRC: 0.045<br />hct116","Diabetes:  NA<br />CRC: 0.004<br />hd","Diabetes:  NA<br />CRC: 0.004<br />hdac","Diabetes:  NA<br />CRC: 0.004<br />hdacis","Diabetes: 0.1<br />CRC:    NA<br />hdl","Diabetes:  NA<br />CRC: 0.007<br />head","Diabetes: 0.1<br />CRC: 0.004<br />healing","Diabetes: 0.1<br />CRC: 0.022<br />health","Diabetes:  NA<br />CRC: 0.187<br />healthy","Diabetes:  NA<br />CRC: 0.004<br />heart","Diabetes: 0.1<br />CRC:    NA<br />heat","Diabetes:  NA<br />CRC: 0.004<br />hek293","Diabetes: 0.2<br />CRC: 0.004<br />hemoglobin","Diabetes:  NA<br />CRC: 0.004<br />hemolysis","Diabetes:  NA<br />CRC: 0.004<br />hemolyzed","Diabetes:  NA<br />CRC: 0.007<br />hepatic","Diabetes:  NA<br />CRC: 0.007<br />hepatocellular","Diabetes:  NA<br />CRC: 0.004<br />hepatoma","Diabetes:  NA<br />CRC: 0.004<br />hepg2","Diabetes:  NA<br />CRC: 0.007<br />hereditary","Diabetes: 0.1<br />CRC:    NA<br />hes1","Diabetes:  NA<br />CRC: 0.004<br />heterocyclic","Diabetes:  NA<br />CRC: 0.004<br />heterogeneities","Diabetes:  NA<br />CRC: 0.004<br />hg","Diabetes:  NA<br />CRC: 0.004<br />hif","Diabetes:  NA<br />CRC: 0.004<br />hilden","Diabetes:  NA<br />CRC: 0.004<br />hispanic","Diabetes:  NA<br />CRC: 0.004<br />histologic","Diabetes: 0.1<br />CRC: 0.004<br />histologically","Diabetes:  NA<br />CRC: 0.007<br />histone","Diabetes:  NA<br />CRC: 0.004<br />histopathologic","Diabetes:  NA<br />CRC: 0.007<br />histopathology","Diabetes:  NA<br />CRC: 0.004<br />hla","Diabetes:  NA<br />CRC: 0.004<br />hmc","Diabetes:  NA<br />CRC: 0.004<br />hmga2","Diabetes:  NA<br />CRC: 0.007<br />hmsh2","Diabetes:  NA<br />CRC: 0.004<br />hmsh6","Diabetes:  NA<br />CRC: 0.004<br />hnscc","Diabetes: 0.1<br />CRC:    NA<br />homa","Diabetes: 0.2<br />CRC:    NA<br />homeostasis","Diabetes:  NA<br />CRC: 0.004<br />homing","Diabetes:  NA<br />CRC: 0.004<br />homo","Diabetes:  NA<br />CRC: 0.004<br />homogeneous","Diabetes: 0.1<br />CRC: 0.037<br />homolog","Diabetes:  NA<br />CRC: 0.004<br />homolog2","Diabetes:  NA<br />CRC: 0.007<br />homologue","Diabetes:  NA<br />CRC: 0.004<br />homozygotes","Diabetes:  NA<br />CRC: 0.004<br />hong","Diabetes:  NA<br />CRC: 0.004<br />hospitalized","Diabetes:  NA<br />CRC: 0.004<br />hot","Diabetes:  NA<br />CRC: 0.004<br />hoxd10","Diabetes:  NA<br />CRC: 0.004<br />hpp","Diabetes:  NA<br />CRC: 0.004<br />hps","Diabetes:  NA<br />CRC: 0.004<br />hpv","Diabetes:  NA<br />CRC: 0.004<br />hrm","Diabetes:  NA<br />CRC: 0.034<br />hsa","Diabetes:  NA<br />CRC: 0.030<br />ht29","Diabetes:  NA<br />CRC: 0.004<br />ht55","Diabetes:  NA<br />CRC: 0.007<br />htert","Diabetes:  NA<br />CRC: 0.004<br />html","Diabetes:  NA<br />CRC: 0.011<br />hybridisation","Diabetes:  NA<br />CRC: 0.007<br />hydrogen","Diabetes:  NA<br />CRC: 0.004<br />hydrophobic","Diabetes:  NA<br />CRC: 0.004<br />hydroxymethylcytosine","Diabetes:  NA<br />CRC: 0.007<br />hyperplastic","Diabetes: 0.1<br />CRC:    NA<br />hypertensive","Diabetes:  NA<br />CRC: 0.004<br />hypertrophic","Diabetes: 0.1<br />CRC:    NA<br />hypoglycemic","Diabetes: 0.1<br />CRC:    NA<br />hypothesised","Diabetes:  NA<br />CRC: 0.007<br />hypoxia","Diabetes:  NA<br />CRC: 0.004<br />hypoxic","Diabetes:  NA<br />CRC: 0.004<br />i2","Diabetes:  NA<br />CRC: 0.007<br />ibds","Diabetes:  NA<br />CRC: 0.007<br />ic50","Diabetes:  NA<br />CRC: 0.004<br />icam","Diabetes:  NA<br />CRC: 0.004<br />icc","Diabetes:  NA<br />CRC: 0.004<br />iccmean","Diabetes:  NA<br />CRC: 0.004<br />iccs","Diabetes:  NA<br />CRC: 0.004<br />iccsingle","Diabetes:  NA<br />CRC: 0.004<br />icmje","Diabetes:  NA<br />CRC: 0.004<br />icos","Diabetes:  NA<br />CRC: 0.004<br />icts","Diabetes:  NA<br />CRC: 0.004<br />identifiable","Diabetes:  NA<br />CRC: 0.004<br />identities","Diabetes:  NA<br />CRC: 0.004<br />ien","Diabetes:  NA<br />CRC: 0.004<br />ifobt","Diabetes:  NA<br />CRC: 0.004<br />igf2bp1","Diabetes:  NA<br />CRC: 0.022<br />il","Diabetes:  NA<br />CRC: 0.004<br />il1b","Diabetes:  NA<br />CRC: 0.011<br />il6","Diabetes:  NA<br />CRC: 0.004<br />il6r","Diabetes:  NA<br />CRC: 0.004<br />il8","Diabetes:  NA<br />CRC: 0.004<br />imac","Diabetes:  NA<br />CRC: 0.004<br />images","Diabetes:  NA<br />CRC: 0.004<br />immortalised","Diabetes:  NA<br />CRC: 0.004<br />immortalization","Diabetes:  NA<br />CRC: 0.004<br />immortalized","Diabetes:  NA<br />CRC: 0.022<br />immune","Diabetes:  NA<br />CRC: 0.007<br />immunoblot","Diabetes:  NA<br />CRC: 0.004<br />immunocaptured","Diabetes:  NA<br />CRC: 0.004<br />immunocytochemistry","Diabetes:  NA<br />CRC: 0.004<br />immunodeficiency","Diabetes:  NA<br />CRC: 0.004<br />immunoexpression","Diabetes:  NA<br />CRC: 0.004<br />immunohistochemically","Diabetes:  NA<br />CRC: 0.019<br />immunoprecipitation","Diabetes:  NA<br />CRC: 0.004<br />immunoreactivities","Diabetes:  NA<br />CRC: 0.004<br />immunostain","Diabetes:  NA<br />CRC: 0.004<br />immunostaining","Diabetes:  NA<br />CRC: 0.004<br />immunotherapy","Diabetes:  NA<br />CRC: 0.004<br />imperfect","Diabetes:  NA<br />CRC: 0.004<br />impossible","Diabetes: 0.1<br />CRC:    NA<br />impressively","Diabetes:  NA<br />CRC: 0.004<br />individualise","Diabetes:  NA<br />CRC: 0.004<br />individualization","Diabetes:  NA<br />CRC: 0.004<br />individualizing","Diabetes:  NA<br />CRC: 0.007<br />inexpensive","Diabetes:  NA<br />CRC: 0.011<br />infected","Diabetes:  NA<br />CRC: 0.011<br />infection","Diabetes:  NA<br />CRC: 0.007<br />inferior","Diabetes:  NA<br />CRC: 0.004<br />infers","Diabetes: 0.1<br />CRC:    NA<br />infiltration","Diabetes:  NA<br />CRC: 0.004<br />infiltrative","Diabetes:  NA<br />CRC: 0.004<br />infinity","Diabetes: 0.3<br />CRC: 0.030<br />inflammation","Diabetes:  NA<br />CRC: 0.015<br />influencing","Diabetes:  NA<br />CRC: 0.007<br />informative","Diabetes:  NA<br />CRC: 0.004<br />ing3","Diabetes:  NA<br />CRC: 0.004<br />ingenuity","Diabetes:  NA<br />CRC: 0.004<br />injecting","Diabetes: 0.1<br />CRC:    NA<br />ink","Diabetes:  NA<br />CRC: 0.007<br />inoculation","Diabetes:  NA<br />CRC: 0.004<br />ins","Diabetes:  NA<br />CRC: 0.004<br />inspected","Diabetes:  NA<br />CRC: 0.004<br />instances","Diabetes: 0.3<br />CRC:    NA<br />insulin","Diabetes:  NA<br />CRC: 0.007<br />integrating","Diabetes:  NA<br />CRC: 0.004<br />integrin","Diabetes:  NA<br />CRC: 0.004<br />intercellular","Diabetes: 0.1<br />CRC: 0.015<br />interleukin","Diabetes:  NA<br />CRC: 0.004<br />intermolecular","Diabetes:  NA<br />CRC: 0.004<br />interobserver","Diabetes: 0.1<br />CRC:    NA<br />interpretation","Diabetes:  NA<br />CRC: 0.004<br />intersample","Diabetes:  NA<br />CRC: 0.004<br />intersected","Diabetes:  NA<br />CRC: 0.004<br />intertumoral","Diabetes:  NA<br />CRC: 0.007<br />intestinal","Diabetes:  NA<br />CRC: 0.004<br />intestine","Diabetes:  NA<br />CRC: 0.004<br />intraarterial","Diabetes:  NA<br />CRC: 0.004<br />intraclass","Diabetes:  NA<br />CRC: 0.007<br />intraepithelial","Diabetes:  NA<br />CRC: 0.004<br />intramuscular","Diabetes:  NA<br />CRC: 0.004<br />intraperitoneal","Diabetes:  NA<br />CRC: 0.007<br />intratumoral","Diabetes:  NA<br />CRC: 0.007<br />intravasation","Diabetes:  NA<br />CRC: 0.004<br />intrinsically","Diabetes:  NA<br />CRC: 0.004<br />invaluable","Diabetes:  NA<br />CRC: 0.004<br />inversion","Diabetes:  NA<br />CRC: 0.004<br />investigates","Diabetes:  NA<br />CRC: 0.007<br />iodide","Diabetes: 0.1<br />CRC:    NA<br />ipcs","Diabetes:  NA<br />CRC: 0.004<br />ipd","Diabetes: 0.1<br />CRC:    NA<br />ir","Diabetes:  NA<br />CRC: 0.004<br />iranian","Diabetes:  NA<br />CRC: 0.004<br />irinotecan","Diabetes:  NA<br />CRC: 0.007<br />irritable","Diabetes:  NA<br />CRC: 0.004<br />irritations","Diabetes:  NA<br />CRC: 0.004<br />irs","Diabetes:  NA<br />CRC: 0.007<br />island","Diabetes:  NA<br />CRC: 0.004<br />iso","Diabetes:  NA<br />CRC: 0.004<br />isoforms","Diabetes:  NA<br />CRC: 0.007<br />isothiocyanate","Diabetes:  NA<br />CRC: 0.004<br />italy","Diabetes:  NA<br />CRC: 0.004<br />iterations","Diabetes:  NA<br />CRC: 0.004<br />itgbeta4","Diabetes: 0.1<br />CRC:    NA<br />itt","Diabetes:  NA<br />CRC: 0.007<br />japan","Diabetes:  NA<br />CRC: 0.004<br />japanese","Diabetes:  NA<br />CRC: 0.004<br />java","Diabetes:  NA<br />CRC: 0.004<br />javascript","Diabetes:  NA<br />CRC: 0.004<br />jq","Diabetes:  NA<br />CRC: 0.007<br />jun","Diabetes:  NA<br />CRC: 0.004<br />junctions","Diabetes:  NA<br />CRC: 0.004<br />kaiser","Diabetes:  NA<br />CRC: 0.004<br />kale","Diabetes: 0.1<br />CRC:    NA<br />kallistatin","Diabetes: 0.1<br />CRC:    NA<br />kallistatin's","Diabetes:  NA<br />CRC: 0.004<br />kanaan","Diabetes:  NA<br />CRC: 0.041<br />kaplan","Diabetes:  NA<br />CRC: 0.022<br />kappab","Diabetes:  NA<br />CRC: 0.004<br />karyotyping","Diabetes:  NA<br />CRC: 0.004<br />kazal","Diabetes:  NA<br />CRC: 0.004<br />kd","Diabetes:  NA<br />CRC: 0.004<br />kda","Diabetes:  NA<br />CRC: 0.004<br />kegg","Diabetes: 0.1<br />CRC:    NA<br />keratinocytes","Diabetes:  NA<br />CRC: 0.004<br />ki","Diabetes:  NA<br />CRC: 0.004<br />ki67","Diabetes: 0.3<br />CRC: 0.004<br />kidney","Diabetes: 0.2<br />CRC:    NA<br />kidneys","Diabetes:  NA<br />CRC: 0.004<br />kilobase","Diabetes:  NA<br />CRC: 0.004<br />kinases","Diabetes:  NA<br />CRC: 0.004<br />kindred","Diabetes:  NA<br />CRC: 0.007<br />klf4","Diabetes:  NA<br />CRC: 0.004<br />km12c","Diabetes:  NA<br />CRC: 0.004<br />kong","Diabetes:  NA<br />CRC: 0.030<br />kras","Diabetes:  NA<br />CRC: 0.004<br />kruppel","Diabetes:  NA<br />CRC: 0.007<br />kruskall","Diabetes:  NA<br />CRC: 0.007<br />l1","Diabetes:  NA<br />CRC: 0.004<br />labeling","Diabetes:  NA<br />CRC: 0.004<br />laboratories","Diabetes:  NA<br />CRC: 0.004<br />ladder","Diabetes:  NA<br />CRC: 0.004<br />lamina","Diabetes:  NA<br />CRC: 0.007<br />laminin","Diabetes:  NA<br />CRC: 0.004<br />larc","Diabetes:  NA<br />CRC: 0.007<br />lasso","Diabetes:  NA<br />CRC: 0.004<br />laterally","Diabetes:  NA<br />CRC: 0.004<br />lcm","Diabetes: 0.1<br />CRC:    NA<br />ldl","Diabetes:  NA<br />CRC: 0.004<br />lean","Diabetes:  NA<br />CRC: 0.007<br />lef","Diabetes:  NA<br />CRC: 0.007<br />let7a","Diabetes:  NA<br />CRC: 0.007<br />let7b","Diabetes: 0.1<br />CRC: 0.004<br />lethal","Diabetes:  NA<br />CRC: 0.004<br />leucine","Diabetes:  NA<br />CRC: 0.004<br />leuconostoc","Diabetes:  NA<br />CRC: 0.007<br />leukemia","Diabetes:  NA<br />CRC: 0.004<br />leukemic","Diabetes:  NA<br />CRC: 0.004<br />leukocyte","Diabetes:  NA<br />CRC: 0.004<br />lg","Diabetes:  NA<br />CRC: 0.004<br />lif","Diabetes:  NA<br />CRC: 0.004<br />lif's","Diabetes: 0.1<br />CRC:    NA<br />lifespan","Diabetes:  NA<br />CRC: 0.004<br />ligand","Diabetes:  NA<br />CRC: 0.004<br />lin28","Diabetes:  NA<br />CRC: 0.004<br />linc01296","Diabetes:  NA<br />CRC: 0.004<br />linearity","Diabetes:  NA<br />CRC: 0.004<br />links.lww.com","Diabetes:  NA<br />CRC: 0.004<br />linregpcr","Diabetes:  NA<br />CRC: 0.004<br />linux","Diabetes: 0.1<br />CRC:    NA<br />lipoprotein","Diabetes:  NA<br />CRC: 0.004<br />liposomal","Diabetes:  NA<br />CRC: 0.004<br />liposome","Diabetes:  NA<br />CRC: 0.075<br />liver","Diabetes:  NA<br />CRC: 0.004<br />lmd","Diabetes:  NA<br />CRC: 0.004<br />lnccasc2","Diabetes:  NA<br />CRC: 0.026<br />lncrna","Diabetes:  NA<br />CRC: 0.015<br />lncrnas","Diabetes:  NA<br />CRC: 0.004<br />localisations","Diabetes:  NA<br />CRC: 0.004<br />localised","Diabetes:  NA<br />CRC: 0.004<br />log10","Diabetes:  NA<br />CRC: 0.004<br />log2","Diabetes:  NA<br />CRC: 0.004<br />longum","Diabetes:  NA<br />CRC: 0.011<br />lovo","Diabetes:  NA<br />CRC: 0.004<br />lowly","Diabetes:  NA<br />CRC: 0.004<br />lrrc24","Diabetes:  NA<br />CRC: 0.015<br />ls174t","Diabetes:  NA<br />CRC: 0.004<br />luca","Diabetes:  NA<br />CRC: 0.007<br />luminal","Diabetes:  NA<br />CRC: 0.056<br />lung","Diabetes:  NA<br />CRC: 0.049<br />lymph","Diabetes:  NA<br />CRC: 0.004<br />lymphatic","Diabetes:  NA<br />CRC: 0.004<br />lymphoblast","Diabetes:  NA<br />CRC: 0.004<br />lymphocytes","Diabetes:  NA<br />CRC: 0.007<br />lymphoid","Diabetes:  NA<br />CRC: 0.007<br />lymphoma","Diabetes:  NA<br />CRC: 0.004<br />lymphomas","Diabetes:  NA<br />CRC: 0.004<br />lymphonodal","Diabetes:  NA<br />CRC: 0.011<br />lymphovascular","Diabetes:  NA<br />CRC: 0.004<br />lynch","Diabetes:  NA<br />CRC: 0.004<br />m2","Diabetes:  NA<br />CRC: 0.004<br />ma","Diabetes:  NA<br />CRC: 0.004<br />macherey","Diabetes:  NA<br />CRC: 0.007<br />machine","Diabetes:  NA<br />CRC: 0.004<br />macrodissected","Diabetes:  NA<br />CRC: 0.007<br />macrophage","Diabetes: 0.1<br />CRC: 0.011<br />macrophages","Diabetes:  NA<br />CRC: 0.004<br />mag3","Diabetes:  NA<br />CRC: 0.004<br />magna","Diabetes:  NA<br />CRC: 0.007<br />magnitude","Diabetes:  NA<br />CRC: 0.004<br />maize","Diabetes:  NA<br />CRC: 0.004<br />maker","Diabetes:  NA<br />CRC: 0.004<br />male","Diabetes:  NA<br />CRC: 0.007<br />males","Diabetes:  NA<br />CRC: 0.004<br />malfunctions","Diabetes:  NA<br />CRC: 0.037<br />malignancies","Diabetes:  NA<br />CRC: 0.037<br />malignancy","Diabetes:  NA<br />CRC: 0.064<br />malignant","Diabetes:  NA<br />CRC: 0.004<br />malnutrition","Diabetes: 0.1<br />CRC:    NA<br />malondialdehyde","Diabetes:  NA<br />CRC: 0.004<br />mammalian","Diabetes:  NA<br />CRC: 0.004<br />mandard's","Diabetes:  NA<br />CRC: 0.007<br />mapk","Diabetes:  NA<br />CRC: 0.004<br />mapk1","Diabetes:  NA<br />CRC: 0.004<br />mapt","Diabetes:  NA<br />CRC: 0.004<br />march","Diabetes:  NA<br />CRC: 0.004<br />margin","Diabetes:  NA<br />CRC: 0.007<br />mark","Diabetes: 0.1<br />CRC: 0.154<br />markers","Diabetes:  NA<br />CRC: 0.004<br />market","Diabetes:  NA<br />CRC: 0.004<br />marking","Diabetes:  NA<br />CRC: 0.004<br />marrow","Diabetes:  NA<br />CRC: 0.004<br />maryland","Diabetes:  NA<br />CRC: 0.004<br />masaryk","Diabetes:  NA<br />CRC: 0.004<br />matches","Diabetes:  NA<br />CRC: 0.004<br />mathematical","Diabetes:  NA<br />CRC: 0.004<br />matrices","Diabetes:  NA<br />CRC: 0.004<br />matrigels","Diabetes:  NA<br />CRC: 0.011<br />matrix","Diabetes:  NA<br />CRC: 0.004<br />mbq","Diabetes:  NA<br />CRC: 0.004<br />mc","Diabetes:  NA<br />CRC: 0.004<br />mcf","Diabetes:  NA<br />CRC: 0.004<br />md","Diabetes:  NA<br />CRC: 0.004<br />mde","Diabetes:  NA<br />CRC: 0.004<br />mdm2","Diabetes:  NA<br />CRC: 0.004<br />meat","Diabetes:  NA<br />CRC: 0.007<br />mechanistically","Diabetes:  NA<br />CRC: 0.004<br />media","Diabetes:  NA<br />CRC: 0.011<br />medicine","Diabetes:  NA<br />CRC: 0.004<br />meditated","Diabetes:  NA<br />CRC: 0.004<br />mef2c","Diabetes:  NA<br />CRC: 0.041<br />meier","Diabetes:  NA<br />CRC: 0.004<br />mek","Diabetes:  NA<br />CRC: 0.011<br />melanoma","Diabetes: 0.3<br />CRC:    NA<br />mellitus","Diabetes:  NA<br />CRC: 0.011<br />membrane","Diabetes:  NA<br />CRC: 0.004<br />menace","Diabetes:  NA<br />CRC: 0.004<br />merited","Diabetes: 0.1<br />CRC:    NA<br />mesangial","Diabetes:  NA<br />CRC: 0.034<br />mesenchymal","Diabetes:  NA<br />CRC: 0.004<br />mesenteric","Diabetes:  NA<br />CRC: 0.004<br />mesenteroides","Diabetes:  NA<br />CRC: 0.004<br />mesh","Diabetes:  NA<br />CRC: 0.004<br />mesorectal","Diabetes: 0.1<br />CRC: 0.007<br />metabolic","Diabetes:  NA<br />CRC: 0.004<br />metabolizing","Diabetes:  NA<br />CRC: 0.004<br />metabolomics","Diabetes: 0.1<br />CRC:    NA<br />metalloproteinases","Diabetes:  NA<br />CRC: 0.045<br />metastases","Diabetes:  NA<br />CRC: 0.187<br />metastasis","Diabetes:  NA<br />CRC: 0.004<br />metastasizing","Diabetes:  NA<br />CRC: 0.086<br />metastatic","Diabetes:  NA<br />CRC: 0.004<br />metformin","Diabetes:  NA<br />CRC: 0.007<br />methyl","Diabetes:  NA<br />CRC: 0.004<br />methylacyl","Diabetes:  NA<br />CRC: 0.007<br />methylated","Diabetes:  NA<br />CRC: 0.022<br />methylation","Diabetes:  NA<br />CRC: 0.007<br />methylator","Diabetes:  NA<br />CRC: 0.004<br />methylcytosine","Diabetes:  NA<br />CRC: 0.004<br />methylguanine","Diabetes:  NA<br />CRC: 0.004<br />methylomic","Diabetes:  NA<br />CRC: 0.004<br />mfge8","Diabetes:  NA<br />CRC: 0.004<br />mfuzz","Diabetes:  NA<br />CRC: 0.007<br />mgb","Diabetes: 0.2<br />CRC:    NA<br />microalbuminuria","Diabetes: 0.1<br />CRC:    NA<br />microbubble","Diabetes:  NA<br />CRC: 0.007<br />microdissected","Diabetes:  NA<br />CRC: 0.007<br />microdissection","Diabetes: 0.1<br />CRC: 0.030<br />microenvironment","Diabetes:  NA<br />CRC: 0.011<br />microfluidic","Diabetes:  NA<br />CRC: 0.004<br />microgravity","Diabetes:  NA<br />CRC: 0.004<br />microribonucleic","Diabetes:  NA<br />CRC: 0.022<br />microsatellite","Diabetes:  NA<br />CRC: 0.004<br />microscopic","Diabetes:  NA<br />CRC: 0.004<br />microsulis","Diabetes:  NA<br />CRC: 0.004<br />microvesicles","Diabetes:  NA<br />CRC: 0.004<br />microvillus","Diabetes:  NA<br />CRC: 0.004<br />microwave","Diabetes:  NA<br />CRC: 0.004<br />migrate","Diabetes:  NA<br />CRC: 0.004<br />migrated","Diabetes:  NA<br />CRC: 0.011<br />migratory","Diabetes:  NA<br />CRC: 0.007<br />milk","Diabetes:  NA<br />CRC: 0.004<br />minimum","Diabetes:  NA<br />CRC: 0.004<br />mip3a","Diabetes:  NA<br />CRC: 0.004<br />mir126","Diabetes:  NA<br />CRC: 0.007<br />mir135a","Diabetes:  NA<br />CRC: 0.011<br />mir135b","Diabetes:  NA<br />CRC: 0.004<br />mir137","Diabetes:  NA<br />CRC: 0.007<br />mir141","Diabetes:  NA<br />CRC: 0.004<br />mir147b","Diabetes:  NA<br />CRC: 0.004<br />mir17hg","Diabetes:  NA<br />CRC: 0.004<br />mir181a","Diabetes: 0.1<br />CRC:    NA<br />mir192","Diabetes:  NA<br />CRC: 0.004<br />mir20a","Diabetes:  NA<br />CRC: 0.004<br />mir212","Diabetes:  NA<br />CRC: 0.007<br />mir214","Diabetes:  NA<br />CRC: 0.004<br />mir215p","Diabetes:  NA<br />CRC: 0.007<br />mir21a","Diabetes:  NA<br />CRC: 0.004<br />mir21hg","Diabetes:  NA<br />CRC: 0.004<br />mir224","Diabetes:  NA<br />CRC: 0.007<br />mir26b","Diabetes:  NA<br />CRC: 0.004<br />mir27b","Diabetes:  NA<br />CRC: 0.004<br />mir29a","Diabetes:  NA<br />CRC: 0.004<br />mir339","Diabetes:  NA<br />CRC: 0.004<br />mir361","Diabetes:  NA<br />CRC: 0.007<br />mir92a","Diabetes:  NA<br />CRC: 0.007<br />mircury","Diabetes:  NA<br />CRC: 0.004<br />mirdip","Diabetes:  NA<br />CRC: 0.004<br />mirneasy","Diabetes:  NA<br />CRC: 0.004<br />mirxes","Diabetes:  NA<br />CRC: 0.004<br />misclassification","Diabetes:  NA<br />CRC: 0.004<br />miscript","Diabetes:  NA<br />CRC: 0.004<br />misexpressed","Diabetes:  NA<br />CRC: 0.022<br />mismatch","Diabetes:  NA<br />CRC: 0.004<br />mitogen","Diabetes:  NA<br />CRC: 0.004<br />mitogenic","Diabetes:  NA<br />CRC: 0.004<br />mlh","Diabetes:  NA<br />CRC: 0.004<br />mlh1","Diabetes:  NA<br />CRC: 0.004<br />mmp","Diabetes:  NA<br />CRC: 0.004<br />mmp2","Diabetes:  NA<br />CRC: 0.007<br />mmr","Diabetes: 0.1<br />CRC: 0.004<br />mobility","Diabetes:  NA<br />CRC: 0.004<br />modeled","Diabetes:  NA<br />CRC: 0.004<br />modeling","Diabetes:  NA<br />CRC: 0.004<br />modulations","Diabetes:  NA<br />CRC: 0.004<br />molt","Diabetes:  NA<br />CRC: 0.004<br />monophosphate","Diabetes:  NA<br />CRC: 0.004<br />morphogenesis","Diabetes:  NA<br />CRC: 0.004<br />motif","Diabetes:  NA<br />CRC: 0.007<br />motility","Diabetes:  NA<br />CRC: 0.004<br />motivation","Diabetes: 0.1<br />CRC: 0.011<br />mouse","Diabetes:  NA<br />CRC: 0.004<br />mrecist","Diabetes:  NA<br />CRC: 0.004<br />msh","Diabetes:  NA<br />CRC: 0.004<br />msh2","Diabetes:  NA<br />CRC: 0.004<br />msh6","Diabetes:  NA<br />CRC: 0.015<br />msi","Diabetes:  NA<br />CRC: 0.004<br />mtor","Diabetes:  NA<br />CRC: 0.004<br />muc17","Diabetes:  NA<br />CRC: 0.004<br />muc3a","Diabetes:  NA<br />CRC: 0.004<br />muc3b","Diabetes:  NA<br />CRC: 0.004<br />mucin","Diabetes:  NA<br />CRC: 0.090<br />mucosa","Diabetes:  NA<br />CRC: 0.004<br />mucosae","Diabetes:  NA<br />CRC: 0.019<br />mucosal","Diabetes:  NA<br />CRC: 0.004<br />mucous","Diabetes:  NA<br />CRC: 0.007<br />mug","Diabetes:  NA<br />CRC: 0.004<br />mul","Diabetes:  NA<br />CRC: 0.004<br />multicolor","Diabetes:  NA<br />CRC: 0.004<br />multidrug","Diabetes:  NA<br />CRC: 0.004<br />multimarker","Diabetes:  NA<br />CRC: 0.007<br />multiplex","Diabetes:  NA<br />CRC: 0.007<br />multivariable","Diabetes:  NA<br />CRC: 0.004<br />mum","Diabetes:  NA<br />CRC: 0.004<br />muni","Diabetes:  NA<br />CRC: 0.007<br />murine","Diabetes:  NA<br />CRC: 0.007<br />muscle","Diabetes:  NA<br />CRC: 0.004<br />mutagen","Diabetes:  NA<br />CRC: 0.019<br />mutation","Diabetes:  NA<br />CRC: 0.007<br />mutational","Diabetes:  NA<br />CRC: 0.037<br />mutations","Diabetes:  NA<br />CRC: 0.004<br />mutator","Diabetes:  NA<br />CRC: 0.007<br />muts","Diabetes:  NA<br />CRC: 0.004<br />mutually","Diabetes:  NA<br />CRC: 0.004<br />mv","Diabetes:  NA<br />CRC: 0.004<br />mybl2","Diabetes:  NA<br />CRC: 0.007<br />myc","Diabetes:  NA<br />CRC: 0.004<br />myd88","Diabetes:  NA<br />CRC: 0.007<br />myofibroblast","Diabetes:  NA<br />CRC: 0.004<br />myofibroblasts","Diabetes:  NA<br />CRC: 0.004<br />mysql","Diabetes:  NA<br />CRC: 0.004<br />n0","Diabetes:  NA<br />CRC: 0.007<br />n1","Diabetes:  NA<br />CRC: 0.004<br />n2","Diabetes: 0.1<br />CRC:    NA<br />nadph","Diabetes:  NA<br />CRC: 0.004<br />nagel","Diabetes:  NA<br />CRC: 0.004<br />nanostring","Diabetes:  NA<br />CRC: 0.004<br />nas","Diabetes:  NA<br />CRC: 0.004<br />nat","Diabetes:  NA<br />CRC: 0.004<br />ncm460","Diabetes:  NA<br />CRC: 0.004<br />ncrt","Diabetes:  NA<br />CRC: 0.004<br />ncs","Diabetes:  NA<br />CRC: 0.007<br />neck","Diabetes: 0.1<br />CRC: 0.011<br />necrosis","Diabetes:  NA<br />CRC: 0.011<br />neoadjuvant","Diabetes:  NA<br />CRC: 0.026<br />neoplasia","Diabetes:  NA<br />CRC: 0.019<br />neoplasm","Diabetes:  NA<br />CRC: 0.015<br />neoplasms","Diabetes:  NA<br />CRC: 0.019<br />neoplastic","Diabetes: 0.2<br />CRC:    NA<br />nephropathy","Diabetes:  NA<br />CRC: 0.004<br />neural","Diabetes:  NA<br />CRC: 0.004<br />neurite","Diabetes:  NA<br />CRC: 0.004<br />neurologic","Diabetes:  NA<br />CRC: 0.004<br />neurotensin","Diabetes:  NA<br />CRC: 0.004<br />neurotrophin","Diabetes: 0.1<br />CRC: 0.026<br />nf","Diabetes:  NA<br />CRC: 0.004<br />nfkb1","Diabetes:  NA<br />CRC: 0.007<br />ngs","Diabetes:  NA<br />CRC: 0.004<br />nhs","Diabetes:  NA<br />CRC: 0.004<br />nighty","Diabetes:  NA<br />CRC: 0.004<br />nkd2","Diabetes:  NA<br />CRC: 0.004<br />nlr","Diabetes:  NA<br />CRC: 0.004<br />nomograms","Diabetes:  NA<br />CRC: 0.011<br />noncancerous","Diabetes:  NA<br />CRC: 0.004<br />nonconventional","Diabetes: 0.1<br />CRC:    NA<br />nonhealing","Diabetes:  NA<br />CRC: 0.004<br />nonintrusive","Diabetes:  NA<br />CRC: 0.004<br />nonpolyposis","Diabetes: 0.1<br />CRC:    NA<br />nonprogressors","Diabetes:  NA<br />CRC: 0.004<br />nonrecurrent","Diabetes:  NA<br />CRC: 0.004<br />nontransformed","Diabetes:  NA<br />CRC: 0.004<br />nontumor","Diabetes:  NA<br />CRC: 0.004<br />nontumorigenic","Diabetes:  NA<br />CRC: 0.004<br />nontumorous","Diabetes:  NA<br />CRC: 0.004<br />norgen","Diabetes:  NA<br />CRC: 0.004<br />normalisation","Diabetes:  NA<br />CRC: 0.004<br />normalise","Diabetes:  NA<br />CRC: 0.004<br />normalised","Diabetes:  NA<br />CRC: 0.004<br />normalisers","Diabetes: 0.1<br />CRC:    NA<br />normoalbuminuric","Diabetes:  NA<br />CRC: 0.004<br />normoxia","Diabetes:  NA<br />CRC: 0.004<br />nos2a","Diabetes:  NA<br />CRC: 0.004<br />notch","Diabetes:  NA<br />CRC: 0.004<br />npv","Diabetes:  NA<br />CRC: 0.004<br />nr2f2","Diabetes: 0.1<br />CRC:    NA<br />nr6a1","Diabetes:  NA<br />CRC: 0.004<br />ns398","Diabetes:  NA<br />CRC: 0.004<br />nt228","Diabetes:  NA<br />CRC: 0.004<br />ntr1","Diabetes:  NA<br />CRC: 0.004<br />nuclease","Diabetes:  NA<br />CRC: 0.007<br />nucleatum","Diabetes:  NA<br />CRC: 0.034<br />nucleic","Diabetes:  NA<br />CRC: 0.007<br />nucleolar","Diabetes:  NA<br />CRC: 0.004<br />nucleus","Diabetes:  NA<br />CRC: 0.026<br />nude","Diabetes:  NA<br />CRC: 0.007<br />nurses","Diabetes:  NA<br />CRC: 0.004<br />nutrition","Diabetes:  NA<br />CRC: 0.004<br />nutritional","Diabetes:  NA<br />CRC: 0.007<br />nx","Diabetes:  NA<br />CRC: 0.004<br />o3a","Diabetes:  NA<br />CRC: 0.004<br />o6","Diabetes: 0.1<br />CRC:    NA<br />obstruction","Diabetes:  NA<br />CRC: 0.007<br />occult","Diabetes:  NA<br />CRC: 0.004<br />ohp","Diabetes:  NA<br />CRC: 0.007<br />oligonucleotide","Diabetes:  NA<br />CRC: 0.004<br />ologic","Diabetes:  NA<br />CRC: 0.004<br />om","Diabetes:  NA<br />CRC: 0.004<br />omega","Diabetes:  NA<br />CRC: 0.019<br />onco","Diabetes:  NA<br />CRC: 0.034<br />oncogene","Diabetes:  NA<br />CRC: 0.026<br />oncogenes","Diabetes:  NA<br />CRC: 0.011<br />oncogenesis","Diabetes:  NA<br />CRC: 0.079<br />oncogenic","Diabetes:  NA<br />CRC: 0.007<br />oncological","Diabetes:  NA<br />CRC: 0.007<br />oncomir","Diabetes:  NA<br />CRC: 0.007<br />oncomirs","Diabetes:  NA<br />CRC: 0.004<br />ontology","Diabetes:  NA<br />CRC: 0.007<br />operated","Diabetes:  NA<br />CRC: 0.004<br />operatively","Diabetes:  NA<br />CRC: 0.004<br />optimism","Diabetes:  NA<br />CRC: 0.004<br />optimizing","Diabetes: 0.1<br />CRC:    NA<br />oral","Diabetes:  NA<br />CRC: 0.004<br />ordinal","Diabetes:  NA<br />CRC: 0.004<br />organoid","Diabetes:  NA<br />CRC: 0.004<br />organotypic","Diabetes:  NA<br />CRC: 0.004<br />orthotopic","Diabetes:  NA<br />CRC: 0.004<br />orthotopically","Diabetes:  NA<br />CRC: 0.004<br />oscillate","Diabetes:  NA<br />CRC: 0.004<br />outgrowths","Diabetes:  NA<br />CRC: 0.004<br />outstanding","Diabetes:  NA<br />CRC: 0.004<br />overabundance","Diabetes:  NA<br />CRC: 0.004<br />overcoming","Diabetes:  NA<br />CRC: 0.004<br />overhang","Diabetes:  NA<br />CRC: 0.004<br />overlapped","Diabetes:  NA<br />CRC: 0.004<br />overproduce","Diabetes:  NA<br />CRC: 0.004<br />ovo","Diabetes:  NA<br />CRC: 0.004<br />oxa","Diabetes:  NA<br />CRC: 0.022<br />oxaliplatin","Diabetes: 0.1<br />CRC:    NA<br />oxidase","Diabetes:  NA<br />CRC: 0.004<br />oxidize","Diabetes:  NA<br />CRC: 0.004<br />oxygenases","Diabetes: 0.1<br />CRC: 0.007<br />p16","Diabetes: 0.1<br />CRC: 0.011<br />p21","Diabetes:  NA<br />CRC: 0.004<br />p38","Diabetes:  NA<br />CRC: 0.015<br />p53","Diabetes:  NA<br />CRC: 0.004<br />p70","Diabetes:  NA<br />CRC: 0.004<br />padj","Diabetes:  NA<br />CRC: 0.004<br />pairwise","Diabetes:  NA<br />CRC: 0.007<br />palliative","Diabetes: 0.1<br />CRC: 0.007<br />pancreas","Diabetes:  NA<br />CRC: 0.004<br />pancreases","Diabetes: 0.1<br />CRC: 0.022<br />pancreatic","Diabetes:  NA<br />CRC: 0.004<br />panitumumab","Diabetes:  NA<br />CRC: 0.004<br />papilla","Diabetes:  NA<br />CRC: 0.004<br />papilloma","Diabetes:  NA<br />CRC: 0.004<br />papillomavirus","Diabetes:  NA<br />CRC: 0.004<br />paradoxical","Diabetes:  NA<br />CRC: 0.004<br />parallels","Diabetes:  NA<br />CRC: 0.004<br />paramagnetic","Diabetes:  NA<br />CRC: 0.011<br />parental","Diabetes:  NA<br />CRC: 0.004<br />participant","Diabetes: 0.1<br />CRC:    NA<br />particles","Diabetes:  NA<br />CRC: 0.004<br />partitioning","Diabetes:  NA<br />CRC: 0.004<br />passive","Diabetes: 0.2<br />CRC: 0.034<br />pathogenesis","Diabetes:  NA<br />CRC: 0.004<br />pathogenicity","Diabetes:  NA<br />CRC: 0.004<br />pathologist","Diabetes: 0.3<br />CRC: 0.584<br />patients","Diabetes:  NA<br />CRC: 0.004<br />pbs","Diabetes:  NA<br />CRC: 0.004<br />pca","Diabetes:  NA<br />CRC: 0.004<br />pcdna","Diabetes:  NA<br />CRC: 0.004<br />pcmv","Diabetes:  NA<br />CRC: 0.004<br />pcnsl","Diabetes:  NA<br />CRC: 0.004<br />pcnsls","Diabetes:  NA<br />CRC: 0.004<br />pcrc","Diabetes:  NA<br />CRC: 0.004<br />pcrcs","Diabetes:  NA<br />CRC: 0.004<br />pdac","Diabetes:  NA<br />CRC: 0.004<br />pdb","Diabetes:  NA<br />CRC: 0.075<br />pdcd4","Diabetes:  NA<br />CRC: 0.004<br />pde4b","Diabetes:  NA<br />CRC: 0.004<br />pdha1","Diabetes:  NA<br />CRC: 0.004<br />pea3","Diabetes:  NA<br />CRC: 0.004<br />pectin","Diabetes:  NA<br />CRC: 0.004<br />pegfp","Diabetes:  NA<br />CRC: 0.004<br />pegylated","Diabetes:  NA<br />CRC: 0.004<br />perfluorooctane","Diabetes: 0.1<br />CRC:    NA<br />perinatal","Diabetes:  NA<br />CRC: 0.007<br />periods","Diabetes:  NA<br />CRC: 0.004<br />permanente","Diabetes:  NA<br />CRC: 0.004<br />permitted","Diabetes:  NA<br />CRC: 0.004<br />perpetuation","Diabetes:  NA<br />CRC: 0.004<br />persons","Diabetes:  NA<br />CRC: 0.004<br />pfos","Diabetes:  NA<br />CRC: 0.011<br />pfs","Diabetes:  NA<br />CRC: 0.007<br />pge2","Diabetes:  NA<br />CRC: 0.004<br />pharmacological","Diabetes:  NA<br />CRC: 0.004<br />phenylimidazo","Diabetes:  NA<br />CRC: 0.004<br />phip","Diabetes:  NA<br />CRC: 0.052<br />phosphatase","Diabetes:  NA<br />CRC: 0.004<br />phosphate","Diabetes:  NA<br />CRC: 0.007<br />phosphatidylinositol","Diabetes:  NA<br />CRC: 0.004<br />phosphorothioate","Diabetes:  NA<br />CRC: 0.004<br />phosphorylase","Diabetes:  NA<br />CRC: 0.004<br />php","Diabetes:  NA<br />CRC: 0.022<br />pi3k","Diabetes:  NA<br />CRC: 0.011<br />pik3ca","Diabetes:  NA<br />CRC: 0.004<br />pinteraction","Diabetes:  NA<br />CRC: 0.004<br />pipeline","Diabetes:  NA<br />CRC: 0.004<br />pitx2","Diabetes:  NA<br />CRC: 0.004<br />pkh67","Diabetes: 0.3<br />CRC: 0.116<br />plasma","Diabetes:  NA<br />CRC: 0.011<br />plasmid","Diabetes: 0.1<br />CRC:    NA<br />plasminogen","Diabetes:  NA<br />CRC: 0.004<br />pleiotropic","Diabetes:  NA<br />CRC: 0.004<br />plr","Diabetes:  NA<br />CRC: 0.007<br />pluripotency","Diabetes:  NA<br />CRC: 0.004<br />pma","Diabetes:  NA<br />CRC: 0.004<br />pmi","Diabetes:  NA<br />CRC: 0.004<br />pmol","Diabetes:  NA<br />CRC: 0.004<br />pnd","Diabetes:  NA<br />CRC: 0.004<br />pnd1","Diabetes:  NA<br />CRC: 0.004<br />pocket","Diabetes:  NA<br />CRC: 0.004<br />polarization","Diabetes:  NA<br />CRC: 0.004<br />polarized","Diabetes:  NA<br />CRC: 0.004<br />polyadenylated","Diabetes:  NA<br />CRC: 0.004<br />polyadenylation","Diabetes:  NA<br />CRC: 0.004<br />polycomb","Diabetes:  NA<br />CRC: 0.007<br />polymorphisms","Diabetes:  NA<br />CRC: 0.004<br />polyol","Diabetes:  NA<br />CRC: 0.011<br />polyp","Diabetes:  NA<br />CRC: 0.007<br />polypoid","Diabetes:  NA<br />CRC: 0.015<br />polyposis","Diabetes:  NA<br />CRC: 0.037<br />polyps","Diabetes:  NA<br />CRC: 0.004<br />polysomal","Diabetes:  NA<br />CRC: 0.004<br />polyunsaturated","Diabetes:  NA<br />CRC: 0.004<br />pooling","Diabetes:  NA<br />CRC: 0.004<br />postdiagnostic","Diabetes:  NA<br />CRC: 0.004<br />postnatal","Diabetes:  NA<br />CRC: 0.004<br />postranscriptional","Diabetes:  NA<br />CRC: 0.004<br />powers","Diabetes:  NA<br />CRC: 0.004<br />ppar","Diabetes: 0.1<br />CRC:    NA<br />ppcs","Diabetes:  NA<br />CRC: 0.007<br />ppi","Diabetes:  NA<br />CRC: 0.004<br />ppp2ca","Diabetes:  NA<br />CRC: 0.004<br />ppv","Diabetes:  NA<br />CRC: 0.004<br />pr","Diabetes:  NA<br />CRC: 0.007<br />precancerous","Diabetes:  NA<br />CRC: 0.004<br />preceded","Diabetes:  NA<br />CRC: 0.004<br />prediagnostic","Diabetes:  NA<br />CRC: 0.004<br />predictors","Diabetes:  NA<br />CRC: 0.007<br />preferable","Diabetes:  NA<br />CRC: 0.004<br />preliminarily","Diabetes:  NA<br />CRC: 0.004<br />premalignant","Diabetes:  NA<br />CRC: 0.004<br />premetastatic","Diabetes:  NA<br />CRC: 0.004<br />preneoplastic","Diabetes:  NA<br />CRC: 0.034<br />preoperative","Diabetes:  NA<br />CRC: 0.004<br />preoperatively","Diabetes:  NA<br />CRC: 0.004<br />prepare","Diabetes:  NA<br />CRC: 0.004<br />presumed","Diabetes:  NA<br />CRC: 0.004<br />prev","Diabetes:  NA<br />CRC: 0.007<br />preventive","Diabetes:  NA<br />CRC: 0.004<br />prg1","Diabetes:  NA<br />CRC: 0.004<br />pri","Diabetes:  NA<br />CRC: 0.112<br />primary","Diabetes:  NA<br />CRC: 0.007<br />primers","Diabetes:  NA<br />CRC: 0.004<br />prl","Diabetes:  NA<br />CRC: 0.004<br />proapoptotic","Diabetes:  NA<br />CRC: 0.004<br />probability","Diabetes:  NA<br />CRC: 0.004<br />probiotic","Diabetes:  NA<br />CRC: 0.004<br />procarcinogenic","Diabetes:  NA<br />CRC: 0.004<br />proceeds","Diabetes:  NA<br />CRC: 0.007<br />professionals","Diabetes: 0.1<br />CRC:    NA<br />profibrotic","Diabetes: 0.2<br />CRC: 0.004<br />progenitor","Diabetes:  NA<br />CRC: 0.004<br />progesterone","Diabetes:  NA<br />CRC: 0.004<br />prognostics","Diabetes:  NA<br />CRC: 0.004<br />progressions","Diabetes: 0.2<br />CRC:    NA<br />progressors","Diabetes:  NA<br />CRC: 0.011<br />proinflammatory","Diabetes: 0.2<br />CRC: 0.142<br />proliferation","Diabetes:  NA<br />CRC: 0.004<br />prometastatic","Diabetes:  NA<br />CRC: 0.004<br />prominently","Diabetes:  NA<br />CRC: 0.037<br />promoter","Diabetes:  NA<br />CRC: 0.007<br />propidium","Diabetes:  NA<br />CRC: 0.004<br />propria","Diabetes:  NA<br />CRC: 0.004<br />proquest","Diabetes:  NA<br />CRC: 0.007<br />prostaglandin","Diabetes:  NA<br />CRC: 0.022<br />prostate","Diabetes:  NA<br />CRC: 0.004<br />prostatic","Diabetes:  NA<br />CRC: 0.004<br />proteasome","Diabetes: 0.1<br />CRC:    NA<br />protection","Diabetes: 0.1<br />CRC: 0.004<br />protective","Diabetes: 0.1<br />CRC:    NA<br />protects","Diabetes: 0.1<br />CRC:    NA<br />proteinuria","Diabetes: 0.1<br />CRC:    NA<br />proteinuric","Diabetes:  NA<br />CRC: 0.004<br />proteomic","Diabetes:  NA<br />CRC: 0.004<br />protrusions","Diabetes:  NA<br />CRC: 0.004<br />proximity","Diabetes:  NA<br />CRC: 0.004<br />psilencer4","Diabetes:  NA<br />CRC: 0.004<br />psoas","Diabetes:  NA<br />CRC: 0.004<br />pt","Diabetes:  NA<br />CRC: 0.004<br />pt1","Diabetes:  NA<br />CRC: 0.075<br />pten","Diabetes:  NA<br />CRC: 0.004<br />ptgs2","Diabetes:  NA<br />CRC: 0.004<br />pth1r","Diabetes:  NA<br />CRC: 0.004<br />ptk2b","Diabetes:  NA<br />CRC: 0.004<br />ptpn12","Diabetes:  NA<br />CRC: 0.004<br />ptprc","Diabetes:  NA<br />CRC: 0.007<br />ptrend","Diabetes:  NA<br />CRC: 0.004<br />pups","Diabetes:  NA<br />CRC: 0.007<br />pure","Diabetes:  NA<br />CRC: 0.004<br />purity","Diabetes:  NA<br />CRC: 0.004<br />pv","Diabetes:  NA<br />CRC: 0.004<br />pv.we","Diabetes:  NA<br />CRC: 0.004<br />pvac","Diabetes:  NA<br />CRC: 0.004<br />pyridine","Diabetes:  NA<br />CRC: 0.004<br />pyruvate","Diabetes:  NA<br />CRC: 0.004<br />qc","Diabetes:  NA<br />CRC: 0.004<br />qualitatively","Diabetes:  NA<br />CRC: 0.004<br />quantitated","Diabetes:  NA<br />CRC: 0.004<br />quantitation","Diabetes:  NA<br />CRC: 0.004<br />quantities","Diabetes:  NA<br />CRC: 0.015<br />quantity","Diabetes:  NA<br />CRC: 0.007<br />quartile","Diabetes:  NA<br />CRC: 0.004<br />quicker","Diabetes:  NA<br />CRC: 0.004<br />r6828","Diabetes:  NA<br />CRC: 0.004<br />rac","Diabetes:  NA<br />CRC: 0.004<br />racemase","Diabetes:  NA<br />CRC: 0.004<br />racial","Diabetes:  NA<br />CRC: 0.004<br />radiation","Diabetes:  NA<br />CRC: 0.011<br />radical","Diabetes:  NA<br />CRC: 0.004<br />radioactive","Diabetes:  NA<br />CRC: 0.004<br />radiochemical","Diabetes:  NA<br />CRC: 0.004<br />radiofrequency","Diabetes:  NA<br />CRC: 0.004<br />radiolabeled","Diabetes:  NA<br />CRC: 0.004<br />radiological","Diabetes:  NA<br />CRC: 0.004<br />radiology","Diabetes:  NA<br />CRC: 0.004<br />radiotherapy","Diabetes:  NA<br />CRC: 0.007<br />raf","Diabetes:  NA<br />CRC: 0.004<br />rage","Diabetes:  NA<br />CRC: 0.004<br />ranges","Diabetes:  NA<br />CRC: 0.004<br />rapamycin","Diabetes:  NA<br />CRC: 0.041<br />ras","Diabetes:  NA<br />CRC: 0.007<br />rasa1","Diabetes: 0.1<br />CRC: 0.007<br />rat","Diabetes: 0.2<br />CRC: 0.004<br />rats","Diabetes: 0.1<br />CRC:    NA<br />rbpj","Diabetes:  NA<br />CRC: 0.004<br />rbpms","Diabetes:  NA<br />CRC: 0.004<br />rc","Diabetes:  NA<br />CRC: 0.007<br />rcc","Diabetes:  NA<br />CRC: 0.004<br />rcs","Diabetes:  NA<br />CRC: 0.004<br />reagent","Diabetes:  NA<br />CRC: 0.015<br />reasons","Diabetes:  NA<br />CRC: 0.004<br />receipt","Diabetes:  NA<br />CRC: 0.004<br />recist","Diabetes:  NA<br />CRC: 0.007<br />recombinant","Diabetes:  NA<br />CRC: 0.004<br />recommend","Diabetes:  NA<br />CRC: 0.004<br />recommendation","Diabetes:  NA<br />CRC: 0.067<br />rectal","Diabetes:  NA<br />CRC: 0.011<br />rectum","Diabetes:  NA<br />CRC: 0.007<br />recurred","Diabetes:  NA<br />CRC: 0.007<br />recurrences","Diabetes:  NA<br />CRC: 0.007<br />reductase","Diabetes:  NA<br />CRC: 0.004<br />reexpression","Diabetes:  NA<br />CRC: 0.004<br />regenerating","Diabetes:  NA<br />CRC: 0.004<br />regimen","Diabetes:  NA<br />CRC: 0.007<br />regorafenib","Diabetes:  NA<br />CRC: 0.004<br />reinforced","Diabetes:  NA<br />CRC: 0.004<br />reinforcing","Diabetes:  NA<br />CRC: 0.004<br />relapsed","Diabetes: 0.3<br />CRC: 0.011<br />renal","Diabetes:  NA<br />CRC: 0.004<br />renewal","Diabetes:  NA<br />CRC: 0.004<br />repairment","Diabetes:  NA<br />CRC: 0.004<br />repeating","Diabetes:  NA<br />CRC: 0.007<br />repressor","Diabetes:  NA<br />CRC: 0.007<br />reproducibility","Diabetes:  NA<br />CRC: 0.004<br />reproducibly","Diabetes:  NA<br />CRC: 0.004<br />res","Diabetes:  NA<br />CRC: 0.004<br />resectable","Diabetes:  NA<br />CRC: 0.034<br />resected","Diabetes:  NA<br />CRC: 0.004<br />resections","Diabetes: 0.2<br />CRC: 0.045<br />resistance","Diabetes:  NA<br />CRC: 0.004<br />resisting","Diabetes:  NA<br />CRC: 0.004<br />resp","Diabetes:  NA<br />CRC: 0.004<br />restrictive","Diabetes:  NA<br />CRC: 0.004<br />resultant","Diabetes: 0.1<br />CRC:    NA<br />retinopathy","Diabetes:  NA<br />CRC: 0.007<br />rf","Diabetes:  NA<br />CRC: 0.004<br />rhob","Diabetes:  NA<br />CRC: 0.004<br />ribosomal","Diabetes:  NA<br />CRC: 0.004<br />rigorous","Diabetes:  NA<br />CRC: 0.004<br />rip1","Diabetes:  NA<br />CRC: 0.004<br />rita","Diabetes:  NA<br />CRC: 0.019<br />rko","Diabetes:  NA<br />CRC: 0.004<br />rnf138","Diabetes:  NA<br />CRC: 0.004<br />rnu2","Diabetes:  NA<br />CRC: 0.004<br />rnu44","Diabetes:  NA<br />CRC: 0.011<br />rnu48","Diabetes:  NA<br />CRC: 0.004<br />rnu6b","Diabetes:  NA<br />CRC: 0.007<br />ros","Diabetes: 0.1<br />CRC:    NA<br />rpbj","Diabetes:  NA<br />CRC: 0.007<br />rq","Diabetes:  NA<br />CRC: 0.007<br />rrna","Diabetes:  NA<br />CRC: 0.004<br />rs13505","Diabetes:  NA<br />CRC: 0.004<br />rs1559931","Diabetes:  NA<br />CRC: 0.004<br />rs4404254","Diabetes:  NA<br />CRC: 0.004<br />rs4675379","Diabetes:  NA<br />CRC: 0.004<br />rs6504593","Diabetes:  NA<br />CRC: 0.004<br />rs7628626","Diabetes:  NA<br />CRC: 0.004<br />rtks","Diabetes:  NA<br />CRC: 0.004<br />runx1","Diabetes:  NA<br />CRC: 0.004<br />runx3","Diabetes:  NA<br />CRC: 0.004<br />s100p","Diabetes:  NA<br />CRC: 0.004<br />s6","Diabetes:  NA<br />CRC: 0.004<br />s6k","Diabetes:  NA<br />CRC: 0.004<br />saliva","Diabetes:  NA<br />CRC: 0.004<br />sapiens","Diabetes:  NA<br />CRC: 0.004<br />sarcomeres","Diabetes:  NA<br />CRC: 0.004<br />sarcopenia","Diabetes:  NA<br />CRC: 0.004<br />satb1","Diabetes:  NA<br />CRC: 0.004<br />satisfying","Diabetes:  NA<br />CRC: 0.004<br />sav1","Diabetes:  NA<br />CRC: 0.004<br />sb501","Diabetes:  NA<br />CRC: 0.004<br />sb521","Diabetes:  NA<br />CRC: 0.004<br />scanty","Diabetes:  NA<br />CRC: 0.004<br />school","Diabetes:  NA<br />CRC: 0.011<br />scid","Diabetes:  NA<br />CRC: 0.004<br />sciencedirect","Diabetes:  NA<br />CRC: 0.004<br />screening.to","Diabetes:  NA<br />CRC: 0.004<br />scrutinize","Diabetes:  NA<br />CRC: 0.004<br />sdc2","Diabetes:  NA<br />CRC: 0.004<br />sec23a","Diabetes:  NA<br />CRC: 0.004<br />secretory","Diabetes:  NA<br />CRC: 0.011<br />section","Diabetes:  NA<br />CRC: 0.004<br />semiquantitative","Diabetes:  NA<br />CRC: 0.004<br />sen","Diabetes: 0.1<br />CRC:    NA<br />senescence","Diabetes:  NA<br />CRC: 0.004<br />sensitization","Diabetes:  NA<br />CRC: 0.011<br />sensitized","Diabetes:  NA<br />CRC: 0.004<br />sensors","Diabetes:  NA<br />CRC: 0.007<br />separately","Diabetes:  NA<br />CRC: 0.004<br />september","Diabetes:  NA<br />CRC: 0.004<br />ser552","Diabetes:  NA<br />CRC: 0.015<br />sera","Diabetes:  NA<br />CRC: 0.004<br />serially","Diabetes:  NA<br />CRC: 0.004<br />serine","Diabetes: 0.1<br />CRC: 0.004<br />serological","Diabetes:  NA<br />CRC: 0.004<br />serpin","Diabetes:  NA<br />CRC: 0.011<br />serrated","Diabetes: 0.2<br />CRC: 0.150<br />serum","Diabetes:  NA<br />CRC: 0.007<br />sessile","Diabetes:  NA<br />CRC: 0.004<br />sevs","Diabetes:  NA<br />CRC: 0.004<br />sfn","Diabetes:  NA<br />CRC: 0.004<br />sfpq","Diabetes:  NA<br />CRC: 0.007<br />sfrp1","Diabetes:  NA<br />CRC: 0.004<br />sfrp2","Diabetes:  NA<br />CRC: 0.004<br />sh","Diabetes:  NA<br />CRC: 0.004<br />shanghai","Diabetes:  NA<br />CRC: 0.004<br />shank2","Diabetes:  NA<br />CRC: 0.004<br />sharply","Diabetes:  NA<br />CRC: 0.004<br />shorten","Diabetes:  NA<br />CRC: 0.007<br />shrna","Diabetes:  NA<br />CRC: 0.004<br />shuttle","Diabetes:  NA<br />CRC: 0.004<br />simplicity","Diabetes:  NA<br />CRC: 0.004<br />simulation","Diabetes:  NA<br />CRC: 0.004<br />singapore","Diabetes: 0.1<br />CRC:    NA<br />sir","Diabetes:  NA<br />CRC: 0.004<br />sirasa1","Diabetes: 0.1<br />CRC:    NA<br />sirt1","Diabetes:  NA<br />CRC: 0.004<br />skeletal","Diabetes:  NA<br />CRC: 0.007<br />skin","Diabetes:  NA<br />CRC: 0.004<br />slc10a1","Diabetes:  NA<br />CRC: 0.004<br />slender","Diabetes:  NA<br />CRC: 0.007<br />slide","Diabetes:  NA<br />CRC: 0.011<br />slides","Diabetes:  NA<br />CRC: 0.004<br />slm","Diabetes:  NA<br />CRC: 0.004<br />slowed","Diabetes:  NA<br />CRC: 0.004<br />slowing","Diabetes:  NA<br />CRC: 0.004<br />smad4","Diabetes: 0.1<br />CRC: 0.004<br />smad7","Diabetes:  NA<br />CRC: 0.004<br />smc4","Diabetes:  NA<br />CRC: 0.004<br />smears","Diabetes:  NA<br />CRC: 0.004<br />smooth","Diabetes:  NA<br />CRC: 0.004<br />smrna","Diabetes:  NA<br />CRC: 0.004<br />sn","Diabetes:  NA<br />CRC: 0.004<br />sn38","Diabetes:  NA<br />CRC: 0.004<br />snail","Diabetes:  NA<br />CRC: 0.004<br />snail1","Diabetes:  NA<br />CRC: 0.004<br />snail2","Diabetes:  NA<br />CRC: 0.004<br />snp","Diabetes:  NA<br />CRC: 0.004<br />snps","Diabetes:  NA<br />CRC: 0.004<br />socs1","Diabetes: 0.1<br />CRC:    NA<br />sod","Diabetes:  NA<br />CRC: 0.011<br />sodium","Diabetes: 0.1<br />CRC: 0.004<br />soft","Diabetes:  NA<br />CRC: 0.004<br />sox","Diabetes: 0.1<br />CRC:    NA<br />sox6","Diabetes:  NA<br />CRC: 0.004<br />sox9","Diabetes:  NA<br />CRC: 0.004<br />sp1","Diabetes:  NA<br />CRC: 0.004<br />spared","Diabetes:  NA<br />CRC: 0.004<br />spe","Diabetes:  NA<br />CRC: 0.004<br />speckle","Diabetes:  NA<br />CRC: 0.004<br />sphere","Diabetes:  NA<br />CRC: 0.004<br />spherical","Diabetes:  NA<br />CRC: 0.004<br />spike","Diabetes:  NA<br />CRC: 0.004<br />spiked","Diabetes:  NA<br />CRC: 0.004<br />spliceosome","Diabetes:  NA<br />CRC: 0.011<br />sponge","Diabetes:  NA<br />CRC: 0.022<br />sporadic","Diabetes:  NA<br />CRC: 0.004<br />sporadically","Diabetes:  NA<br />CRC: 0.004<br />sprague","Diabetes:  NA<br />CRC: 0.004<br />spreading","Diabetes:  NA<br />CRC: 0.004<br />sprouts","Diabetes:  NA<br />CRC: 0.004<br />sprouty","Diabetes:  NA<br />CRC: 0.011<br />sprouty2","Diabetes:  NA<br />CRC: 0.011<br />spry2","Diabetes:  NA<br />CRC: 0.007<br />squamous","Diabetes:  NA<br />CRC: 0.004<br />square2","Diabetes:  NA<br />CRC: 0.007<br />src","Diabetes:  NA<br />CRC: 0.007<br />ssa","Diabetes:  NA<br />CRC: 0.004<br />ssas","Diabetes:  NA<br />CRC: 0.004<br />stabilize","Diabetes:  NA<br />CRC: 0.004<br />stacking","Diabetes:  NA<br />CRC: 0.004<br />stacks","Diabetes:  NA<br />CRC: 0.004<br />stain","Diabetes:  NA<br />CRC: 0.004<br />standardize","Diabetes:  NA<br />CRC: 0.004<br />standing","Diabetes:  NA<br />CRC: 0.004<br />starch","Diabetes:  NA<br />CRC: 0.004<br />starvation","Diabetes:  NA<br />CRC: 0.015<br />stat3","Diabetes: 0.1<br />CRC: 0.045<br />stem","Diabetes:  NA<br />CRC: 0.004<br />stemness","Diabetes: 0.1<br />CRC:    NA<br />steroidogenic","Diabetes:  NA<br />CRC: 0.004<br />stimuli","Diabetes:  NA<br />CRC: 0.007<br />stomach","Diabetes:  NA<br />CRC: 0.026<br />stool","Diabetes:  NA<br />CRC: 0.004<br />stools","Diabetes:  NA<br />CRC: 0.004<br />strata","Diabetes: 0.2<br />CRC:    NA<br />streptozotocin","Diabetes:  NA<br />CRC: 0.019<br />stroma","Diabetes:  NA<br />CRC: 0.004<br />stroma.by","Diabetes:  NA<br />CRC: 0.045<br />stromal","Diabetes:  NA<br />CRC: 0.004<br />studio","Diabetes:  NA<br />CRC: 0.004<br />study's","Diabetes: 0.2<br />CRC:    NA<br />stz","Diabetes:  NA<br />CRC: 0.007<br />subcutaneous","Diabetes:  NA<br />CRC: 0.004<br />subjective","Diabetes:  NA<br />CRC: 0.004<br />submitted","Diabetes:  NA<br />CRC: 0.004<br />suboptimal","Diabetes:  NA<br />CRC: 0.004<br />subunit","Diabetes:  NA<br />CRC: 0.004<br />suffer","Diabetes:  NA<br />CRC: 0.007<br />suited","Diabetes:  NA<br />CRC: 0.004<br />suites","Diabetes:  NA<br />CRC: 0.011<br />sulfate","Diabetes:  NA<br />CRC: 0.004<br />sulfonate","Diabetes:  NA<br />CRC: 0.004<br />sulforaphane","Diabetes: 0.1<br />CRC:    NA<br />superoxide","Diabetes: 0.1<br />CRC: 0.004<br />supplement","Diabetes: 0.1<br />CRC: 0.004<br />supplementation","Diabetes:  NA<br />CRC: 0.004<br />supposed","Diabetes:  NA<br />CRC: 0.004<br />surgeon","Diabetes:  NA<br />CRC: 0.052<br />surgery","Diabetes:  NA<br />CRC: 0.034<br />surgical","Diabetes:  NA<br />CRC: 0.015<br />surgically","Diabetes:  NA<br />CRC: 0.004<br />surrounded","Diabetes:  NA<br />CRC: 0.206<br />survival","Diabetes:  NA<br />CRC: 0.007<br />survived","Diabetes:  NA<br />CRC: 0.007<br />survivin","Diabetes:  NA<br />CRC: 0.064<br />sw480","Diabetes:  NA<br />CRC: 0.011<br />sw620","Diabetes:  NA<br />CRC: 0.004<br />sw837","Diabetes:  NA<br />CRC: 0.004<br />switches","Diabetes:  NA<br />CRC: 0.004<br />syber","Diabetes:  NA<br />CRC: 0.004<br />sybergreen","Diabetes:  NA<br />CRC: 0.015<br />sybr","Diabetes:  NA<br />CRC: 0.004<br />symmetric","Diabetes:  NA<br />CRC: 0.007<br />synchronous","Diabetes:  NA<br />CRC: 0.007<br />synthase","Diabetes: 0.1<br />CRC:    NA<br />t1d","Diabetes: 0.3<br />CRC:    NA<br />t2dm","Diabetes:  NA<br />CRC: 0.004<br />t4","Diabetes:  NA<br />CRC: 0.004<br />t7","Diabetes:  NA<br />CRC: 0.004<br />ta","Diabetes:  NA<br />CRC: 0.004<br />tace","Diabetes: 0.1<br />CRC:    NA<br />taipei","Diabetes: 0.1<br />CRC: 0.004<br />taiwan","Diabetes:  NA<br />CRC: 0.004<br />targetlink","Diabetes:  NA<br />CRC: 0.004<br />taylor","Diabetes:  NA<br />CRC: 0.004<br />tb","Diabetes:  NA<br />CRC: 0.004<br />tbcs","Diabetes:  NA<br />CRC: 0.004<br />tbr","Diabetes: 0.1<br />CRC: 0.004<br />tc","Diabetes:  NA<br />CRC: 0.007<br />tcf","Diabetes:  NA<br />CRC: 0.011<br />tcf4","Diabetes:  NA<br />CRC: 0.004<br />tcf7l2","Diabetes:  NA<br />CRC: 0.011<br />tcga","Diabetes:  NA<br />CRC: 0.004<br />teer","Diabetes: 0.1<br />CRC: 0.007<br />telomerase","Diabetes:  NA<br />CRC: 0.004<br />telotaggg","Diabetes:  NA<br />CRC: 0.007<br />temperature","Diabetes:  NA<br />CRC: 0.037<br />tensin","Diabetes:  NA<br />CRC: 0.004<br />tet","Diabetes:  NA<br />CRC: 0.004<br />tet1","Diabetes:  NA<br />CRC: 0.004<br />tetraspanin","Diabetes:  NA<br />CRC: 0.004<br />tetraspanin3","Diabetes:  NA<br />CRC: 0.007<br />tetrazolium","Diabetes:  NA<br />CRC: 0.004<br />tfs","Diabetes: 0.1<br />CRC:    NA<br />tg","Diabetes: 0.3<br />CRC: 0.026<br />tgf","Diabetes:  NA<br />CRC: 0.007<br />tgfbeta","Diabetes:  NA<br />CRC: 0.004<br />tgfbetar2","Diabetes:  NA<br />CRC: 0.004<br />tgfbr2","Diabetes:  NA<br />CRC: 0.004<br />thbs1","Diabetes:  NA<br />CRC: 0.011<br />therapeutics","Diabetes:  NA<br />CRC: 0.004<br />thirteen","Diabetes:  NA<br />CRC: 0.004<br />threonine","Diabetes:  NA<br />CRC: 0.004<br />thrombosis","Diabetes:  NA<br />CRC: 0.004<br />thrombus","Diabetes:  NA<br />CRC: 0.004<br />throws","Diabetes:  NA<br />CRC: 0.004<br />thymidine","Diabetes:  NA<br />CRC: 0.004<br />thymidylate","Diabetes:  NA<br />CRC: 0.004<br />thyroid","Diabetes:  NA<br />CRC: 0.007<br />tiam1","Diabetes: 0.1<br />CRC:    NA<br />timp3","Diabetes:  NA<br />CRC: 0.004<br />tlr2","Diabetes:  NA<br />CRC: 0.004<br />tlr4","Diabetes:  NA<br />CRC: 0.004<br />tlr7","Diabetes:  NA<br />CRC: 0.004<br />tm","Diabetes:  NA<br />CRC: 0.004<br />tma","Diabetes:  NA<br />CRC: 0.004<br />tme","Diabetes: 0.2<br />CRC: 0.011<br />tnf","Diabetes:  NA<br />CRC: 0.004<br />tnfalpha","Diabetes:  NA<br />CRC: 0.041<br />tnm","Diabetes:  NA<br />CRC: 0.004<br />tnm1","Diabetes: 0.1<br />CRC:    NA<br />tocotrienol","Diabetes: 0.1<br />CRC:    NA<br />tolerance","Diabetes:  NA<br />CRC: 0.007<br />toll","Diabetes:  NA<br />CRC: 0.004<br />topoisomerase","Diabetes:  NA<br />CRC: 0.007<br />toxic","Diabetes:  NA<br />CRC: 0.007<br />toxicant","Diabetes:  NA<br />CRC: 0.007<br />toxicities","Diabetes:  NA<br />CRC: 0.004<br />toxicol","Diabetes:  NA<br />CRC: 0.011<br />tp53","Diabetes:  NA<br />CRC: 0.004<br />tp53inp1","Diabetes:  NA<br />CRC: 0.004<br />tract","Diabetes:  NA<br />CRC: 0.041<br />training","Diabetes:  NA<br />CRC: 0.004<br />transactivation","Diabetes:  NA<br />CRC: 0.004<br />transarterial","Diabetes:  NA<br />CRC: 0.045<br />transcriptase","Diabetes:  NA<br />CRC: 0.007<br />transdifferentiation","Diabetes:  NA<br />CRC: 0.004<br />transepithelial","Diabetes:  NA<br />CRC: 0.011<br />transferred","Diabetes:  NA<br />CRC: 0.011<br />transformed","Diabetes: 0.1<br />CRC: 0.011<br />transforming","Diabetes: 0.1<br />CRC:    NA<br />transporting","Diabetes:  NA<br />CRC: 0.004<br />trappc3","Diabetes:  NA<br />CRC: 0.007<br />trg","Diabetes:  NA<br />CRC: 0.004<br />trg0","Diabetes:  NA<br />CRC: 0.004<br />trg1","Diabetes:  NA<br />CRC: 0.004<br />trg4","Diabetes: 0.1<br />CRC:    NA<br />triacylglycerol","Diabetes:  NA<br />CRC: 0.004<br />tricarboxylic","Diabetes:  NA<br />CRC: 0.004<br />trimethylation","Diabetes:  NA<br />CRC: 0.004<br />trizol","Diabetes:  NA<br />CRC: 0.004<br />trna","Diabetes:  NA<br />CRC: 0.011<br />true","Diabetes:  NA<br />CRC: 0.004<br />truseq","Diabetes:  NA<br />CRC: 0.004<br />tsss","Diabetes: 0.2<br />CRC: 0.007<br />tubular","Diabetes:  NA<br />CRC: 0.004<br />tubulovillous","Diabetes: 0.1<br />CRC: 0.483<br />tumor","Diabetes:  NA<br />CRC: 0.007<br />tumoral","Diabetes:  NA<br />CRC: 0.052<br />tumorigenesis","Diabetes:  NA<br />CRC: 0.019<br />tumorous","Diabetes:  NA<br />CRC: 0.187<br />tumors","Diabetes:  NA<br />CRC: 0.097<br />tumour","Diabetes:  NA<br />CRC: 0.052<br />tumours","Diabetes:  NA<br />CRC: 0.004<br />turkish","Diabetes:  NA<br />CRC: 0.004<br />twist","Diabetes:  NA<br />CRC: 0.004<br />tyramide","Diabetes: 0.1<br />CRC: 0.011<br />tyrosine","Diabetes:  NA<br />CRC: 0.004<br />u2","Diabetes:  NA<br />CRC: 0.004<br />u251","Diabetes:  NA<br />CRC: 0.004<br />u4a","Diabetes:  NA<br />CRC: 0.007<br />u87","Diabetes:  NA<br />CRC: 0.011<br />uc","Diabetes:  NA<br />CRC: 0.004<br />ucat","Diabetes:  NA<br />CRC: 0.019<br />ulcerative","Diabetes:  NA<br />CRC: 0.004<br />umol","Diabetes:  NA<br />CRC: 0.004<br />unambiguous","Diabetes:  NA<br />CRC: 0.004<br />unambiguously","Diabetes:  NA<br />CRC: 0.004<br />unanimous","Diabetes:  NA<br />CRC: 0.004<br />underexpression","Diabetes:  NA<br />CRC: 0.007<br />underscores","Diabetes:  NA<br />CRC: 0.004<br />undertook","Diabetes:  NA<br />CRC: 0.004<br />unfavorable","Diabetes:  NA<br />CRC: 0.004<br />uninfected","Diabetes:  NA<br />CRC: 0.004<br />unisp2","Diabetes:  NA<br />CRC: 0.004<br />unisp3","Diabetes:  NA<br />CRC: 0.004<br />unisp6","Diabetes:  NA<br />CRC: 0.007<br />unlabelled","Diabetes:  NA<br />CRC: 0.007<br />unnecessary","Diabetes:  NA<br />CRC: 0.004<br />unpleasant","Diabetes:  NA<br />CRC: 0.007<br />unresectable","Diabetes:  NA<br />CRC: 0.004<br />unsatisfactorily","Diabetes:  NA<br />CRC: 0.004<br />unsupervised","Diabetes: 0.1<br />CRC:    NA<br />ureteral","Diabetes:  NA<br />CRC: 0.007<br />usa","Diabetes:  NA<br />CRC: 0.004<br />usefully","Diabetes:  NA<br />CRC: 0.004<br />ut3","Diabetes:  NA<br />CRC: 0.004<br />utah","Diabetes: 0.1<br />CRC:    NA<br />uuo","Diabetes:  NA<br />CRC: 0.004<br />v2.0","Diabetes:  NA<br />CRC: 0.004<br />v3.0","Diabetes:  NA<br />CRC: 0.004<br />v3.1.1","Diabetes:  NA<br />CRC: 0.004<br />vaco4s","Diabetes:  NA<br />CRC: 0.004<br />validating","Diabetes:  NA<br />CRC: 0.004<br />valine","Diabetes:  NA<br />CRC: 0.004<br />variances","Diabetes:  NA<br />CRC: 0.004<br />vasation","Diabetes: 0.1<br />CRC: 0.011<br />vascular","Diabetes:  NA<br />CRC: 0.004<br />vater's","Diabetes:  NA<br />CRC: 0.004<br />vegetables","Diabetes:  NA<br />CRC: 0.004<br />vegf","Diabetes:  NA<br />CRC: 0.004<br />vegfa","Diabetes:  NA<br />CRC: 0.007<br />vein","Diabetes:  NA<br />CRC: 0.019<br />venous","Diabetes:  NA<br />CRC: 0.007<br />vesicles","Diabetes: 0.1<br />CRC:    NA<br />vessel","Diabetes:  NA<br />CRC: 0.004<br />video","Diabetes:  NA<br />CRC: 0.004<br />villin","Diabetes:  NA<br />CRC: 0.007<br />vimentin","Diabetes:  NA<br />CRC: 0.004<br />virus","Diabetes:  NA<br />CRC: 0.004<br />vogelstein","Diabetes:  NA<br />CRC: 0.004<br />volumetry","Diabetes:  NA<br />CRC: 0.004<br />vp","Diabetes:  NA<br />CRC: 0.004<br />vsnl1","Diabetes:  NA<br />CRC: 0.004<br />waltham","Diabetes:  NA<br />CRC: 0.004<br />weighting","Diabetes:  NA<br />CRC: 0.007<br />west","Diabetes:  NA<br />CRC: 0.007<br />white","Diabetes:  NA<br />CRC: 0.004<br />whites","Diabetes:  NA<br />CRC: 0.007<br />widr","Diabetes:  NA<br />CRC: 0.007<br />wilcoxon","Diabetes: 0.1<br />CRC: 0.019<br />wnt","Diabetes:  NA<br />CRC: 0.011<br />women","Diabetes:  NA<br />CRC: 0.004<br />worthwhile","Diabetes: 0.1<br />CRC:    NA<br />wound","Diabetes: 0.1<br />CRC:    NA<br />wounds","Diabetes:  NA<br />CRC: 0.007<br />wst","Diabetes:  NA<br />CRC: 0.004<br />ww45","Diabetes:  NA<br />CRC: 0.004<br />wwox","Diabetes:  NA<br />CRC: 0.004<br />xcelligence","Diabetes:  NA<br />CRC: 0.004<br />xenobiotic","Diabetes:  NA<br />CRC: 0.041<br />xenograft","Diabetes:  NA<br />CRC: 0.007<br />xenografted","Diabetes:  NA<br />CRC: 0.004<br />xenografting","Diabetes:  NA<br />CRC: 0.030<br />xenografts","Diabetes:  NA<br />CRC: 0.004<br />xinhua","Diabetes:  NA<br />CRC: 0.004<br />ybx1","Diabetes:  NA<br />CRC: 0.019<br />yl","Diabetes:  NA<br />CRC: 0.004<br />z30","Diabetes:  NA<br />CRC: 0.004<br />zeste","Diabetes:  NA<br />CRC: 0.004<br />zone","Diabetes:  NA<br />CRC: 0.004<br />zymbal's","Diabetes:  NA<br />CRC: 0.004<br />zymo","Diabetes: 0.2<br />CRC:    NA<br />α","Diabetes: 0.3<br />CRC:    NA<br />β","Diabetes: 0.2<br />CRC:    NA<br />β1","Diabetes: 0.1<br />CRC:    NA<br />δt3","Diabetes: 0.1<br />CRC:    NA<br />κb"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,0,0,1)","opacity":0.1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,0,0,1)"}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":45.3588829941669,"r":7.30593607305936,"b":41.7789743183678,"l":60.6392694063927},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"title":{"text":"Comparison of miRNA-term association for miR-21","font":{"color":"rgba(0,0,0,1)","family":"","size":17.5342465753425},"x":0,"xref":"paper"},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1.09449353411648,-0.108721221054746],"tickmode":"array","ticktext":["10%","30%","50%"],"tickvals":[-1,-0.522878745280338,-0.301029995663981],"categoryorder":"array","categoryarray":["10%","30%","50%"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"Diabetes","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-2.56682383973338,0.050032712996108],"tickmode":"array","ticktext":["1.0%","10.0%","100.0%"],"tickvals":[-2,-1,0],"categoryorder":"array","categoryarray":["1.0%","10.0%","100.0%"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":{"text":"CRC","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"121631730afc4":{"intercept":{},"slope":{},"type":"scatter"},"121637d6378c0":{"x":{},"y":{},"text":{}}},"cur_data":"121631730afc4","visdat":{"121631730afc4":["function (y) ","x"],"121637d6378c0":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="identify-unique-term-associations" class="section level3">
<h3 class="hasAnchor">
<a href="#identify-unique-term-associations" class="anchor"></a>Identify unique term associations</h3>
<p>The most frequent terms a miRNA is associated with in one, but not another topic can be displayed with <code><a href="../reference/compare_mir_terms_unique.html">compare_mir_terms_unique()</a></code>.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identify top 10 unique terms of miR-21 per topic</span>
<span class="fu"><a href="../reference/compare_mir_terms_unique.html">compare_mir_terms_unique</a></span><span class="op">(</span><span class="va">df_combined</span>,
                         mir <span class="op">=</span> <span class="st">"miR-21"</span>, top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-37-1.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div id="compare-target-interactions" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-target-interactions" class="anchor"></a>Compare target interactions</h3>
<p>As described previously, targets can be added from an excel file with <code><a href="../reference/join_targets.html">join_targets()</a></code>. When a data frame contains miRNA-target interactions in multiple subjects, <code><a href="../reference/plot_target_mir_scatter.html">plot_target_mir_scatter()</a></code> colours the miRNA-target interaction by subject, thereby allowing easy comparison of miRNA-target interactions across fields. See the Case study below for examples.</p>
</div>
</div>
</div>
<div id="case-study" class="section level1">
<h1 class="hasAnchor">
<a href="#case-study" class="anchor"></a>Case study</h1>
<p>In the last section, we are going to apply miRetrieve in a small case study.</p>
<div id="introduction-1" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-1" class="anchor"></a>Introduction</h2>
<p>In this fictive case study, our lab detected miR-21 to be aberrantly expressed in colorectal cancer (CRC). Using miRetrieve, we characterize the role of miR-21 in CRC and compare it to its role in pancreatic cancer.</p>
</div>
<div id="data-preparation-1" class="section level2">
<h2 class="hasAnchor">
<a href="#data-preparation-1" class="anchor"></a>Data preparation</h2>
<p>To investigate the role of miR-21 in CRC, we load all PubMed abstracts matching the keywords <em>colorectal cancer mirna</em> with <code><a href="../reference/read_pubmed.html">read_pubmed()</a></code>. Then, we keep only abstracts of original research articles, using <code><a href="../reference/subset_research.html">subset_research()</a></code>, and subsequently extract their miRNA names with <code><a href="../reference/extract_mir_df.html">extract_mir_df()</a></code>.</p>
<p>For illustration purposes (and to keep the size of sample files small), we are again going to use a toy dataset on miRNAs in CRC that has a heavy focus on miR-21, but also contains some randomly chosen abstracts.</p>
<p>During our analysis, we use the <code><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></code> operator from the <strong>magrittr</strong> package. <code><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></code> passes the result of one function straight into the following function, making our code easier to write, read, and maintain.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load miRetrieve</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">miRetrieve</span><span class="op">)</span>
<span class="co"># Load magrittr</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org">magrittr</a></span><span class="op">)</span>

<span class="co"># Path to PubMed-file</span>
<span class="va">crc_pubmed</span> <span class="op">&lt;-</span> <span class="st">"CRC_PubMed.txt"</span>

<span class="co"># Load PubMed-file</span>
<span class="va">df_crc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="va">crc_pubmed</span>,
                      topic <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Keep abstracts of original research articles</span>
  <span class="fu"><a href="../reference/subset_research.html">subset_research</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Extract miRNA names</span>
  <span class="fu"><a href="../reference/extract_mir_df.html">extract_mir_df</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p>After loading and filtering the abstracts, we have a data frame with 508 rows and 8 columns, named <em>PMID, Year, Title, Abstract, Language, Type, Topic, miRNA</em>. As each miRNA name occupies one row in the <code>miRNA</code> column, and each abstract can mention more than one miRNA, there are more rows in the data frame than abstracts under investigation.</p>
</div>
<div id="indicate-mir-21-in-crc-and-save-file" class="section level2">
<h2 class="hasAnchor">
<a href="#indicate-mir-21-in-crc-and-save-file" class="anchor"></a>Indicate miR-21 in CRC and save file</h2>
<p>We know that we are going to need information on miR-21 in the future. We therefore label all abstracts mentioning miR-21 with <code><a href="../reference/indicate_mir.html">indicate_mir()</a></code>, and save the table as an .xlsx-file. Saving the table locally will allow us to read and compare PubMed abstracts mentioning miR-21 comfortably.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Label all abstracts mentioning miR-21 with "Yes"</span>
<span class="va">df_mir21</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/indicate_mir.html">indicate_mir</a></span><span class="op">(</span><span class="va">df_crc</span>,
                         indicate.mir <span class="op">=</span> <span class="st">"miR-21"</span><span class="op">)</span>

<span class="co"># Save as an .xlsx file</span>
<span class="fu"><a href="../reference/save_excel.html">save_excel</a></span><span class="op">(</span><span class="va">df_mir21</span>,
           excel_file <span class="op">=</span> <span class="st">"miR21_crc.xlsx"</span><span class="op">)</span></code></pre></div>
</div>
<div id="count-mirnas-in-crc" class="section level2">
<h2 class="hasAnchor">
<a href="#count-mirnas-in-crc" class="anchor"></a>Count miRNAs in CRC</h2>
<p>To begin our analysis, we count the top miRNAs in CRC using <code><a href="../reference/plot_mir_count.html">plot_mir_count()</a></code>. If miR-21 should not be among the top miRNAs, we plan on counting all miRNAs with <code><a href="../reference/count_mir.html">count_mir()</a></code>, filter the resulting data frame for miR-21 with <code><a href="../reference/subset_df.html">subset_df()</a></code>, and distinctively determine the count of miR-21.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot count of top miRNAs in CRC</span>
<span class="fu"><a href="../reference/plot_mir_count.html">plot_mir_count</a></span><span class="op">(</span><span class="va">df_crc</span>,
               title <span class="op">=</span> <span class="st">"Most frequently mentioned miRNAs in CRC"</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-40-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>In our toy dataset of CRC, miR-21 seems to be the most frequently mentioned miRNA, being reported in more than 250 abstracts. Next to miR-21, miR-145 and miR-34<br>
are also mentioned frequently in CRC.</p>
</div>
<div id="plot-mir-21-associated-terms-in-crc" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-mir-21-associated-terms-in-crc" class="anchor"></a>Plot miR-21 associated terms in CRC</h2>
<p>As miR-21 is likely well investigated in CRC, we identify which terms miR-21 is associated with using <code><a href="../reference/plot_mir_terms.html">plot_mir_terms()</a></code>.</p>
<p>To analyze the terms miR-21 is associated with, we choose to tokenize the text into single words and into 2-grams.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot top single terms associated with miR-21 in CRC</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_crc</span>,
               <span class="st">"miR-21"</span>,
               top <span class="op">=</span> <span class="fl">30</span>,
               title <span class="op">=</span> <span class="st">"Top terms for miR-21 in CRC"</span><span class="op">)</span>

<span class="co"># Plot top 2-grams associated with miR-21 in CRC</span>
<span class="fu"><a href="../reference/plot_mir_terms.html">plot_mir_terms</a></span><span class="op">(</span><span class="va">df_crc</span>,
               <span class="st">"miR-21"</span>,
               token <span class="op">=</span> <span class="st">"ngrams"</span>,
               n <span class="op">=</span> <span class="fl">2</span>,
               title <span class="op">=</span> <span class="st">"Top 2-grams for miR-21 in CRC"</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-41-1.png" width="50%"><img src="miRetrieve_files/figure-html/unnamed-chunk-41-2.png" width="50%"></p>
<p>The top single terms for miR-21 suggest that miR-21 is associated with <em>survival</em>, even though <em>survival</em> is most likely a word common in cancer literature. Furthermore, it is implied that miR-21 targets <em>PTEN</em> and <em>PDCD4</em>. As miR-21 is also often associated with <em>serum</em>, <em>plasma</em>, and <em>biomarker</em>, miR-21 might be a biomarker.</p>
<p>The 2-gram tokenization implies that miR-21 is frequently mentioned with miR-145, miR-20a, miR-92a, and miR-31, which hints at possible miR-miR interactions and co-regulations.</p>
</div>
<div id="mirnas-as-biomarkers-in-crc" class="section level2">
<h2 class="hasAnchor">
<a href="#mirnas-as-biomarkers-in-crc" class="anchor"></a>miRNAs as biomarkers in CRC</h2>
<p>As miR-21 might be a biomarker in CRC, we determine the most frequently mentioned miRNAs in biomarker abstracts.</p>
<p>Here, we first subset for biomarker abstracts with <code><a href="../reference/calculate_score_biomarker.html">calculate_score_biomarker()</a></code> and setting <code>discard = TRUE</code>.<br>
As <code><a href="../reference/calculate_score_biomarker.html">calculate_score_biomarker()</a></code> requires a threshold to distinguish abstracts with and without biomarker, we try to determine a reliable threshold using <code><a href="../reference/plot_score_biomarker.html">plot_score_biomarker()</a></code>.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot score distribution for biomarker in CRC</span>
<span class="fu"><a href="../reference/plot_score_biomarker.html">plot_score_biomarker</a></span><span class="op">(</span><span class="va">df_crc</span>,
                     title <span class="op">=</span> <span class="st">"Biomarker score distribution in CRC"</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-42-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>While the majority of abstracts do not seem to report miRNAs as biomarkers in CRC, using a threshold above 5 in <code><a href="../reference/calculate_score_biomarker.html">calculate_score_biomarker()</a></code> seems reasonable. After identifying abstracts describing biomarkers, we count the top miRNAs with <code><a href="../reference/plot_mir_count.html">plot_mir_count()</a></code>.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identify abstracts reporting miRNAs as biomarker in CRC</span>
<span class="va">crc_biomarker</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_score_biomarker.html">calculate_score_biomarker</a></span><span class="op">(</span><span class="va">df_crc</span>,
                                           threshold <span class="op">=</span> <span class="fl">5</span>,
                                           discard <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Plot top potential biomarker miRNAs in CRC</span>
<span class="fu"><a href="../reference/plot_mir_count.html">plot_mir_count</a></span><span class="op">(</span><span class="va">crc_biomarker</span>,
               title <span class="op">=</span> <span class="st">"Likely biomarker in CRC"</span>,
               top <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-43-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>miR-21 is mentioned in more than 50 abstracts potentially reporting biomarkers, and thus very likely to be a biomarker in CRC.</p>
</div>
<div id="mirnas-as-biomarkers-in-pancreatic-cancer" class="section level2">
<h2 class="hasAnchor">
<a href="#mirnas-as-biomarkers-in-pancreatic-cancer" class="anchor"></a>miRNAs as biomarkers in pancreatic cancer</h2>
<p>To determine if miR-21 is a specific biomarker for CRC, we compare it to possible miRNA biomarkers in another cancer entity, namely pancreatic cancer.</p>
<p>First, we load abstracts matching the keywords <em>pancreatic cancer mirna</em>, keep only abstracts of original research articles, and extract their miRNA names.</p>
<p>This dataset is again very focused on miR-21 to keep the file size small.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Path to PubMed-file</span>
<span class="va">panc_pubmed</span> <span class="op">&lt;-</span> <span class="st">"Pancreas_PubMed.txt"</span>

<span class="co"># Load PubMed-file</span>
<span class="va">df_panc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pubmed.html">read_pubmed</a></span><span class="op">(</span><span class="va">panc_pubmed</span>,
                       topic <span class="op">=</span> <span class="st">"Pancreas"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Keep original research articles</span>
  <span class="fu"><a href="../reference/subset_research.html">subset_research</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Extract miRNA names</span>
  <span class="fu"><a href="../reference/extract_mir_df.html">extract_mir_df</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Next, we identify the top possible biomarker miRNAs with <code><a href="../reference/plot_score_biomarker.html">plot_score_biomarker()</a></code>, <code><a href="../reference/calculate_score_biomarker.html">calculate_score_biomarker()</a></code>, and <code>plot_mir_count</code>. If miR-21 is not among the top biomarker miRNAs in pancreatic cancer, we subset all biomarker abstracts for miR-21 with <code><a href="../reference/subset_df.html">subset_df()</a></code> and count miR-21 selectively with <code><a href="../reference/count_mir.html">count_mir()</a></code>.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot score distribution for biomarker in pancreatic cancer</span>
<span class="fu"><a href="../reference/plot_score_biomarker.html">plot_score_biomarker</a></span><span class="op">(</span><span class="va">df_panc</span>,
                     title <span class="op">=</span> <span class="st">"Biomarker score distribution in pancreatic cancer"</span><span class="op">)</span>

<span class="co"># Identify abstracts reporting miRNAs as biomarker in pancreatic cancer</span>
<span class="va">panc_biomarker</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_score_biomarker.html">calculate_score_biomarker</a></span><span class="op">(</span><span class="va">df_panc</span>,
                                           threshold <span class="op">=</span> <span class="fl">6</span>,
                                           indicate <span class="op">=</span> <span class="cn">TRUE</span>,
                                           discard <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Plot top potential biomarker miRNAs in pancreatic cancer</span>
<span class="fu"><a href="../reference/plot_mir_count.html">plot_mir_count</a></span><span class="op">(</span><span class="va">panc_biomarker</span>,
               title <span class="op">=</span> <span class="st">"Likely biomarker in pancreatic cancer"</span>,
               top <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-45-1.png" width="50%"><img src="miRetrieve_files/figure-html/unnamed-chunk-45-2.png" width="50%"></p>
<p>miR-21 is mentioned in about 35 abstracts reporting miRNAs as biomarkers in pancreatic cancer. This suggests that miR-21 is most likely a biomarker in pancreatic cancer, but hence no biomarker that is specific for either CRC or pancreatic cancer.</p>
</div>
<div id="target-interactions-of-mir-21-in-crc-and-pancreatic-cancer" class="section level2">
<h2 class="hasAnchor">
<a href="#target-interactions-of-mir-21-in-crc-and-pancreatic-cancer" class="anchor"></a>Target interactions of miR-21 in CRC and pancreatic cancer</h2>
<p>As miR-21 seems to be a biomarker in CRC and pancreatic, we determine if miR-21 also shares miRNA-target interactions in both tumor entities.</p>
<p>First, we combine the CRC and pancreatic cancer data frames with <code><a href="../reference/combine_df.html">combine_df()</a></code>. Next, we look up the experimentally validated targets by adding the <em>miRTarBase</em><span class="citation">[<a href="#ref-mirtarbase" role="doc-biblioref">4</a>]</span> database with <code><a href="../reference/join_targets.html">join_targets()</a></code>.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Combine CRC and pancreatic cancer data frames</span>
<span class="va">df_crc_panc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_df.html">combine_df</a></span><span class="op">(</span><span class="va">df_crc</span>, <span class="va">df_panc</span><span class="op">)</span>

<span class="co"># Path to miRTarBase (see "References")</span>
<span class="va">target_db</span> <span class="op">&lt;-</span> <span class="st">"miRTarBase_MTI.xlsx"</span>

<span class="co"># Add miRTarBase targets to `df_crc_panc`</span>
<span class="va">df_targets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/join_targets.html">join_targets</a></span><span class="op">(</span><span class="va">df_crc_panc</span>,
                           excel_file <span class="op">=</span> <span class="va">target_db</span>,
                           col.pmid.excel <span class="op">=</span> <span class="st">"References (PMID)"</span>,
                           col.target.excel <span class="op">=</span> <span class="st">"Target Gene"</span>,
                           col.mir.excel <span class="op">=</span> <span class="st">"miRNA"</span>,
                           stem_mir_excel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Finally, we plot the targets of miR-21 with <code><a href="../reference/plot_target_mir_scatter.html">plot_target_mir_scatter()</a></code>.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot top targets for miR-21 in CRC and pancreatic cancer</span>
<span class="fu"><a href="../reference/plot_target_mir_scatter.html">plot_target_mir_scatter</a></span><span class="op">(</span><span class="va">df_targets</span>,
                        mir <span class="op">=</span> <span class="st">"miR-21"</span>,
                        col.mir <span class="op">=</span> <span class="va">miRNA_excel</span>,
                        top <span class="op">=</span> <span class="fl">10</span>,
                        filter_for <span class="op">=</span> <span class="st">"target"</span>,
                        title <span class="op">=</span> <span class="st">"miRNA-target interaction of miR-21 in CRC and pancreatic cancer"</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-47-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>According to <em>miRTarBase</em>, miR-21 shares at least three targets across CRC and pancreatic cancer, namely <em>PDCD4</em>, <em>PTEN</em>, and <em>RPS7</em>. However, miR-21 is also known to target only specific genes in one subject so far, such as <em>RASA1</em>, <em>SPRY2</em>, or <em>TIAM1</em> in CRC, or <em>BCL2</em>, <em>FASLG</em>, <em>HIF1A</em>, <em>MMP2</em>, or <em>MMP9</em> in pancreatic cancer.</p>
<p>For our project, it is therefore interesting to investigate if miR-21 regulates also one of the uniquely in pancreatic cancer reported targets in CRC.</p>
</div>
<div id="top-mirna-target-interactions-in-crc-and-pancreatic-cancer" class="section level2">
<h2 class="hasAnchor">
<a href="#top-mirna-target-interactions-in-crc-and-pancreatic-cancer" class="anchor"></a>Top miRNA-target interactions in CRC and pancreatic cancer</h2>
<p>Lastly, we identify the three genes that are targeted by most miRNAs in CRC and pancreatic cancer with <code>plot_mir_scatter()</code>.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot top 3 miRNA targets in CRC and pancreatic cancer</span>
<span class="fu"><a href="../reference/plot_target_mir_scatter.html">plot_target_mir_scatter</a></span><span class="op">(</span><span class="va">df_targets</span>,
                        col.mir <span class="op">=</span> <span class="va">miRNA_excel</span>,
                        top <span class="op">=</span> <span class="fl">3</span>,
                        filter_for <span class="op">=</span> <span class="st">"target"</span>,
                        title <span class="op">=</span> <span class="st">"Top 3 miRNA targets in CRC and pancreatic cancer"</span><span class="op">)</span></code></pre></div>
<p><img src="miRetrieve_files/figure-html/unnamed-chunk-48-1.png" width="50%" style="display: block; margin: auto;"></p>
<p>According to <em>miRTarBase</em>, <em>PTEN</em> and <em>PDCD4</em> are both genes targeted by miR-21 in CRC and pancreatic cancer. As these were just toy datasets, an interesting next step would be to apply miRTarBase to a larger collection of abstracts and compare similarities and difference of miRNA targets between tumor entities!</p>
</div>
<div id="conclusion" class="section level2">
<h2 class="hasAnchor">
<a href="#conclusion" class="anchor"></a>Conclusion</h2>
<p>With few lines of code, we determined that miR-21 is a frequently mentioned and thus most likely well investigated miRNA in CRC. Furthermore, we revealed that miR-21 is possibly a non-specific biomarker for CRC and pancreatic cancer. Next to that, we gained insight into common and distinct targets of miR-21 in both diseases, while observing that <em>PTEN</em> and <em>PDCD4</em> are targeted by miR-21 in both fields.</p>
<p>While the mention of a miRNA and the terms it is associated with in an abstract need to be interpreted carefully, text mining miRNAs with miRetrieve provides the opportunity to generate and test hypotheses on the fly, which can serve as a starting point for subsequent research.</p>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p><em>Disclaimer: The links to external websites provided below are for further information. Although we try to ensure these links are accurate and safe, we cannot take responsibility for pages maintained by external providers.</em></p>
<div id="refs" class="references csl-bib-body">
<div id="ref-tidytext" class="csl-entry">
<div class="csl-left-margin">[1] </div>
<div class="csl-right-inline">Silge J, Robinson D. Tidytext: Text mining and analysis using tidy data principles in r. JOSS 2016;1. <a href="https://doi.org/10.21105/joss.00037">https://doi.org/10.21105/joss.00037</a>.</div>
</div>
<div id="ref-topicmodels" class="csl-entry">
<div class="csl-left-margin">[2] </div>
<div class="csl-right-inline">Grün B, Hornik K. <span class="nocase">topicmodels</span>: An <span>R</span> package for fitting topic models. Journal of Statistical Software 2011;40:1–30. <a href="https://doi.org/10.18637/jss.v040.i13">https://doi.org/10.18637/jss.v040.i13</a>.</div>
</div>
<div id="ref-wordcloud" class="csl-entry">
<div class="csl-left-margin">[3] </div>
<div class="csl-right-inline">Fellows I. Wordcloud: Word clouds. 2018.</div>
</div>
<div id="ref-mirtarbase" class="csl-entry">
<div class="csl-left-margin">[4] </div>
<div class="csl-right-inline">Huang HY, Lin YC, Li J, Huang KY, Shrestha S, Hong HC, et al. miRTarBase 2020: Updates to the experimentally validated microRNA-target interaction database. Nucleic Acids Res 2020;48:D148–54. <a href="https://doi.org/10.1093/nar/gkz896">https://doi.org/10.1093/nar/gkz896</a>.</div>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="https://www.lexico.com/en/definition/natural_language_processing" class="uri">https://www.lexico.com/en/definition/natural_language_processing</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="https://en.wikipedia.org/wiki/Regular_expression" class="uri">https://en.wikipedia.org/wiki/Regular_expression</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="http://www.mirbase.org" class="uri">http://www.mirbase.org</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>English stop words are based on <code><a href="https://rdrr.io/pkg/tidytext/man/stop_words.html">tidytext::stop_words</a></code><span class="citation">[<a href="#ref-tidytext" role="doc-biblioref">1</a>]</span>.<a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>English stop words removed for n-gram tokenization are based on <code><a href="https://rdrr.io/pkg/tidytext/man/stop_words.html">tidytext::stop_words</a></code><span class="citation">[<a href="#ref-tidytext" role="doc-biblioref">1</a>]</span>.<a href="#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p>If there is interest in a miRNA with trailing letters, e.g. <em>miR-23a</em> or <em>miR-23b</em>, an alternative is to plot the top associated 2-grams of the miRNA with <code><a href="../reference/plot_mir_terms.html">plot_mir_terms()</a></code>, which can display the most frequent combination of miRNA/trailing letter in literature<a href="#fnref6" class="footnote-back">↩︎</a></p></li>
<li id="fn7"><p><code><a href="../reference/plot_wordcloud.html">plot_wordcloud()</a></code> is based on the <strong>wordcloud</strong> package<span class="citation">[<a href="#ref-wordcloud" role="doc-biblioref">3</a>]</span>.<a href="#fnref7" class="footnote-back">↩︎</a></p></li>
<li id="fn8"><p>The plot created by <code><a href="../reference/compare_mir_terms_log2.html">compare_mir_terms_log2()</a></code> is greatly inspired by <em>Text Mining with R</em> by Silge and Robinson<span class="citation">[<a href="#ref-tidytext" role="doc-biblioref">1</a>]</span>.<a href="#fnref8" class="footnote-back">↩︎</a></p></li>
<li id="fn9"><p>The plot generated by <code><a href="../reference/compare_mir_terms_scatter.html">compare_mir_terms_scatter()</a></code> is greatly inspired by <em>Text Mining with R</em> by Silge and Robinson<span class="citation">[<a href="#ref-tidytext" role="doc-biblioref">1</a>]</span>.<a href="#fnref9" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Julian Friedrich, Hans-Peter Hammes, Guido Krenning.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
